// @multiversx/sdk-dapp/__commonjs/UI/SignTransactionsModals/SignWithDeviceModal/SignWithDeviceModal.js
"use strict";var np=Object.create;var Co=Object.defineProperty,tp=Object.defineProperties,rp=Object.getOwnPropertyDescriptor,ap=Object.getOwnPropertyDescriptors,ip=Object.getOwnPropertyNames,ln=Object.getOwnPropertySymbols,cp=Object.getPrototypeOf,ut=Object.prototype.hasOwnProperty,Kr=Object.prototype.propertyIsEnumerable;var qr=(e,o,n)=>o in e?Co(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,x=(e,o)=>{for(var n in o||(o={}))ut.call(o,n)&&qr(e,n,o[n]);if(ln)for(var n of ln(o))Kr.call(o,n)&&qr(e,n,o[n]);return e},k=(e,o)=>tp(e,ap(o));var _n=(e,o)=>{var n={};for(var r in e)ut.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&ln)for(var r of ln(e))o.indexOf(r)<0&&Kr.call(e,r)&&(n[r]=e[r]);return n};var w=(e,o)=>()=>(e&&(o=e(e=0)),o);var Q=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Me=(e,o)=>{for(var n in o)Co(e,n,{get:o[n],enumerable:!0})},Xr=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let s of ip(o))!ut.call(e,s)&&s!==n&&Co(e,s,{get:()=>o[s],enumerable:!(r=rp(o,s))||r.enumerable});return e};var y=(e,o,n)=>(n=e!=null?np(cp(e)):{},Xr(o||!e||!e.__esModule?Co(n,"default",{value:e,enumerable:!0}):n,e)),No=e=>Xr(Co({},"__esModule",{value:!0}),e);var v=(e,o,n)=>new Promise((r,s)=>{var p=l=>{try{d(n.next(l))}catch(u){s(u)}},m=l=>{try{d(n.throw(l))}catch(u){s(u)}},d=l=>l.done?r(l.value):Promise.resolve(l.value).then(p,m);d((n=n.apply(e,o)).next())});var Jr=Q(gn=>{"use strict";t();gn.byteLength=pp;gn.toByteArray=dp;gn.fromByteArray=gp;var le=[],ae=[],sp=typeof Uint8Array!="undefined"?Uint8Array:Array,ft="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Re=0,Yr=ft.length;Re<Yr;++Re)le[Re]=ft[Re],ae[ft.charCodeAt(Re)]=Re;var Re,Yr;ae["-".charCodeAt(0)]=62;ae["_".charCodeAt(0)]=63;function Zr(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function pp(e){var o=Zr(e),n=o[0],r=o[1];return(n+r)*3/4-r}function mp(e,o,n){return(o+n)*3/4-n}function dp(e){var o,n=Zr(e),r=n[0],s=n[1],p=new sp(mp(e,r,s)),m=0,d=s>0?r-4:r,l;for(l=0;l<d;l+=4)o=ae[e.charCodeAt(l)]<<18|ae[e.charCodeAt(l+1)]<<12|ae[e.charCodeAt(l+2)]<<6|ae[e.charCodeAt(l+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return s===2&&(o=ae[e.charCodeAt(l)]<<2|ae[e.charCodeAt(l+1)]>>4,p[m++]=o&255),s===1&&(o=ae[e.charCodeAt(l)]<<10|ae[e.charCodeAt(l+1)]<<4|ae[e.charCodeAt(l+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function lp(e){return le[e>>18&63]+le[e>>12&63]+le[e>>6&63]+le[e&63]}function _p(e,o,n){for(var r,s=[],p=o;p<n;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),s.push(lp(r));return s.join("")}function gp(e){for(var o,n=e.length,r=n%3,s=[],p=16383,m=0,d=n-r;m<d;m+=p)s.push(_p(e,m,m+p>d?d:m+p));return r===1?(o=e[n-1],s.push(le[o>>2]+le[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],s.push(le[o>>10]+le[o>>4&63]+le[o<<2&63]+"=")),s.join("")}});var Qr=Q(Tt=>{t();Tt.read=function(e,o,n,r,s){var p,m,d=s*8-r-1,l=(1<<d)-1,u=l>>1,f=-7,T=n?s-1:0,b=n?-1:1,h=e[o+T];for(T+=b,p=h&(1<<-f)-1,h>>=-f,f+=d;f>0;p=p*256+e[o+T],T+=b,f-=8);for(m=p&(1<<-f)-1,p>>=-f,f+=r;f>0;m=m*256+e[o+T],T+=b,f-=8);if(p===0)p=1-u;else{if(p===l)return m?NaN:(h?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-u}return(h?-1:1)*m*Math.pow(2,p-r)};Tt.write=function(e,o,n,r,s,p){var m,d,l,u=p*8-s-1,f=(1<<u)-1,T=f>>1,b=s===23?Math.pow(2,-24)-Math.pow(2,-77):0,h=r?0:p-1,A=r?1:-1,D=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(d=isNaN(o)?1:0,m=f):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+T>=1?o+=b/l:o+=b*Math.pow(2,1-T),o*l>=2&&(m++,l/=2),m+T>=f?(d=0,m=f):m+T>=1?(d=(o*l-1)*Math.pow(2,s),m=m+T):(d=o*Math.pow(2,T-1)*Math.pow(2,s),m=0));s>=8;e[n+h]=d&255,h+=A,d/=256,s-=8);for(m=m<<s|d,u+=s;u>0;e[n+h]=m&255,h+=A,m/=256,u-=8);e[n+h-A]|=D*128}});var la=Q(oo=>{"use strict";t();var ht=Jr(),eo=Qr(),ea=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;oo.Buffer=g;oo.SlowBuffer=yp;oo.INSPECT_MAX_BYTES=50;var un=2147483647;oo.kMaxLength=un;g.TYPED_ARRAY_SUPPORT=up();!g.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function up(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(g.prototype,"parent",{enumerable:!0,get:function(){if(!!g.isBuffer(this))return this.buffer}});Object.defineProperty(g.prototype,"offset",{enumerable:!0,get:function(){if(!!g.isBuffer(this))return this.byteOffset}});function be(e){if(e>un)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,g.prototype),o}function g(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return St(e)}return ta(e,o,n)}g.poolSize=8192;function ta(e,o,n){if(typeof e=="string")return Tp(e,o);if(ArrayBuffer.isView(e))return hp(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(_e(e,ArrayBuffer)||e&&_e(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(_e(e,SharedArrayBuffer)||e&&_e(e.buffer,SharedArrayBuffer)))return yt(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return g.from(r,o,n);var s=xp(e);if(s)return s;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return g.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}g.from=function(e,o,n){return ta(e,o,n)};Object.setPrototypeOf(g.prototype,Uint8Array.prototype);Object.setPrototypeOf(g,Uint8Array);function ra(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function fp(e,o,n){return ra(e),e<=0?be(e):o!==void 0?typeof n=="string"?be(e).fill(o,n):be(e).fill(o):be(e)}g.alloc=function(e,o,n){return fp(e,o,n)};function St(e){return ra(e),be(e<0?0:wt(e)|0)}g.allocUnsafe=function(e){return St(e)};g.allocUnsafeSlow=function(e){return St(e)};function Tp(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!g.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=aa(e,o)|0,r=be(n),s=r.write(e,o);return s!==n&&(r=r.slice(0,s)),r}function xt(e){for(var o=e.length<0?0:wt(e.length)|0,n=be(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function hp(e){if(_e(e,Uint8Array)){var o=new Uint8Array(e);return yt(o.buffer,o.byteOffset,o.byteLength)}return xt(e)}function yt(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,g.prototype),r}function xp(e){if(g.isBuffer(e)){var o=wt(e.length)|0,n=be(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||vt(e.length)?be(0):xt(e);if(e.type==="Buffer"&&Array.isArray(e.data))return xt(e.data)}function wt(e){if(e>=un)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+un.toString(16)+" bytes");return e|0}function yp(e){return+e!=e&&(e=0),g.alloc(+e)}g.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==g.prototype};g.compare=function(o,n){if(_e(o,Uint8Array)&&(o=g.from(o,o.offset,o.byteLength)),_e(n,Uint8Array)&&(n=g.from(n,n.offset,n.byteLength)),!g.isBuffer(o)||!g.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,s=n.length,p=0,m=Math.min(r,s);p<m;++p)if(o[p]!==n[p]){r=o[p],s=n[p];break}return r<s?-1:s<r?1:0};g.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};g.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return g.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var s=g.allocUnsafe(n),p=0;for(r=0;r<o.length;++r){var m=o[r];if(_e(m,Uint8Array))p+m.length>s.length?g.from(m).copy(s,p):Uint8Array.prototype.set.call(s,m,p);else if(g.isBuffer(m))m.copy(s,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return s};function aa(e,o){if(g.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||_e(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var s=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return bt(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return da(e).length;default:if(s)return r?-1:bt(e).length;o=(""+o).toLowerCase(),s=!0}}g.byteLength=aa;function bp(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Np(this,o,n);case"utf8":case"utf-8":return ca(this,o,n);case"ascii":return Ep(this,o,n);case"latin1":case"binary":return Cp(this,o,n);case"base64":return Ap(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Dp(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}g.prototype._isBuffer=!0;function Oe(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}g.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)Oe(this,n,n+1);return this};g.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)Oe(this,n,n+3),Oe(this,n+1,n+2);return this};g.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)Oe(this,n,n+7),Oe(this,n+1,n+6),Oe(this,n+2,n+5),Oe(this,n+3,n+4);return this};g.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?ca(this,0,o):bp.apply(this,arguments)};g.prototype.toLocaleString=g.prototype.toString;g.prototype.equals=function(o){if(!g.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:g.compare(this,o)===0};g.prototype.inspect=function(){var o="",n=oo.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};ea&&(g.prototype[ea]=g.prototype.inspect);g.prototype.compare=function(o,n,r,s,p){if(_e(o,Uint8Array)&&(o=g.from(o,o.offset,o.byteLength)),!g.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),s===void 0&&(s=0),p===void 0&&(p=this.length),n<0||r>o.length||s<0||p>this.length)throw new RangeError("out of range index");if(s>=p&&n>=r)return 0;if(s>=p)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,s>>>=0,p>>>=0,this===o)return 0;for(var m=p-s,d=r-n,l=Math.min(m,d),u=this.slice(s,p),f=o.slice(n,r),T=0;T<l;++T)if(u[T]!==f[T]){m=u[T],d=f[T];break}return m<d?-1:d<m?1:0};function ia(e,o,n,r,s){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,vt(n)&&(n=s?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(s)return-1;n=e.length-1}else if(n<0)if(s)n=0;else return-1;if(typeof o=="string"&&(o=g.from(o,r)),g.isBuffer(o))return o.length===0?-1:oa(e,o,n,r,s);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?s?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):oa(e,[o],n,r,s);throw new TypeError("val must be string, number or Buffer")}function oa(e,o,n,r,s){var p=1,m=e.length,d=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,d/=2,n/=2}function l(h,A){return p===1?h[A]:h.readUInt16BE(A*p)}var u;if(s){var f=-1;for(u=n;u<m;u++)if(l(e,u)===l(o,f===-1?0:u-f)){if(f===-1&&(f=u),u-f+1===d)return f*p}else f!==-1&&(u-=u-f),f=-1}else for(n+d>m&&(n=m-d),u=n;u>=0;u--){for(var T=!0,b=0;b<d;b++)if(l(e,u+b)!==l(o,b)){T=!1;break}if(T)return u}return-1}g.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};g.prototype.indexOf=function(o,n,r){return ia(this,o,n,r,!0)};g.prototype.lastIndexOf=function(o,n,r){return ia(this,o,n,r,!1)};function Sp(e,o,n,r){n=Number(n)||0;var s=e.length-n;r?(r=Number(r),r>s&&(r=s)):r=s;var p=o.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var d=parseInt(o.substr(m*2,2),16);if(vt(d))return m;e[n+m]=d}return m}function wp(e,o,n,r){return fn(bt(o,e.length-n),e,n,r)}function vp(e,o,n,r){return fn(Rp(o),e,n,r)}function kp(e,o,n,r){return fn(da(o),e,n,r)}function Ip(e,o,n,r){return fn(Op(o,e.length-n),e,n,r)}g.prototype.write=function(o,n,r,s){if(n===void 0)s="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")s=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,s===void 0&&(s="utf8")):(s=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-n;if((r===void 0||r>p)&&(r=p),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");s||(s="utf8");for(var m=!1;;)switch(s){case"hex":return Sp(this,o,n,r);case"utf8":case"utf-8":return wp(this,o,n,r);case"ascii":case"latin1":case"binary":return vp(this,o,n,r);case"base64":return kp(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ip(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+s);s=(""+s).toLowerCase(),m=!0}};g.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Ap(e,o,n){return o===0&&n===e.length?ht.fromByteArray(e):ht.fromByteArray(e.slice(o,n))}function ca(e,o,n){n=Math.min(e.length,n);for(var r=[],s=o;s<n;){var p=e[s],m=null,d=p>239?4:p>223?3:p>191?2:1;if(s+d<=n){var l,u,f,T;switch(d){case 1:p<128&&(m=p);break;case 2:l=e[s+1],(l&192)===128&&(T=(p&31)<<6|l&63,T>127&&(m=T));break;case 3:l=e[s+1],u=e[s+2],(l&192)===128&&(u&192)===128&&(T=(p&15)<<12|(l&63)<<6|u&63,T>2047&&(T<55296||T>57343)&&(m=T));break;case 4:l=e[s+1],u=e[s+2],f=e[s+3],(l&192)===128&&(u&192)===128&&(f&192)===128&&(T=(p&15)<<18|(l&63)<<12|(u&63)<<6|f&63,T>65535&&T<1114112&&(m=T))}}m===null?(m=65533,d=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),s+=d}return Lp(r)}var na=4096;function Lp(e){var o=e.length;if(o<=na)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=na));return n}function Ep(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]&127);return r}function Cp(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]);return r}function Np(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var s="",p=o;p<n;++p)s+=Fp[e[p]];return s}function Dp(e,o,n){for(var r=e.slice(o,n),s="",p=0;p<r.length-1;p+=2)s+=String.fromCharCode(r[p]+r[p+1]*256);return s}g.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var s=this.subarray(o,n);return Object.setPrototypeOf(s,g.prototype),s};function H(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}g.prototype.readUintLE=g.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||H(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return s};g.prototype.readUintBE=g.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||H(o,n,this.length);for(var s=this[o+--n],p=1;n>0&&(p*=256);)s+=this[o+--n]*p;return s};g.prototype.readUint8=g.prototype.readUInt8=function(o,n){return o=o>>>0,n||H(o,1,this.length),this[o]};g.prototype.readUint16LE=g.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||H(o,2,this.length),this[o]|this[o+1]<<8};g.prototype.readUint16BE=g.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||H(o,2,this.length),this[o]<<8|this[o+1]};g.prototype.readUint32LE=g.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||H(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};g.prototype.readUint32BE=g.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||H(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};g.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||H(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return p*=128,s>=p&&(s-=Math.pow(2,8*n)),s};g.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||H(o,n,this.length);for(var s=n,p=1,m=this[o+--s];s>0&&(p*=256);)m+=this[o+--s]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*n)),m};g.prototype.readInt8=function(o,n){return o=o>>>0,n||H(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};g.prototype.readInt16LE=function(o,n){o=o>>>0,n||H(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};g.prototype.readInt16BE=function(o,n){o=o>>>0,n||H(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};g.prototype.readInt32LE=function(o,n){return o=o>>>0,n||H(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};g.prototype.readInt32BE=function(o,n){return o=o>>>0,n||H(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};g.prototype.readFloatLE=function(o,n){return o=o>>>0,n||H(o,4,this.length),eo.read(this,o,!0,23,4)};g.prototype.readFloatBE=function(o,n){return o=o>>>0,n||H(o,4,this.length),eo.read(this,o,!1,23,4)};g.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||H(o,8,this.length),eo.read(this,o,!0,52,8)};g.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||H(o,8,this.length),eo.read(this,o,!1,52,8)};function Z(e,o,n,r,s,p){if(!g.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>s||o<p)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}g.prototype.writeUintLE=g.prototype.writeUIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;Z(this,o,n,r,p,0)}var m=1,d=0;for(this[n]=o&255;++d<r&&(m*=256);)this[n+d]=o/m&255;return n+r};g.prototype.writeUintBE=g.prototype.writeUIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;Z(this,o,n,r,p,0)}var m=r-1,d=1;for(this[n+m]=o&255;--m>=0&&(d*=256);)this[n+m]=o/d&255;return n+r};g.prototype.writeUint8=g.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,1,255,0),this[n]=o&255,n+1};g.prototype.writeUint16LE=g.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};g.prototype.writeUint16BE=g.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};g.prototype.writeUint32LE=g.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};g.prototype.writeUint32BE=g.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};g.prototype.writeIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);Z(this,o,n,r,p-1,-p)}var m=0,d=1,l=0;for(this[n]=o&255;++m<r&&(d*=256);)o<0&&l===0&&this[n+m-1]!==0&&(l=1),this[n+m]=(o/d>>0)-l&255;return n+r};g.prototype.writeIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);Z(this,o,n,r,p-1,-p)}var m=r-1,d=1,l=0;for(this[n+m]=o&255;--m>=0&&(d*=256);)o<0&&l===0&&this[n+m+1]!==0&&(l=1),this[n+m]=(o/d>>0)-l&255;return n+r};g.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};g.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};g.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};g.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};g.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||Z(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function sa(e,o,n,r,s,p){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function pa(e,o,n,r,s){return o=+o,n=n>>>0,s||sa(e,o,n,4,34028234663852886e22,-34028234663852886e22),eo.write(e,o,n,r,23,4),n+4}g.prototype.writeFloatLE=function(o,n,r){return pa(this,o,n,!0,r)};g.prototype.writeFloatBE=function(o,n,r){return pa(this,o,n,!1,r)};function ma(e,o,n,r,s){return o=+o,n=n>>>0,s||sa(e,o,n,8,17976931348623157e292,-17976931348623157e292),eo.write(e,o,n,r,52,8),n+8}g.prototype.writeDoubleLE=function(o,n,r){return ma(this,o,n,!0,r)};g.prototype.writeDoubleBE=function(o,n,r){return ma(this,o,n,!1,r)};g.prototype.copy=function(o,n,r,s){if(!g.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!s&&s!==0&&(s=this.length),n>=o.length&&(n=o.length),n||(n=0),s>0&&s<r&&(s=r),s===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(s<0)throw new RangeError("sourceEnd out of bounds");s>this.length&&(s=this.length),o.length-n<s-r&&(s=o.length-n+r);var p=s-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,s):Uint8Array.prototype.set.call(o,this.subarray(r,s),n),p};g.prototype.fill=function(o,n,r,s){if(typeof o=="string"){if(typeof n=="string"?(s=n,n=0,r=this.length):typeof r=="string"&&(s=r,r=this.length),s!==void 0&&typeof s!="string")throw new TypeError("encoding must be a string");if(typeof s=="string"&&!g.isEncoding(s))throw new TypeError("Unknown encoding: "+s);if(o.length===1){var p=o.charCodeAt(0);(s==="utf8"&&p<128||s==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var d=g.isBuffer(o)?o:g.from(o,s),l=d.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=d[m%l]}return this};var Pp=/[^+/0-9A-Za-z-_]/g;function Mp(e){if(e=e.split("=")[0],e=e.trim().replace(Pp,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function bt(e,o){o=o||1/0;for(var n,r=e.length,s=null,p=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!s){if(n>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&p.push(239,191,189);continue}s=n;continue}if(n<56320){(o-=3)>-1&&p.push(239,191,189),s=n;continue}n=(s-55296<<10|n-56320)+65536}else s&&(o-=3)>-1&&p.push(239,191,189);if(s=null,n<128){if((o-=1)<0)break;p.push(n)}else if(n<2048){if((o-=2)<0)break;p.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;p.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;p.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return p}function Rp(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function Op(e,o){for(var n,r,s,p=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,s=n%256,p.push(s),p.push(r);return p}function da(e){return ht.toByteArray(Mp(e))}function fn(e,o,n,r){for(var s=0;s<r&&!(s+n>=o.length||s>=e.length);++s)o[s+n]=e[s];return s}function _e(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function vt(e){return e!==e}var Fp=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,s=0;s<16;++s)o[r+s]=e[n]+e[s];return o}()});var Ta=Q((s_,fa)=>{t();var U=fa.exports={},ge,ue;function kt(){throw new Error("setTimeout has not been defined")}function It(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ge=setTimeout:ge=kt}catch(e){ge=kt}try{typeof clearTimeout=="function"?ue=clearTimeout:ue=It}catch(e){ue=It}})();function _a(e){if(ge===setTimeout)return setTimeout(e,0);if((ge===kt||!ge)&&setTimeout)return ge=setTimeout,setTimeout(e,0);try{return ge(e,0)}catch(o){try{return ge.call(null,e,0)}catch(n){return ge.call(this,e,0)}}}function Bp(e){if(ue===clearTimeout)return clearTimeout(e);if((ue===It||!ue)&&clearTimeout)return ue=clearTimeout,clearTimeout(e);try{return ue(e)}catch(o){try{return ue.call(null,e)}catch(n){return ue.call(this,e)}}}var Se=[],no=!1,Fe,Tn=-1;function Up(){!no||!Fe||(no=!1,Fe.length?Se=Fe.concat(Se):Tn=-1,Se.length&&ga())}function ga(){if(!no){var e=_a(Up);no=!0;for(var o=Se.length;o;){for(Fe=Se,Se=[];++Tn<o;)Fe&&Fe[Tn].run();Tn=-1,o=Se.length}Fe=null,no=!1,Bp(e)}}U.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];Se.push(new ua(e,o)),Se.length===1&&!no&&_a(ga)};function ua(e,o){this.fun=e,this.array=o}ua.prototype.run=function(){this.fun.apply(null,this.array)};U.title="browser";U.browser=!0;U.env={};U.argv=[];U.version="";U.versions={};function we(){}U.on=we;U.addListener=we;U.once=we;U.off=we;U.removeListener=we;U.removeAllListeners=we;U.emit=we;U.prependListener=we;U.prependOnceListener=we;U.listeners=function(e){return[]};U.binding=function(e){throw new Error("process.binding is not supported")};U.cwd=function(){return"/"};U.chdir=function(e){throw new Error("process.chdir is not supported")};U.umask=function(){return 0}});var i,c,Wp,a,t=w(()=>{i=y(la()),c=y(Ta()),Wp=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object),a=Wp});var At,Be=w(()=>{"use strict";t();At="Transaction canceled"});var ha=w(()=>{"use strict";t()});var xa=w(()=>{"use strict";t()});var ya=w(()=>{"use strict";t()});var O=w(()=>{"use strict";t()});var ba=w(()=>{"use strict";t()});var Sa=w(()=>{"use strict";t()});var J=w(()=>{"use strict";t()});var wa=w(()=>{"use strict";t()});var va=w(()=>{"use strict";t()});var $=w(()=>{"use strict";t();ha();xa();ya();O();ba();Sa();J();wa();va()});var Lt,ka,L_,Ia,E_,Aa,C_,N_,D_,La=w(()=>{"use strict";t();$();Lt={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",apiAddress:"https://devnet-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",apiAddress:"https://testnet-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",apiAddress:"https://api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",apiTimeout:"4000"}},{chainId:ka,egldLabel:L_}=Lt["devnet"],{chainId:Ia,egldLabel:E_}=Lt["testnet"],{chainId:Aa,egldLabel:C_}=Lt["mainnet"],N_={["devnet"]:ka,["testnet"]:Ia,["mainnet"]:Aa},D_={[ka]:"devnet",[Ia]:"testnet",[Aa]:"mainnet"}});var Et,Ct=w(()=>{"use strict";t();Et={36864:{code:"codeSuccess",message:"Success"},27013:{code:"ERR_USER_DENIED",message:"Rejected by user"},27904:{code:"ERR_UNKNOWN_INSTRUCTION",message:"Unknown instruction"},28160:{code:"ERR_WRONG_CLA",message:"Wrong CLA"},28161:{code:"ERR_INVALID_ARGUMENTS",message:"Invalid arguments"},28162:{code:"ERR_INVALID_MESSAGE",message:"Invalid message"},28163:{code:"ERR_INVALID_P1",message:"Invalid P1"},28164:{code:"ERR_MESSAGE_TOO_LONG",message:"Message too long"},28165:{code:"ERR_RECEIVER_TOO_LONG",message:"Receiver too long"},28166:{code:"ERR_AMOUNT_TOO_LONG",message:"Amount too long"},28167:{code:"ERR_CONTRACT_DATA_DISABLED",message:"Contract data disabled in app options"},28168:{code:"ERR_MESSAGE_INCOMPLETE",message:"Message incomplete"},28176:{code:"ERR_SIGNATURE_FAILED",message:"Signature failed"},28169:{code:"ERR_WRONG_TX_VERSION",message:"Wrong TX version"},28170:{code:"ERR_NONCE_TOO_LONG",message:"Nonce too long"},28171:{code:"ERR_INVALID_AMOUNT",message:"Invalid amount"},28172:{code:"ERR_INVALID_FEE",message:"Invalid fee"},28173:{code:"ERR_PRETTY_FAILED",message:"Pretty failed"},28174:{code:"ERR_DATA_TOO_LONG",message:"Data too long"},28175:{code:"ERR_WRONG_TX_OPTIONS",message:"Invalid transaction options"},28177:{code:"ERR_SIGN_TX_DEPRECATED",message:"Regular transaction signing is deprecated in this version. Use hash signing."}}});var Ea=w(()=>{"use strict";t()});var Do,Nt=w(()=>{"use strict";t();Do="transaction-status-toast-id"});var fe,Ca=w(()=>{"use strict";t();fe=require("@multiversx/sdk-web-wallet-provider")});var Na=w(()=>{"use strict";t()});var Po,hn,Te,xn,Da,Pa,Mo,yn,Ma,Ra,Ue,P=w(()=>{"use strict";t();Be();La();Ct();Ea();Nt();Ca();Na();Po=5e4,hn=1e9,Te=18,xn=4,Da=1,Pa="/dapp/init",Mo="signSession",yn="logout",Ma="login",Ra="N/A",Ue="0"});var Ro,bn=w(()=>{"use strict";t();Ro=()=>Date.now()/1e3});var Oa=w(()=>{"use strict";t()});var Fa=w(()=>{"use strict";t()});var Dt=w(()=>{"use strict";t();bn();Oa();Fa()});var Mt={};Me(Mt,{getItem:()=>$p,localStorageKeys:()=>Ce,removeItem:()=>Vp,setItem:()=>Hp});var Ce,Pt,Hp,$p,Vp,to=w(()=>{"use strict";t();Dt();Ce={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event"},Pt=typeof localStorage!="undefined",Hp=({key:e,data:o,expires:n})=>{!Pt||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},$p=e=>{if(!Pt)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Ro()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},Vp=e=>{!Pt||localStorage.removeItem(String(e))}});var Rt={};Me(Rt,{clear:()=>Ga,getItem:()=>Ua,removeItem:()=>Wa,setItem:()=>Ba,storage:()=>jp});var Ba,Ua,Wa,Ga,jp,Ha=w(()=>{"use strict";t();Ba=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Ua=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Wa=e=>sessionStorage.removeItem(String(e)),Ga=()=>sessionStorage.clear(),jp={setItem:Ba,getItem:Ua,removeItem:Wa,clear:Ga}});var We,ro=w(()=>{"use strict";t();to();Ha();We={session:Rt,local:Mt}});var Ot,V,he,K=w(()=>{"use strict";t();Ot=require("@reduxjs/toolkit");P();V=(0,Ot.createAction)(yn),he=(0,Ot.createAction)(Ma,e=>({payload:e}))});var Bt,$a,Va,Sn,Ft,ja,Ut,mg,Wt,dg,zp,qp,lg,_g,gg,Kp,wn,vn=w(()=>{"use strict";t();Bt=require("@multiversx/sdk-core"),$a=require("@reduxjs/toolkit"),Va=require("redux-persist");P();ro();to();K();Sn={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:Ue},Ft={address:"",websocketEvent:null,accounts:{"":Sn},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},ja=(0,$a.createSlice)({name:"accountInfoSlice",initialState:Ft,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new Bt.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:Sn},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(V,()=>(We.local.removeItem(Ce.loginExpiresAt),Ft)),e.addCase(he,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new Bt.Address(r).hex()}),e.addCase(Va.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:s,shard:p,accounts:m,publicKey:d}=r;o.address=s,o.shard=p;let l=m&&s in m;o.accounts=l?m:Ft.accounts,o.publicKey=d})}}),{setAccount:Ut,setAddress:mg,setAccountNonce:Wt,setAccountShard:dg,setLedgerAccount:zp,updateLedgerAccount:qp,setWalletConnectAccount:lg,setIsAccountLoading:_g,setAccountLoadingError:gg,setWebsocketEvent:Kp}=ja.actions,wn=ja.reducer});function Oo(){return new Date().setHours(new Date().getHours()+24)}function Fo(e){We.local.setItem({key:Ce.loginExpiresAt,data:e,expires:e})}var Gt=w(()=>{"use strict";t();ro();to()});var qa,za,Ka,vg,Xa,Xp,Yp,kg,Zp,Ya,Ig,kn,In=w(()=>{"use strict";t();qa=require("@reduxjs/toolkit");Gt();O();K();za={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,operaLogin:null,isLoginSessionInvalid:!1},Ka=(0,qa.createSlice)({name:"loginInfoSlice",initialState:za,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload}},extraReducers:e=>{e.addCase(V,()=>za),e.addCase(he,(o,n)=>{o.loginMethod=n.payload.loginMethod,Fo(Oo())})}}),{setLoginMethod:vg,setWalletConnectLogin:Xa,setLedgerLogin:Xp,setTokenLogin:Yp,setTokenLoginSignature:kg,setWalletLogin:Zp,invalidateLoginSession:Ya,setLogoutRoute:Ig}=Ka.actions,kn=Ka.reducer});var Ja,Za,Qa,Cg,Jp,Ng,Qp,An,Ln=w(()=>{"use strict";t();Ja=require("@reduxjs/toolkit");K();Za={},Qa=(0,Ja.createSlice)({name:"modalsSlice",initialState:Za,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(V,()=>Za)}}),{setTxSubmittedModal:Cg,setNotificationModal:Jp,clearTxSubmittedModal:Ng,clearNotificationModal:Qp}=Qa.actions,An=Qa.reducer});var em,En,Bo=w(()=>{"use strict";t();em=!1,En=(e,o=0)=>{em||setTimeout(()=>{!window||(window.location.href=e,window.location.assign(e))},o)}});var ei=w(()=>{"use strict";t();Bo()});function Ht(e){return e[Math.floor(Math.random()*e.length)]}var oi=w(()=>{"use strict";t()});function $t(e){let o=null;if((e==null?void 0:e.statusCode)in Et){let n=e==null?void 0:e.statusCode,{message:r}=Et[n];o=tm.includes(n)?ni:r}return{errorMessage:o,defaultErrorMessage:ni}}var ni,om,nm,tm,Vt=w(()=>{"use strict";t();Ct();ni="Check if the MultiversX app is open on Ledger",om=28161,nm=28160,tm=[om,nm]});var Ge=w(()=>{"use strict";t();ei();oi();Vt()});var ti,ri,jt,rm,ai,jg,zg,Cn,Nn=w(()=>{"use strict";t();ti=require("@reduxjs/toolkit"),ri=y(require("lodash.omit"));Ge();jt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},rm={network:jt,chainID:"-1"},ai=(0,ti.createSlice)({name:"appConfig",initialState:rm,reducers:{initializeNetworkConfig:(e,o)=>{let n=Ht(o.payload.walletConnectBridgeAddresses),r=Ht(o.payload.walletConnectV2RelayAddresses),s=(0,ri.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=k(x(x({},e.network),s),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:jg,setChainID:zg}=ai.actions,Cn=ai.reducer});var ii,zt,ci,Jg,Qg,eu,Dn,Pn=w(()=>{"use strict";t();ii=require("@reduxjs/toolkit");$();K();zt={isSigning:!1,signedSessions:{}},ci=(0,ii.createSlice)({name:"signedMessageInfoSliceState",initialState:zt,reducers:{setSignSession:(e,o)=>{let{sessionId:n,signedSession:r,errorMessage:s}=o.payload;s&&(e.errorMessage=s),e.isSigning=r.status==="pending",e.signedSessions[n]=r},setSignSessionState:(e,o)=>x(x({},e),o.payload),clearSignedMessageInfo:()=>zt},extraReducers:e=>{e.addCase(V,()=>zt)}}),{setSignSession:Jg,clearSignedMessageInfo:Qg,setSignSessionState:eu}=ci.actions,Dn=ci.reducer});var pi,mi,si,di,am,Mn,su,pu,mu,du,Rn,On=w(()=>{"use strict";t();pi=require("@reduxjs/toolkit"),mi=require("redux-persist");Nt();$();bn();K();si={customToasts:[],transactionToasts:[],failTransactionToast:null},di=(0,pi.createSlice)({name:"toastsSlice",initialState:si,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(k(x({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Ro(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=k(x({},o.payload),{toastId:Do})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(V,()=>si),e.addCase(mi.REHYDRATE,(o,n)=>{var s,p;let r=(p=(s=n.customToasts)==null?void 0:s.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=r})}}),{addCustomToast:am,removeCustomToast:Mn,addTransactionToast:su,removeTransactionToast:pu,addFailTransactionToast:mu,removeFailTransactionToast:du}=di.actions,Rn=di.reducer});var li,Kt,Xt,Yt,im,qt,_i,uu,cm,Uo,Fn,Bn=w(()=>{"use strict";t();li=require("@reduxjs/toolkit");K();Kt="Transaction failed",Xt="Transaction successful",Yt="Processing transaction",im="Transaction submitted",qt={},_i=(0,li.createSlice)({name:"transactionsInfo",initialState:qt,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||Kt,successMessage:(r==null?void 0:r.successMessage)||Xt,processingMessage:(r==null?void 0:r.processingMessage)||Yt,submittedMessage:(r==null?void 0:r.submittedMessage)||im,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>qt},extraReducers:e=>{e.addCase(V,()=>qt)}}),{clearTransactionsInfo:uu,setTransactionsDisplayInfo:cm,clearTransactionsInfoForSessionId:Uo}=_i.actions,Fn=_i.reducer});function io(e){return e!=null&&(um(e)||xm(e))}function co(e){return e!=null&&(fm(e)||ym(e))}function so(e){return e!=null&&(Tm(e)||bm(e))}function Zt(e){return e!=null&&hm(e)}function um(e){return e!=null&&sm.includes(e)}function fm(e){return e!=null&&pm.includes(e)}function Tm(e){return e!=null&&mm.includes(e)}function hm(e){return e!=null&&dm.includes(e)}function xm(e){return e!=null&&lm.includes(e)}function ym(e){return e!=null&&_m.includes(e)}function bm(e){return e!=null&&gm.includes(e)}var sm,pm,mm,dm,lm,_m,gm,po=w(()=>{"use strict";t();O();sm=["sent"],pm=["success"],mm=["fail","cancelled","timedOut"],dm=["timedOut"],lm=["pending"],_m=["success"],gm=["fail","invalid"]});var gi,ui,Wo,Sm,fi,Ti,hi,wm,mo,vm,km,vu,xi,lo,$e,Un,Wn=w(()=>{"use strict";t();gi=require("@reduxjs/toolkit"),ui=require("redux-persist");O();po();K();Wo={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Sm={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},fi=(0,gi.createSlice)({name:"transactionsSlice",initialState:Wo,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,u;let{sessionId:n,transactions:r,errorMessage:s,status:p,redirectRoute:m}=o.payload,d=((l=e.customTransactionInformationForSessionId)==null?void 0:l[n])||Sm;e.signedTransactions[n]={transactions:r,status:p,errorMessage:s,redirectRoute:m,customTransactionInformation:d},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=Wo.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=x(x({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:s,transactions:p}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,s!=null&&(e.signedTransactions[n].errorMessage=s),p!=null&&(e.signedTransactions[n].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var d,l,u,f,T,b;let{sessionId:n,status:r,errorMessage:s,transactionHash:p}=o.payload,m=(l=(d=e.signedTransactions)==null?void 0:d[n])==null?void 0:l.transactions;if(m!=null){e.signedTransactions[n].transactions=m.map(D=>D.hash===p?k(x({},D),{status:r,errorMessage:s}):D);let h=(f=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:f.every(D=>co(D.status)),A=(b=(T=e.signedTransactions[n])==null?void 0:T.transactions)==null?void 0:b.every(D=>so(D.status));h&&(e.signedTransactions[n].status="success"),A&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=Wo.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=Wo.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(V,()=>Wo),e.addCase(ui.REHYDRATE,(o,n)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:s}=n.payload.transactions,p=Object.entries(r).reduce((d,[l,u])=>{let f=new Date(l),T=new Date;return T.setHours(T.getHours()+5),T-f>0||(d[l]=u),d},{});s!=null&&(o.customTransactionInformationForSessionId=s),r!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:Ti,updateSignedTransactions:hi,setTransactionsToSign:wm,clearAllTransactionsToSign:mo,clearAllSignedTransactions:vm,clearSignedTransaction:km,clearTransactionToSign:vu,setSignTransactionsError:xi,setSignTransactionsCancelMessage:lo,moveTransactionsToSignedState:$e}=fi.actions,Un=fi.reducer});var M=w(()=>{"use strict";t();vn();In();Ln();Nn();Pn();On();Bn();Wn()});var Jt=w(()=>{"use strict";t()});var vi,wi,ki,Qt,Ku,Xu,Yu,Zu,Ii,Ai,er=w(()=>{"use strict";t();vi=require("@reduxjs/toolkit");K();wi={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},ki=(0,vi.createSlice)({name:"dappModalsSlice",initialState:wi,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=x({},o.payload)}},extraReducers:e=>{e.addCase(V,()=>wi)}}),{setDappModalVisibility:Qt,setDappModalHeaderVisibility:Ku,setDappModalFooterVisibility:Xu,setDappModalHeaderText:Yu,setDappModalFooterText:Zu,setDappModalConfig:Ii}=ki.actions,Ai=ki.reducer});var Li,Em,Cm,Ve,Vn=w(()=>{"use strict";t();Li=require("@reduxjs/toolkit");Jt();vn();er();In();Ln();Nn();Pn();On();Bn();Wn();Em={["account"]:wn,["dappModal"]:Ai,["loginInfo"]:kn,["modals"]:An,["networkConfig"]:Cn,["signedMessageInfo"]:Dn,["toasts"]:Rn,["transactionsInfo"]:Fn,["transactions"]:Un},Cm=(e={})=>(0,Li.combineReducers)(x(x({},Em),e)),Ve=Cm});var Di={};Me(Di,{default:()=>jm,sessionStorageReducers:()=>or});function Ne(e,o=[]){return{key:e,version:1,storage:Ci.default,blacklist:o}}var oe,Ei,Ci,Nm,Ho,Dm,Pm,Mm,Rm,Om,Fm,Bm,Um,Wm,Ni,Gm,or,Hm,$m,Vm,jm,Pi=w(()=>{"use strict";t();oe=require("redux-persist"),Ei=y(require("redux-persist/lib/storage")),Ci=y(require("redux-persist/lib/storage/session"));Vn();M();vn();In();Ln();Nn();Pn();On();Bn();Wn();Jt();Nm={persistReducersStorageType:"localStorage"},Ho={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Dm=Ne(Ho["account"]),Pm=Ne(Ho["loginInfo"]),Mm=Ne(Ho["modals"]),Rm=Ne(Ho["networkConfig"]),Om={2:e=>k(x({},e),{networkConfig:jt})};Fm=Ne("sdk-dapp-transactionsInfo"),Bm=Ne("sdk-dapp-transactions",["transactionsToSign"]),Um=Ne("sdk-dapp-toasts"),Wm=Ne("sdk-dapp-signedMessageInfo"),Ni={key:"sdk-dapp-store",version:2,storage:Ei.default,whitelist:Object.keys(Ho),migrate:(0,oe.createMigrate)(Om,{debug:!1})},Gm=k(x({},Ni),{whitelist:[]}),or={["toasts"]:(0,oe.persistReducer)(Um,Rn),["transactions"]:(0,oe.persistReducer)(Bm,Un),["transactionsInfo"]:(0,oe.persistReducer)(Fm,Fn),["signedMessageInfo"]:(0,oe.persistReducer)(Wm,Dn)},Hm=k(x({},or),{["account"]:(0,oe.persistReducer)(Dm,wn),["loginInfo"]:(0,oe.persistReducer)(Pm,kn),["modals"]:(0,oe.persistReducer)(Mm,An),["networkConfig"]:(0,oe.persistReducer)(Rm,Cn)}),$m=Nm.persistReducersStorageType==="localStorage",Vm=$m?(0,oe.persistReducer)(Ni,Ve(or)):(0,oe.persistReducer)(Gm,Ve(Hm)),jm=Vm});var Mi={};Me(Mi,{default:()=>zm});var zm,Ri=w(()=>{"use strict";t();Vn();zm=Ve()});var Oi={};Me(Oi,{default:()=>Km});var ie,qm,Km,Fi=w(()=>{"use strict";t();ie=require("redux-persist"),qm=[ie.FLUSH,ie.REHYDRATE,ie.PAUSE,ie.PERSIST,ie.PURGE,ie.REGISTER],Km=qm});var Wi={};Me(Wi,{default:()=>Ui});function Ui(e){return(0,Bi.persistStore)(e)}var Bi,Gi=w(()=>{"use strict";t();Bi=require("redux-persist")});var Zc=Q((IL,Yc)=>{t();var rl=typeof a=="object"&&a&&a.Object===Object&&a;Yc.exports=rl});var Qc=Q((AL,Jc)=>{t();var al=Zc(),il=typeof self=="object"&&self&&self.Object===Object&&self,cl=al||il||Function("return this")();Jc.exports=cl});var Nr=Q((LL,es)=>{t();var sl=Qc(),pl=sl.Symbol;es.exports=pl});var rs=Q((EL,ts)=>{t();var os=Nr(),ns=Object.prototype,ml=ns.hasOwnProperty,dl=ns.toString,Qo=os?os.toStringTag:void 0;function ll(e){var o=ml.call(e,Qo),n=e[Qo];try{e[Qo]=void 0;var r=!0}catch(p){}var s=dl.call(e);return r&&(o?e[Qo]=n:delete e[Qo]),s}ts.exports=ll});var is=Q((CL,as)=>{t();var _l=Object.prototype,gl=_l.toString;function ul(e){return gl.call(e)}as.exports=ul});var ms=Q((NL,ps)=>{t();var cs=Nr(),fl=rs(),Tl=is(),hl="[object Null]",xl="[object Undefined]",ss=cs?cs.toStringTag:void 0;function yl(e){return e==null?e===void 0?xl:hl:ss&&ss in Object(e)?fl(e):Tl(e)}ps.exports=yl});var ls=Q((DL,ds)=>{t();var bl=Array.isArray;ds.exports=bl});var gs=Q((PL,_s)=>{t();function Sl(e){return e!=null&&typeof e=="object"}_s.exports=Sl});var fs=Q((ML,us)=>{t();var wl=ms(),vl=ls(),kl=gs(),Il="[object String]";function Al(e){return typeof e=="string"||!vl(e)&&kl(e)&&wl(e)==Il}us.exports=Al});var t_={};Me(t_,{SignWithDeviceModal:()=>n_});module.exports=No(t_);t();var lt=y(require("react")),zr=y(require("classnames"));t();var Gp=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Gp));var _={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};t();t();t();var qi=require("react"),_o=require("react-redux");t();var nr=require("@reduxjs/toolkit"),ji=require("react-redux/lib/utils/Subscription");M();t();var bi=y(require("lodash.throttle"));P();t();var yi=y(require("lodash.isequal")),Gn=require("reselect");$();var S=(0,Gn.createSelectorCreator)(Gn.defaultMemoize,yi.default),Go=(e,o)=>e!=""&&Boolean(o);M();Gt();ro();to();var Im=[yn],Hn=!1,Am=(0,bi.default)(()=>{Fo(Oo())},5e3),Si=e=>o=>n=>{var l;if(Im.includes(n.type))return o(n);let r=e.getState(),s=We.local.getItem(Ce.loginExpiresAt);if(!Go((l=r==null?void 0:r.loginInfo)==null?void 0:l.loginMethod,r==null?void 0:r.account.address))return o(n);if(s==null)return Fo(Oo());let m=Date.now();return s-m<0&&!Hn?setTimeout(()=>{Hn=!0,e.dispatch(Ya())},1e3):(Hn&&(Hn=!1),Am()),o(n)};t();t();function $n(){return typeof sessionStorage!="undefined"}var Hi=$n()?(Pi(),No(Di)).default:(Ri(),No(Mi)).default,$i=$n()?(Fi(),No(Oi)).default:[],Vi=$n()?(Gi(),No(Wi)).default:e=>e;Vn();var L=(0,nr.configureStore)({reducer:Hi,middleware:e=>e({serializableCheck:{ignoredActions:[...$i,Wt.type,Ut.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Si)}),zi=(0,ji.createSubscription)(L),Ef=Vi(L),Cf=(0,nr.configureStore)({reducer:Ve});var Xm={store:L,subscription:zi},tr=(0,qi.createContext)(Xm),Mf=(0,_o.createStoreHook)(tr),G=(0,_o.createDispatchHook)(tr),I=(0,_o.createSelectorHook)(tr);t();t();M();var ve=e=>e.account,je=S(ve,e=>e.address),go=S(ve,je,(e,o)=>o in e.accounts?e.accounts[o]:Sn),Ki=S(ve,go,(e,o)=>{let s=e,{accounts:n}=s,r=_n(s,["accounts"]);return k(x({},r),{account:o})}),Ym=S(go,e=>e.balance),Xi=S(go,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Bf=S(ve,e=>e.shard),Zm=S(ve,e=>e.ledgerAccount),Uf=S(ve,e=>e.walletConnectAccount),Wf=S(ve,e=>e.isAccountLoading),Gf=S(ve,e=>e.accountLoadingError),Yi=S(ve,e=>e.websocketEvent);t();var me=e=>e.loginInfo,jf=S(me,e=>e.loginMethod),rr=S(me,je,(e,o)=>Go(e.loginMethod,o)),zf=S(me,e=>e.walletConnectLogin),qf=S(me,e=>e.ledgerLogin),Jm=S(me,e=>e.walletLogin),Kf=S(me,e=>e.isLoginSessionInvalid),Qm=S(me,e=>e.tokenLogin),Xf=S(me,e=>e.logoutRoute);t();var Zi=e=>e.modals,Jf=S(Zi,e=>e.txSubmittedModal),ed=S(Zi,e=>e.notificationModal);t();var de=e=>e.networkConfig,jn=S(de,e=>e.chainID),od=S(de,e=>e.network.walletConnectBridgeAddress),nd=S(de,e=>e.network.walletConnectV2RelayAddress),td=S(de,e=>e.network.walletConnectV2ProjectId),rd=S(de,e=>e.network.walletConnectV2Options),Ji=S(de,e=>e.network.walletConnectDeepLink),ke=S(de,e=>e.network),ar=S(ke,e=>e.apiAddress),Qi=S(ke,e=>e.explorerAddress),zn=S(ke,e=>e.egldLabel);t();var qn=e=>e.signedMessageInfo,tT=S(qn,e=>e.isSigning),rT=S(qn,e=>e.errorMessage),aT=S(qn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return e.signedSessions[o[n-1]]}),iT=S(qn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return o.length>0?o[n-1]:""});t();var ir=e=>e.toasts,pT=S(ir,e=>e.customToasts),mT=S(ir,e=>e.transactionToasts),dT=S(ir,e=>e.failTransactionToast);t();M();var ec={errorMessage:Kt,successMessage:Xt,processingMessage:Yt},ad=e=>e.transactionsInfo,id=S(ad,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||ec);t();t();var ne=require("@multiversx/sdk-core");P();t();function cr(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function $o(e){var s,p,m;let{data:o}=e,n=cr(o!=null?o:"")?ne.TransactionPayload.fromEncoded(o):new ne.TransactionPayload(o),r=new ne.Transaction(x({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new ne.Address(e.receiver),sender:new ne.Address(e.sender),gasLimit:(s=e.gasLimit.valueOf())!=null?s:Po,gasPrice:(p=e.gasPrice.valueOf())!=null?p:hn,chainID:e.chainID.valueOf(),version:new ne.TransactionVersion((m=e.version)!=null?m:Da)},e.options?{options:new ne.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new ne.Address(e.sender)),r}t();t();t();t();t();var sd=require("@multiversx/sdk-core/out");t();P();t();var nc=require("@multiversx/sdk-core");$();t();var oc=require("@multiversx/sdk-core");function pd(e){try{let o=new oc.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Vo(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&pd(e)}t();t();t();var mr=require("@multiversx/sdk-core"),dr=y(require("bignumber.js"));O();t();t();function sr(e){for(let o=0;o<e.length;o++)if(e.charCodeAt(o)>127)return!1;return!0}function ze(e){let o=e;try{let n=i.Buffer.from(e,"hex").toString();sr(n)&&n.length>1&&(o=n)}catch(n){}return o}var pr={tokenId:"",amount:""},md=e=>e.split("@").map((p,m)=>[2,3].includes(m)?p:ze(p));function lr(e){if(!e)return pr;let o=e.startsWith("ESDTTransfer"),n=e.startsWith("ESDTNFTTransfer")&&e.includes("@");if(o){let[,r,s]=e.split("@");try{let p=i.Buffer.from(r,"hex").toString("ascii");if(!p)return pr;let m=new dr.default("0x"+s.replace("0x","")).toString(10);return{tokenId:p,amount:m}}catch(p){console.error("Error getting token from transaction data",p)}}if(n)try{let[,r,s,p,m]=md(e);if([r,s,p,m].every(d=>Boolean(d))&&Vo(new mr.Address(m).bech32()))return{tokenId:`${r}-${s}`,amount:new dr.default(p,16).toString(10),collection:r,nonce:s,receiver:new mr.Address(m).bech32()}}catch(r){}return pr}t();t();t();var jo="accounts";var _r="blocks";var gr="collections";var ur="tokens";t();t();t();t();J();t();t();J();t();t();t();var _d=y(require("bignumber.js"));P();t();var cc=require("@multiversx/sdk-core"),De=y(require("bignumber.js"));P();t();var ic=y(require("bignumber.js")),fr=(e,o=!0)=>{let n=String(e);if(!n.match(/^[-]?\d+$/))return!1;let r=new ic.default(n),s=o?0:-1;return r.toString(10)===n&&r.comparedTo(0)>=s};t();function Ie(e){return{if:function(o){return o?{then:n=>n instanceof Function?Ie(n(e)):Ie(n)}:{then:()=>Ie(e)}},then:o=>o instanceof Function?Ie(o(e)):Ie(o),valueOf:function(){return e}}}De.default.config({ROUNDING_MODE:De.default.ROUND_FLOOR});function qe({input:e,decimals:o=Te,digits:n=xn,showLastNonZeroDecimal:r=!0,showIsLessThanDecimalsLabel:s=!1,addCommas:p=!1}){if(!fr(e,!1))throw new Error("Invalid input");let m=new De.default(e).isNegative(),d=e;return m&&(d=e.substring(1)),Ie(d).then(()=>cc.TokenPayment.fungibleFromBigInteger("",d,o).toRationalNumber()).then(l=>{let u=new De.default(l);if(u.isZero())return Ue;let f=u.toString(10),[T,b]=f.split("."),h=new De.default(b||0),A=Ie(0).if(Boolean(b&&r)).then(()=>Math.max(b.length,n)).if(h.isZero()&&!r).then(0).if(Boolean(b&&!r)).then(()=>Math.min(b.length,n)).valueOf(),D=b&&n>=1&&n<=b.length&&h.isGreaterThan(0)&&new De.default(b.substring(0,n)).isZero(),B=u.toFormat(A);return Ie(f).if(p).then(B).if(Boolean(D)).then(E=>{let R=new De.default(T).isZero(),[z,re]=E.split("."),ye=new Array(n-1).fill(0),se=[...ye,0].join(""),Ee=[...ye,1].join("");return R?s?`<${z}.${Ee}`:r?`${z}.${re}`:z:`${z}.${se}`}).if(Boolean(!D&&b)).then(E=>{let[R]=E.split("."),z=b.substring(0,A);if(r){let re=n-z.length;if(re>0){let ye=Array(re).fill(0).join("");return z=`${z}${ye}`,`${R}.${z}`}return E}return z?`${R}.${z}`:R}).valueOf()}).if(m).then(l=>`-${l}`).valueOf()}t();t();t();P();t();t();t();t();P();t();P();t();var gd=require("@multiversx/sdk-core");P();t();var zo=require("@multiversx/sdk-core"),Td=require("@multiversx/sdk-network-providers");P();t();t();var ud=y(require("bignumber.js"));P();t();P();t();var Tr={isEsdt:!1,isNft:!1,isEgld:!1};function Ke(e){let o=e==null?void 0:e.split("-").length;return o===2?k(x({},Tr),{isEsdt:!0}):o===3?k(x({},Tr),{isNft:!0}):k(x({},Tr),{isEgld:!0})}t();t();t();t();P();t();J();var xd=["reDelegateRewards","claimRewards","unBond"],yd=["wrapEgld","unwrapEgld"],bd=["unStake"],Sd=["unDelegate"];t();t();t();P();t();var kd=y(require("bignumber.js"));t();t();t();t();J();t();var Id=y(require("bignumber.js"));t();t();function Xn({tokenId:e,erdLabel:o}){return Boolean(e&&e!==o)}t();var xr=y(require("bignumber.js"));$();t();var mc=(e,o)=>{let n=[],r=e.indexOf(o,0);for(;r>=0;)n.push(r),r=e.indexOf(o,r+1);return n};function dc(e){let o=[],n=0;try{if((e==null?void 0:e.startsWith("MultiESDTNFTTransfer"))&&(e==null?void 0:e.includes("@"))){let[,r,s,...p]=e==null?void 0:e.split("@");if(r){let m=new xr.default(s,16).toNumber(),d=0;for(let T=0;T<m;T++){let b={type:"nftTransaction",data:"",receiver:r};for(let h=0;h<3;h++){switch(h){case 0:b.token=ze(p[d]),b.data=p[d];break;case 1:{let A=p[d]&&p[d].length?p[d]:"";A&&A!=="00"?b.nonce=A:b.type="esdtTransaction",b.data=`${b.data}@${p[d]}`;break}case 2:b.amount=new xr.default(p[d],16).toString(10),b.data=`${b.data}@${p[d]}`;break;default:break}n=d+1,d++}o[T]=b}let l=o.length!==m,u=o.some(T=>mc(T.data,"@").length!==2),f=o.some(T=>T.data.startsWith("@"));if(l||u||f)return[];if(p[n]){let T=p[n];for(let b=n+1;b<p.length;b++)T+="@"+p[b];o[m]={type:"scCall",data:T,receiver:r}}}}}catch(r){return console.error("failed parsing tx",r),o}return o}t();t();t();O();function qo(e){let n=Object.getPrototypeOf(e).toPlainObject!=null?e:$o(e);return k(x({},n.toPlainObject()),{hash:n.getHash().hex(),status:"pending"})}t();t();t();t();J();t();t();t();var Ld=y(require("linkifyjs"));t();P();t();var Ed=y(require("bignumber.js"));t();$();t();t();O();t();O();t();t();t();J();t();J();t();var Cd=y(require("bignumber.js"));P();J();t();J();t();var _c=require("react");$();t();t();J();t();t();var Nd=require("@multiversx/sdk-core/out"),Dd=y(require("bignumber.js"));J();t();$();t();t();$();var $0=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var gc=require("react");$();t();var Md=require("react");J();var Y0=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];po();var Zn=e=>e.transactions,uo=S(Zn,e=>e.signedTransactions),Rd=S(Zn,e=>e.signTransactionsError),yr=S(Zn,e=>e.signTransactionsCancelMessage),Jn=e=>o=>Object.entries(o).reduce((n,[r,s])=>(e(s.status)&&(n[r]=s),n),{}),uc=S(uo,Jn(io)),fc=S(uo,Jn(co)),Tc=S(uo,Jn(so)),Od=S(uo,Jn(Zt)),Qn=S(Zn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:k(x({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>$o(n)))||[]})}),Fd=S(uo,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});var Ko=()=>I(de);t();var hc=require("react");t();M();t();t();t();t();t();t();t();t();var xc=require("@multiversx/sdk-extension-provider"),yc=require("@multiversx/sdk-hw-provider"),bc=require("@multiversx/sdk-opera-provider"),Sc=require("@multiversx/sdk-web-wallet-provider");P();O();t();var et=fo,br=et;var To=e=>{switch(e==null?void 0:e.constructor){case Sc.WalletProvider:return"wallet";case et:return"walletconnect";case br:return"walletconnectv2";case yc.HWProvider:return"ledger";case xc.ExtensionProvider:return"extension";case bc.OperaProvider:return"opera";case fo:return"";default:return"extra"}};var Ae=e=>`Unable to perform ${e}, Provider not initialized`,fo=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Ae(`login with options: ${o}`))}logout(o){throw new Error(Ae(`logout with options: ${o}`))}getAddress(){throw new Error(Ae("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Ae(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Ae(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Ae(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Ae(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Ae(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Ae(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Sr=new fo;var Bd=Sr;function xe(){return Bd||Sr}M();t();t();var vc=y(require("axios"));t();t();var wc=e=>{let o=e!=null?e:ar(L.getState());return o.endsWith("/")?o.slice(0,-1):o};function kc(e){return v(this,null,function*(){if(!e)return null;let n=`${wc()}/${jo}/${e}`;try{let{data:r}=yield vc.default.get(n);return r}catch(r){console.error("error fetching configuration for ",n)}return null})}function Xo(e){return v(this,null,function*(){return yield kc(e)})}t();O();t();t();function Ic(){let e=xe();return To(e)}function ho(e){return Ic()===e}t();function ot(e){let o=L.getState(),n=Xi(o);return e?n&&!isNaN(n)?Math.max(n,e.nonce):e.nonce:n}po();t();t();t();var Ud=y(require("axios"));t();var Wd=y(require("axios"));t();var Hd=y(require("axios"));t();t();var $d=y(require("axios"));t();var Vd=y(require("axios"));t();M();O();t();t();M();t();t();var Ec=require("@multiversx/sdk-core");t();K();$();t();po();t();M();O();t();M();$();t();t();t();$();t();t();t();t();var Cc=y(require("axios")),Nc=y(require("swr"));var qd=e=>Cc.default.get(e).then(o=>o.data);function Dc({tokenId:e}){var b;let{network:o}=Ko(),{isEsdt:n,isNft:r}=Ke(e),s=n?ur:gr,p=e;if(r){let[h,A]=e.split("-");p=`${h}-${A}`}let{data:m,error:d}=(0,Nc.default)(Boolean(p)?`${o.apiAddress}/${s}/${p}`:null,qd);if(!p)return{tokenDecimals:Number(o.decimals),tokenLabel:"",tokenAvatar:""};let l=m?m==null?void 0:m.decimals:Number(o.decimals),u=m?m==null?void 0:m.name:"",f=m?`${(b=m==null?void 0:m.assets)==null?void 0:b.svgUrl}`:"",T=m==null?void 0:m.assets;return{tokenDecimals:l,tokenLabel:u,tokenAvatar:f,assets:T,error:d}}t();t();var Yo=require("react");var Pc={tokenId:"",amount:"",type:"",multiTxData:"",receiver:""};function Mc({transactions:e}){let[o,n]=(0,Yo.useState)({}),[r,s]=(0,Yo.useState)([]);function p(l,u){n(f=>k(x({},f),{[l]:u}))}function m(l,u){return o==null?Pc:l in o?o[l]:u!=null&&String(u)in o?o[u]:Pc}function d(){if(e&&e.length>0){let l=[];e.forEach((u,f)=>{let T=u.getData().toString(),b=dc(T);if(b.length>0)b.forEach((h,A)=>{let D={transaction:u,multiTxData:h.data,transactionIndex:A};p(h.data,{tokenId:h.token?h.token:"",amount:h.amount?h.amount:"",type:h.type,nonce:h.nonce?h.nonce:"",multiTxData:h.data,receiver:h.receiver}),l.push(D)});else{let{tokenId:h,amount:A}=lr(u.getData().toString());h&&p(u.getData().toString(),{tokenId:h,amount:A,receiver:u.getReceiver().bech32()}),l.push({transaction:u,transactionIndex:f})}}),s(l)}}return(0,Yo.useEffect)(()=>{d()},[e==null?void 0:e.length]),{parsedTransactionsByDataField:o,getTxInfoByDataField:m,allTransactions:r}}t();var Fc=require("react"),tt=require("@multiversx/sdk-web-wallet-provider"),Bc=y(require("qs"));P();M();O();t();var nt=e=>{let o=new URLSearchParams(e).toString(),n=window==null?void 0:window.location.pathname,r=o?`?${o}`:"",s=n?`${n}${r}`:"./";setTimeout(()=>{window==null||window.history.replaceState({},document==null?void 0:document.title,s)})};t();var Rc,Kd={search:(Rc=window==null?void 0:window.location)==null?void 0:Rc.search,removeParams:[]},Oc=(e,o=Kd)=>{var p,m;let n={},r=(m=o.search)!=null?m:(p=window==null?void 0:window.location)==null?void 0:p.search;if(r){let d=r?new URLSearchParams(r):[];n=Object.fromEntries(d)}let s={};return e.forEach(d=>{s[d]=n[d],delete n[d]}),o.removeParams!=null&&Object.keys(n).forEach(d=>{var u,f;let[l]=d.split("[");(((u=o.removeParams)==null?void 0:u.includes(d))||((f=o.removeParams)==null?void 0:f.includes(l)))&&delete n[d]}),{remainingParams:s,params:n,clearNavigationHistory:()=>nt(n)}};function wr(e){let o=window==null?void 0:window.location.search,n=I(ke),r=G();(0,Fc.useEffect)(()=>{if(o!=null){let s=Bc.default.parse(o.replace("?",""));if(s&&Mo in s){let p=String(s[Mo]),m=new tt.WalletProvider(`${n.walletAddress}${Pa}`).getTransactionsFromWalletUrl();if(s.status==="cancelled"){r($e({sessionId:p,status:"cancelled"})),e(),history.pushState({},document==null?void 0:document.title,"?");return}if(m.length>0){r($e({sessionId:p,status:"signed",transactions:m.map(u=>qo(u))}));let[d]=m,{params:l}=Oc([],{removeParams:[...Object.keys(d),tt.WALLET_PROVIDER_CALLBACK_PARAM,Mo]});nt(l)}}}},[o])}t();var Xe=require("react");Vt();var rt={};function Uc({isLedger:e=!1,transactionsToSign:o,egldLabel:n,address:r,onCancel:s,onSignTransaction:p,onTransactionsSignError:m,onTransactionsSignSuccess:d,onGetScamAddressData:l}){var Ao,sn,pn;let[u,f]=(0,Xe.useState)(0),[T,b]=(0,Xe.useState)(),[h,A]=(0,Xe.useState)(null),[D,B]=(0,Xe.useState)(!1),{getTxInfoByDataField:X,allTransactions:E}=Mc({transactions:o}),R=u===E.length-1;(0,Xe.useEffect)(()=>{z()},[u,E]);function z(){return v(this,null,function*(){var Eo;let N=E[u];if(N==null)return;let{transaction:W,multiTxData:pe}=N,Je=W.getData().toString(),mn=X(W.getData().toString(),pe),{tokenId:dn}=mn,Pe=W.getReceiver().toString(),Lo=r!==Pe,gt=Pe in rt;if(Lo&&!gt&&l!=null){let Qe=yield l(Pe);rt=x(x({},rt),Qe!=null&&Qe.scamInfo?{[Pe]:Qe.scamInfo}:{})}let q=Boolean(dn&&Xn({tokenId:dn,erdLabel:n}));A({transaction:W,receiverScamInfo:((Eo=rt[Pe])==null?void 0:Eo.info)||null,transactionTokenInfo:mn,isTokenTransaction:q,dataField:Je})})}function re(){f(0),b(void 0),B(!1)}function ye(){return v(this,null,function*(){try{if(h==null)return;B(e);let N=yield p(h.transaction),W={[u]:N},pe=T?x(x({},T),W):W;b(pe),R?pe&&(d(Object.values(pe)),re()):(f(Je=>Je+1),B(!1))}catch(N){console.error(N,"sign error");let{message:W}=N,pe=e?$t(N).errorMessage:null;re(),m(pe!=null?pe:W)}})}function se(){try{if(h==null)return;h.transaction.getSignature().hex()?R||f(W=>W+1):ye()}catch(N){ye()}}let Ee=u===0;function rn(){Ee?s==null||s():f(N=>N-1)}let an=Boolean(((Ao=h==null?void 0:h.transactionTokenInfo)==null?void 0:Ao.type)&&((sn=h==null?void 0:h.transactionTokenInfo)==null?void 0:sn.multiTxData)&&!(h!=null&&h.dataField.endsWith((pn=h==null?void 0:h.transactionTokenInfo)==null?void 0:pn.multiTxData)));function Ze(){an?f(N=>N+1):se()}function _t(){f(N=>{let W=N+1;return W>(E==null?void 0:E.length)?N:W})}function cn(){f(N=>{let W=N-1;return W<0?N:W})}return{allTransactions:E,onSignTransaction:Ze,onNext:_t,onPrev:cn,waitingForDevice:D,onAbort:rn,isLastTransaction:R,isFirstTransaction:u===0,hasMultipleTransactions:E.length>1,shouldContinueWithoutSigning:an,currentStep:u,signedTransactions:T,currentTransaction:h}}t();var $c=require("react"),Zd=require("@multiversx/sdk-extension-provider");P();t();var xo=()=>{let e=xe(),o=To(e);return{provider:e,providerType:o}};M();O();t();t();function vr(e){let o=Qn(L.getState()),n=Object.keys(e).length===(o==null?void 0:o.transactions.length);return e&&(e&&n)}t();var Xd=(e,o)=>o.map((n,r)=>(n.setNonce(e+r),n)),kr=()=>{let e=I(je);return o=>v(void 0,null,function*(){let n=yield Xo(e),r=ot(n);return Xd(r,o)})};t();var yo=require("react"),Ar=require("@multiversx/sdk-extension-provider");P();t();t();var Wc=()=>I(Ki);t();var Zo=()=>I(go);t();t();t();t();t();var Jo=y(require("react"));var B1=(0,Jo.createContext)({}),U1=L.getState();t();var Hc=y(require("react"));t();var Ir=y(require("react")),Yd=y(require("axios"));t();t();M();var Lr=()=>{let e=G(),{provider:o}=xo(),{nonce:n}=Zo(),[r,s]=(0,yo.useState)(null),[p,m]=(0,yo.useState)(),[d,l]=(0,yo.useState)([]),u=kr(),f=I(Qn),T=I(yr),b=()=>v(void 0,null,function*(){var E;let B=Boolean(f==null?void 0:f.transactions),X=(E=f==null?void 0:f.transactions)!=null?E:[];if(B){let R=yield u(X);l(R)}m(B)});(0,yo.useEffect)(()=>{b()},[f,n]);let h=()=>{s(null),e(lo(null))},A=B=>{h(),D(B)};wr(A);function D(B){var E,R;let X=o instanceof Ar.ExtensionProvider;e(mo()),e(Uo(B)),e(Mn(Do)),X&&(h(),(R=(E=Ar.ExtensionProvider.getInstance())==null?void 0:E.cancelAction)==null||R.call(E))}return{error:r,canceledTransactionsMessage:T,clearTransactionStatusMessage:h,onAbort:A,setError:s,hasTransactions:p,transactionsToSign:f&&k(x({},f),{transactions:d})}};t();t();var Vc=y(require("axios"));function jc(e){return v(this,null,function*(){let{network:{apiAddress:o,apiTimeout:n}}=de(L.getState()),{data:r}=yield Vc.default.get(`/${jo}/${e}`,{baseURL:o,timeout:Number(n)});return r})}M();$();Bo();t();Be();M();function zc(){let e=G();return o=>{e(mo()),e(Uo(o)),e(lo(At))}}function Er({onCancel:e,verifyReceiverScam:o=!0}){let{transactionsToSign:n,hasTransactions:r}=Lr(),s=I(zn),{account:{address:p}}=Wc(),{provider:m}=xo(),d=G(),l=zc(),{transactions:u,sessionId:f,callbackRoute:T,customTransactionInformation:b}=n||{};function h(R){f&&d($e({sessionId:f,status:"cancelled"})),d(xi(R))}let A=T!=null&&(window==null?void 0:window.location.pathname.includes(T));function D(R){!vr(R)||f&&(d($e({sessionId:f,status:"signed",transactions:R.map(re=>qo(re))})),T!=null&&(b==null?void 0:b.redirectAfterSign)&&!A&&En(T))}function B(){e(),l(f)}function X(R){return v(this,null,function*(){return yield m.signTransaction(R)})}let E=Uc({address:p,egldLabel:s,transactionsToSign:r?u:[],onGetScamAddressData:o?jc:null,isLedger:ho("ledger"),onCancel:B,onSignTransaction:X,onTransactionsSignError:h,onTransactionsSignSuccess:D});return k(x({},E),{callbackRoute:T})}t();t();var Nl=require("react");t();t();t();M();t();t();var el=y(require("bignumber.js"));P();M();$();t();var Qd=y(require("bignumber.js"));P();t();var ol=require("@multiversx/sdk-core"),nl=y(require("bignumber.js"));P();t();t();t();t();var Xc=y(require("axios"));t();t();t();function qc(e){return v(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}t();var Kc=(e,o,n,r=0)=>v(void 0,null,function*(){try{return yield e(...n)}catch(s){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield qc(o.delay)),yield Kc(e,o,n,r+1)):null}}),Cr=(e,o={retries:5,delay:500})=>(...n)=>v(void 0,null,function*(){return yield Kc(e,o,n)});var tl=4;var vL=Cr((e,o)=>v(void 0,null,function*(){let{data:n}=yield Xc.default.get(`${e}/${_r}?from=${tl}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[r]=n;return r}));t();t();bn();t();t();var Ll=y(fs());t();var Ts,zL={origin:typeof window!="undefined"?(Ts=window==null?void 0:window.location)==null?void 0:Ts.origin:"",apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};t();O();po();t();var hs=require("react");t();t();var Pl=require("react"),Ml=require("@multiversx/sdk-extension-provider");Be();K();O();t();Ge();t();var Dl=require("react"),xs=require("@multiversx/sdk-core");M();t();var Rl=require("react"),Ol=require("@multiversx/sdk-opera-provider");Be();K();O();Ge();t();var Ul=require("react");Be();M();t();t();Dt();t();t();var Fl=y(require("platform"));t();K();$();Bo();ro();to();t();t();t();t();t();function ct(){return zn(L.getState())}t();t();t();t();O();t();Bo();ro();t();t();t();M();t();t();var Bl=y(require("axios"));t();t();var ys=require("react"),Gl=require("@multiversx/sdk-hw-provider");Be();K();M();O();Ge();t();var Wl=require("react");t();var Pr=require("react");K();M();O();Ge();t();var Mr=require("react");K();M();O();Ge();t();t();var bs=require("react");t();var zl=require("react");t();var Rr=require("react"),jl=require("socket.io-client");M();t();t();t();t();var Ss=e=>e.dappModal,ws=S(Ss,e=>e.visible),So=S(Ss,e=>e.modal),AD=S(So,e=>e.showHeader),LD=S(So,e=>e.showFooter),ED=S(So,e=>e.headerText),CD=S(So,e=>e.footerText);var vs=()=>I(ws);t();var Ds=y(require("react"));t();t();t();var vo=y(require("react")),Wr=y(require("classnames")),Cs=y(require("react-dom"));t();var ql=`.dapp-core-component__dappModalStyles__dappModal {
  position: fixed;
  z-index: 999;
  padding-top: 100px;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  align-items: center;
}

.dapp-core-component__dappModalStyles__dappModalContent {
  background: #fff;
  border: 2px solid #aaa;
  border-radius: 5px;
  z-index: 999;
  max-width: 500px;
  margin: auto;
  padding: 0;
  position: relative;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.dapp-core-component__dappModalStyles__dappModalHeader {
  padding: 2px 16px;
  background-color: #fff;
  font-size: 1.25rem;
}

.dapp-core-component__dappModalStyles__dappModalHeaderText {
  padding: 0 1rem;
}

.dapp-core-component__dappModalStyles__dappModalBody {
  padding: 16px 0;
}

.dapp-core-component__dappModalStyles__dappModalFooter {
  background-color: #fff;
  padding: 1rem 0.5rem;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton {
  position: absolute;
  top: 0;
  right: 0;
  float: right;
  font-weight: bold;
  border-radius: 5px;
  font-size: 1rem;
  padding: 0.5rem 1rem;
  border: 0;
}

.dapp-core-component__dappModalStyles__dappModalCloseButton:hover,
.dapp-core-component__dappModalStyles__dappModalCloseButton:focus {
  text-decoration: none;
  cursor: pointer;
  color: #212529;
  background-color: #dee2e6;
  border-color: #e9ecef;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ql));var ce={dappModal:"dapp-core-component__dappModalStyles__dappModal",dappModalContent:"dapp-core-component__dappModalStyles__dappModalContent",dappModalHeader:"dapp-core-component__dappModalStyles__dappModalHeader",dappModalHeaderText:"dapp-core-component__dappModalStyles__dappModalHeaderText",dappModalBody:"dapp-core-component__dappModalStyles__dappModalBody",dappModalFooter:"dapp-core-component__dappModalStyles__dappModalFooter",dappModalCloseButton:"dapp-core-component__dappModalStyles__dappModalCloseButton"};t();var ks=y(require("react")),Is=y(require("classnames"));var Or=({className:e,children:o})=>ks.default.createElement("div",{className:(0,Is.default)(ce.dappModalBody,e)},o);t();var Fr=y(require("react")),As=y(require("classnames"));var Br=({visible:e,customFooter:o,className:n,footerText:r})=>e?Fr.default.createElement("div",{className:(0,As.default)(ce.dappModalFooter,n)},o!=null?o:Fr.default.createElement("div",null,r)):null;t();var wo=y(require("react")),Ls=require("@fortawesome/free-solid-svg-icons"),Es=require("@fortawesome/react-fontawesome"),on=y(require("classnames"));var Ur=({visible:e,headerText:o,customHeader:n,className:r,closeButtonClassName:s,headerTextClassName:p,onHide:m})=>e?n?wo.default.createElement("div",{className:(0,on.default)(ce.dappModalHeader,r)},n):wo.default.createElement("div",{className:(0,on.default)(ce.dappModalHeader,r)},wo.default.createElement("div",{className:(0,on.default)(ce.dappModalHeaderText,p)},o),wo.default.createElement("button",{onClick:m,className:(0,on.default)(ce.dappModalCloseButton,_.btn,_.btnLight,s)},wo.default.createElement(Es.FontAwesomeIcon,{size:"lg",icon:Ls.faTimes}))):null;var Kl={showHeader:!0,showFooter:!1,headerText:"",footerText:""},Gr=({id:e="dapp-modal",visible:o,onHide:n,parentElement:r,config:s=Kl,children:p,className:m="dapp-modal-dialog-wrapper"})=>{if(!o)return null;let{showHeader:d,showFooter:l,headerText:u,footerText:f,modalDialogClassName:T="dapp-modal-dialog",modalContentClassName:b="dapp-modal-dialog-content",modalHeaderClassName:h="dapp-modal-dialog-header",modalHeaderTextClassName:A="dapp-modal-dialog-header-text",modalCloseButtonClassName:D="dapp-modal-dialog-close-button",modalBodyClassName:B="dapp-modal-dialog-content-body",modalFooterClassName:X="dapp-modal-dialog-footer",customModalHeader:E,customModalFooter:R}=s;return Cs.default.createPortal(vo.default.createElement("div",{id:e,role:"dialog","aria-modal":"true",className:(0,Wr.default)(T,ce.dappModal,m)},vo.default.createElement("div",{className:(0,Wr.default)(ce.dappModalContent,b)},vo.default.createElement(Ur,{visible:d,headerText:u,customHeader:E,className:h,headerTextClassName:A,closeButtonClassName:D,onHide:n}),vo.default.createElement(Or,{className:B},p),vo.default.createElement(Br,{visible:l,customFooter:R,footerText:f,className:X}))),r!=null?r:document==null?void 0:document.body)};t();t();var ko=require("react");er();var Ns=e=>{var d;let o=G(),n=vs(),r=I(So),s=()=>{o(Qt(!0))},p=()=>{o(Qt(!1))},m=(0,ko.useCallback)(l=>{o(Ii(l))},[]);return(0,ko.useEffect)(()=>{e!=null&&e.modalConfig&&m(e==null?void 0:e.modalConfig)},[e==null?void 0:e.modalConfig]),(0,ko.useEffect)(()=>{var l;return(l=e==null?void 0:e.config)!=null&&l.openOnMount&&s(),()=>{p()}},[]),(0,ko.useEffect)(()=>{var l;typeof((l=e==null?void 0:e.config)==null?void 0:l.visible)=="boolean"&&(e.config.visible?s():p())},[(d=e==null?void 0:e.config)==null?void 0:d.visible]),{handleShowModal:s,handleHideModal:p,setModalConfig:m,visible:n,config:r}};t();var Ps=e=>{let{handleHideModal:o,visible:n,config:r}=Ns({modalConfig:e.modalConfig,config:k(x({},e.modalInteractionConfig),{visible:e.visible})});return Ds.default.createElement(Gr,{onHide:()=>{var p;o(),(p=e.onClose)==null||p.call(e)},visible:n,config:r,className:e.className},e.children)};t();var C=y(require("react")),dt=require("@fortawesome/free-solid-svg-icons"),Js=require("@fortawesome/react-fontawesome"),Qs=require("@multiversx/sdk-core/out"),te=y(require("classnames"));t();t();var Io=y(require("react")),Rs=require("@fortawesome/react-fontawesome"),nn=y(require("classnames"));t();var Xl=`.dapp-core-component__pageState-styles__wrapper {
  color: inherit;
}

.dapp-core-component__pageState-styles__state {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Xl));var Ms={wrapper:"dapp-core-component__pageState-styles__wrapper",state:"dapp-core-component__pageState-styles__state"};var Os=({icon:e,title:o,action:n,iconClass:r,dataTestId:s,description:p,iconBgClass:m,iconSize:d="5x",className:l="dapp-page-state"})=>{let u={wrapper:(0,nn.default)(Ms.state,_.mAuto,_.p4,_.textCenter,l),iconContainer:(0,nn.default)(_.iconState,_.mxAuto,{[m!=null?m:""]:Boolean(m)}),iconClass:(0,nn.default)(r!=null&&r),title:(0,nn.default)(_.h4,_.my4),description:_.mb3};return Io.default.createElement("div",{className:u.wrapper,"data-testid":s},e&&Io.default.createElement("span",{className:u.iconContainer},Io.default.createElement(Rs.FontAwesomeIcon,{icon:e,className:u.iconClass,size:d})),o&&Io.default.createElement("p",{className:u.title},o),p&&Io.default.createElement("div",{className:u.description},p),n)};t();var tn=y(require("react")),st=y(require("classnames"));t();var Fs=y(require("react")),Bs=y(require("classnames"));var Us=({color:e,"data-testid":o,className:n="dapp-dot"})=>Fs.default.createElement("span",{className:(0,Bs.default)(_.dot,e,n),"data-testid":o});var Ws=({totalSteps:e,currentStep:o,className:n="dapp-progress-steps"})=>{let r=[];for(let s=1;s<=e;s+=1){let p=o===s,m=p||s<o?_.bgPrimary:_.bgSecondary;r.push(tn.default.createElement(Us,{"data-testid":`step${s}${p?"active":""}`,key:s,color:m,className:p?"active":""}))}return tn.default.createElement("div",{className:(0,st.default)(_.progressSteps,_.flexRow,_.justifyContentCenter,n)},tn.default.createElement("div",{className:(0,st.default)(_.steps,_.flexRow,_.justifyContentCenter,_.alignItemsCenter,_.positionRelative)},tn.default.createElement("hr",{className:(0,st.default)(_.positionAbsolute,_.w100,_.m0)}),r))};t();var j=y(require("react")),zs=require("@fortawesome/free-solid-svg-icons"),qs=require("@fortawesome/react-fontawesome");t();var pt=y(require("react")),Yl=e=>pt.createElement("svg",x({viewBox:"0 0 300 300",fill:"none",xmlns:"http://www.w3.org/2000/svg",className:"expanded"},e),pt.createElement("rect",{width:300,height:300,rx:150,fill:"black",id:"egld-token"}),pt.createElement("path",{d:"M158.482 149.928L228.714 112.529L216.919 90L152.575 115.854C150.923 116.523 149.077 116.523 147.425 115.854L83.0814 90L71.25 112.602L141.482 150L71.25 187.398L83.0814 210L147.425 183.948C149.077 183.279 150.923 183.279 152.575 183.948L216.919 209.874L228.75 187.272L158.482 149.928Z",fill:"#23F7DD"})),Gs=Yl;t();var Zl=`.dapp-core-component__tokenDetailsStyles__token-symbol-custom-token {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Zl));var Hs={"token-symbol-custom-token":"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token",tokenSymbolCustomToken:"dapp-core-component__tokenDetailsStyles__token-symbol-custom-token"};t();var Hr=y(require("react")),$r=y(require("classnames"));t();var Jl=`.dapp-core-component__tokenSymbol__token-symbol,
.dapp-core-component__tokenSymbol__token-symbol-combined {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
  border-radius: 50%;
}
.dapp-core-component__tokenSymbol__token-symbol svg #dapp-core-component__tokenSymbol__egld-token,
.dapp-core-component__tokenSymbol__token-symbol-combined svg #dapp-core-component__tokenSymbol__egld-token {
  fill: #000;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  background-color: rgba(0, 0, 0, 0);
}

.dapp-core-component__tokenSymbol__token-symbol {
  border: 1px solid #000;
  width: 2rem;
  height: 2rem;
}
.dapp-core-component__tokenSymbol__token-symbol img {
  width: 100%;
  height: 100%;
}
.dapp-core-component__tokenSymbol__token-symbol svg.dapp-core-component__tokenSymbol__expanded {
  width: 100%;
  height: 100%;
}

.dapp-core-component__tokenSymbol__token-symbol-combined {
  width: 4.5rem;
  height: 4.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg circle {
  fill: #fff;
}
.dapp-core-component__tokenSymbol__token-symbol-combined svg {
  width: 4.25rem;
  height: 4.25rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small {
  width: 2.5rem;
  min-width: 2.5rem;
  height: 2.5rem;
}
.dapp-core-component__tokenSymbol__token-symbol-combined.dapp-core-component__tokenSymbol__small svg {
  width: 2.5rem;
  height: 2.5rem;
}

.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol {
  width: 3rem;
  height: 3rem;
}
.dapp-core-component__tokenSymbol__token-icon-lg .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.25rem;
  height: 2.25rem;
}

.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol {
  width: 4rem;
  height: 4rem;
}
.dapp-core-component__tokenSymbol__token-icon-xl .dapp-core-component__tokenSymbol__token-symbol svg {
  width: 2.3rem;
  height: 2.3rem;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Jl));var mt={"token-symbol":"dapp-core-component__tokenSymbol__token-symbol",tokenSymbol:"dapp-core-component__tokenSymbol__token-symbol","token-symbol-combined":"dapp-core-component__tokenSymbol__token-symbol-combined",tokenSymbolCombined:"dapp-core-component__tokenSymbol__token-symbol-combined","egld-token":"dapp-core-component__tokenSymbol__egld-token",egldToken:"dapp-core-component__tokenSymbol__egld-token",expanded:"dapp-core-component__tokenSymbol__expanded",small:"dapp-core-component__tokenSymbol__small","token-icon-lg":"dapp-core-component__tokenSymbol__token-icon-lg",tokenIconLg:"dapp-core-component__tokenSymbol__token-icon-lg","token-icon-xl":"dapp-core-component__tokenSymbol__token-icon-xl",tokenIconXl:"dapp-core-component__tokenSymbol__token-icon-xl"};var $s=({children:e,className:o="dapp-simple-token-symbol"})=>Hr.default.createElement("div",{className:(0,$r.default)(mt.tokenSymbol,o)},e),Vs=({small:e,children:o,className:n="dapp-combined-token-symbol"})=>Hr.default.createElement("div",{className:(0,$r.default)(mt.tokenSymbolCombined,{[mt.small]:e},n)},o);var Ks=e=>{let o=e.split("-");return e.includes("-")?`${o[0]}-${o[1]}`:e};function Ql(e,o){return o&&o!=="undefined"?j.default.createElement("img",{className:Hs.tokenSymbolCustomToken,src:o}):e?j.default.createElement(Gs,null):j.default.createElement(qs.FontAwesomeIcon,{icon:zs.faDiamond})}var Vr=(e,o)=>{let n=ct(),r=e===n;return{token:e,symbol:e?e.split("-")[0]:"",label:e,icon:()=>Ql(r,o)}},js=e=>{let{isNft:o}=Ke(e);return o?e:Ks(e)},Le=class extends j.Component{render(){return null}};Le.Token=o=>j.default.createElement(j.default.Fragment,null,o.token),Le.Symbol=o=>j.default.createElement(j.default.Fragment,null,Vr(Ks(o.token),o.tokenAvatar).symbol),Le.Label=o=>j.default.createElement(j.default.Fragment,null,Vr(js(o.token),o.tokenAvatar).label),Le.Icon=o=>{var r;let n=Vr(js(o.token),o.tokenAvatar).icon;return j.default.createElement("div",{className:(r=o.className)!=null?r:"dapp-token-details-icon"},o.combined?j.default.createElement(Vs,{small:o.small},j.default.createElement(n,null)):j.default.createElement($s,null,j.default.createElement(n,null)))};t();var F=y(require("react")),Ye=y(require("classnames"));P();var e_=(e,o)=>[...e.matchAll(new RegExp(o,"gi"))].map(n=>n.index),Xs=({data:e,highlight:o,isScCall:n,className:r="dapp-transaction-data",innerTransactionDataClasses:s})=>{let{transactionDataInputLabelClassName:p,transactionDataInputValueClassName:m}=s||{},d=F.default.createElement(F.default.Fragment,null,e),[l,...u]=o&&n?o.split("@"):[];if(e&&o&&e_(e,o).length===1)switch(!0){case e.startsWith(o):{let[,f]=e.split(o);d=F.default.createElement(F.default.Fragment,null,o,F.default.createElement("span",{className:_.textMuted},f));break}case e.endsWith(o):{let[f]=e.split(o);d=F.default.createElement(F.default.Fragment,null,F.default.createElement("span",{className:_.textMuted},f),o);break}default:{let[f,T]=e.split(o);d=F.default.createElement(F.default.Fragment,null,F.default.createElement("span",{className:_.textMuted},f),F.default.createElement("span",{className:_.highlighted},o),F.default.createElement("span",{className:_.textMuted},T));break}}return F.default.createElement(F.default.Fragment,null,l&&F.default.createElement("div",{className:(0,Ye.default)(_.formGroup,_.mb0,_.dataField,_.mw100,_.w100,r)},F.default.createElement("span",{className:(0,Ye.default)(_.formLabel,_.textSecondary,_.dBlock,p)},"SC Call"),F.default.createElement("div",{"data-testid":"confirmScCall",className:(0,Ye.default)(_.textarea,_.scCall,_.formControl,_.mt1,_.textBreak,_.w100,m)},[ze(l),...u].join("@"))),F.default.createElement("div",{className:(0,Ye.default)(_.formGroup,_.mb0,_.dataField,_.mw100,_.w100,r)},F.default.createElement("span",{className:(0,Ye.default)(_.formLabel,_.textSecondary,_.dBlock,p)},"Data"),F.default.createElement("div",{"data-testid":"confirmData",className:(0,Ye.default)(_.textarea,_.formControl,_.mt1,_.textBreak,_.w100,m)},e?d:Ra)))};t();var Ys=require("react"),Y=y(require("classnames"));var Zs=(e=null)=>(0,Ys.useMemo)(()=>({formGroup:(0,Y.default)(_.formGroup,_.textBreak,_.textLeft),formLabel:(0,Y.default)(_.textBreak,_.textSecondary),icon:_.textWhite,contentWrapper:(0,Y.default)(_.dFlex,_.flexColumn,_.justifyContentStart,_.flexRow,_.justifyContentBetween,_.mb3),tokenWrapper:(0,Y.default)(_.mb3,_.mb0,_.dFlex,_.flexColumn,_.alignItemsStart),tokenLabel:(0,Y.default)(_.textSecondary,_.textLeft),tokenValue:(0,Y.default)(_.dFlex,_.alignItemsCenter),scamReport:_.textWarning,scamReportIcon:(0,Y.default)(_.textWarning,_.mr1),tokenAmountLabel:(0,Y.default)(_.textSecondary,_.textLeft),tokenAmountValue:(0,Y.default)(_.dFlex,_.alignItemsCenter),dataFormGroup:(0,Y.default)(_.formGroup,_.textLeft),errorMessage:(0,Y.default)(_.textDanger,_.dFlex,_.justifyContentCenter,_.alignItemsCenter),buttonsWrapper:(0,Y.default)(_.dFlex,_.alignItemsCenter,_.justifyContentEnd,_.mt1),cancelButton:(0,Y.default)(_.btn,_.btnDark,_.textWhite,_.dFlex,_.mr2),signButton:(0,Y.default)(_.btn,e?_.btnWarning:_.btnPrimary,_.dFlex,_.ml2)}),[e]);var ep=({onSignTransaction:e,handleClose:o,onPrev:n,title:r,waitingForDevice:s,currentTransaction:p,error:m,allTransactions:d,isLastTransaction:l,currentStep:u,className:f,signStepInnerClasses:T})=>{let b=ct();if(!p)return null;let h=p.transaction.getData().toString(),{network:A}=Ko(),{buttonsWrapperClassName:D,inputGroupClassName:B,inputLabelClassName:X,inputValueClassName:E,errorClassName:R,scamAlertClassName:z,buttonClassName:re,progressClassName:ye}=T||{},{tokenId:se,nonce:Ee,amount:rn,type:an,multiTxData:Ze,receiver:_t}=p.transactionTokenInfo,cn=Boolean(se&&Xn({tokenId:se,erdLabel:b})),Ao=u===0,sn=op=>{op.preventDefault(),Ao?o():n()},pn=an&&Ze&&!h.endsWith(Ze),N="Sign & Continue";N=s?"Check your Ledger":N,N=l&&!s?"Sign & Submit":N,N=pn?"Continue":N;let{isNft:W}=Ke(se),pe=Ee?`-${Ee}`:"",Je=`${se}${pe}`,{tokenDecimals:mn,tokenAvatar:dn}=Dc({tokenId:Ee&&Ee.length>0?Je:se}),Pe=qe({input:cn?rn:p.transaction.getValue().toString(),decimals:cn?mn:Number(A.decimals),digits:Number(A.digits),showLastNonZeroDecimal:!1,addCommas:!0}),Lo=p.receiverScamInfo,gt=d.length>1,q=Zs(Lo),Eo=W?Je:se||b,Qe=W?rn:Pe;return C.default.createElement(Os,{icon:m?dt.faTimes:null,iconClass:q.icon,iconBgClass:m?_.bgDanger:_.bgWarning,iconSize:"3x",className:f,title:r||"Confirm on Ledger",description:C.default.createElement(C.default.Fragment,null,p.transaction&&C.default.createElement(C.default.Fragment,null,gt&&C.default.createElement(Ws,{totalSteps:d.length,currentStep:u+1,className:(0,te.default)(_.mb4,ye)}),C.default.createElement("div",{"data-testid":"transactionTitle",className:(0,te.default)(q.formGroup,B)},C.default.createElement("div",{className:(0,te.default)(q.formLabel,X)},"To"),C.default.createElement("div",{className:E,"data-testid":"confirmReceiver"},Ze?new Qs.Address(_t).bech32():p.transaction.getReceiver().toString()),Lo&&C.default.createElement("div",{className:(0,te.default)(q.scamReport,z)},C.default.createElement("span",null,C.default.createElement(Js.FontAwesomeIcon,{icon:dt.faExclamationTriangle,className:q.scamReportIcon}),C.default.createElement("small",{"data-testid":"confirmScamReport"},Lo)))),C.default.createElement("div",{className:(0,te.default)(q.tokenWrapper,B)},C.default.createElement("div",{className:(0,te.default)(q.tokenLabel,X)},"Token"),C.default.createElement("div",{className:E,"data-testid":"confirmToken"},C.default.createElement("div",{className:q.tokenValue},C.default.createElement(Le.Icon,{tokenAvatar:dn,token:Eo}),C.default.createElement("div",{className:_.mr2}),C.default.createElement(Le.Label,{token:Eo})))),C.default.createElement("div",{className:B},C.default.createElement("div",{className:(0,te.default)(q.tokenAmountLabel,X)},"Amount"),C.default.createElement("div",{className:(0,te.default)(q.tokenAmountValue,E),"data-testid":"confirmAmount"},Qe)),p.transaction.getData()&&C.default.createElement(Xs,{isScCall:!se,data:p.transaction.getData().toString(),highlight:Ze,className:B,innerTransactionDataClasses:{transactionDataInputLabelClassName:X,transactionDataInputValueClassName:E}}),m&&C.default.createElement("p",{className:(0,te.default)(q.errorMessage,R)},m))),action:C.default.createElement("div",{className:(0,te.default)(q.buttonsWrapper,D)},C.default.createElement("button",{id:"closeButton","data-testid":"closeButton",onClick:sn,className:(0,te.default)(q.cancelButton,re)},Ao?"Cancel":"Back"),C.default.createElement("button",{type:"button",className:(0,te.default)(q.signButton,re),id:"signBtn","data-testid":"signBtn",onClick:e,disabled:s},N))})};t();var o_=`.dapp-core-component__signWithDeviceModalStyles__wallet-connect {
  color: inherit;
}

.dapp-core-component__signWithDeviceModalStyles__modal-container {
  color: inherit;
}

.dapp-core-component__signWithDeviceModalStyles__formGroup {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(o_));var jr={"wallet-connect":"dapp-core-component__signWithDeviceModalStyles__wallet-connect",walletConnect:"dapp-core-component__signWithDeviceModalStyles__wallet-connect","modal-container":"dapp-core-component__signWithDeviceModalStyles__modal-container",modalContainer:"dapp-core-component__signWithDeviceModalStyles__modal-container",formGroup:"dapp-core-component__signWithDeviceModalStyles__formGroup"};var n_=({handleClose:e,error:o,className:n="dapp-device-modal",verifyReceiverScam:r=!0,title:s="Confirm transaction",signStepInnerClasses:p})=>{let{onSignTransaction:m,onPrev:d,allTransactions:l,waitingForDevice:u,onAbort:f,isLastTransaction:T,currentStep:b,callbackRoute:h,currentTransaction:A}=Er({onCancel:e,verifyReceiverScam:r}),D={wrapper:(0,zr.default)(jr.modalContainer,jr.walletConnect,n),container:(0,zr.default)(_.card,_.container),cardBody:_.cardBody};return lt.default.createElement(Ps,{onClose:e,modalConfig:{modalDialogClassName:D.wrapper},visible:A!=null},lt.default.createElement("div",{className:D.cardBody},lt.default.createElement(ep,{onSignTransaction:m,allTransactions:l,onPrev:d,waitingForDevice:u,currentStep:b,isLastTransaction:T,callbackRoute:h,currentTransaction:A,handleClose:f,error:o,title:s,signStepInnerClasses:p})))};0&&(module.exports={SignWithDeviceModal});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
