// @multiversx/sdk-dapp/__commonjs/UI/TransactionInfo/components/transactionInfoFields/TransactionInfoMethod/TransactionInfoMethod.js
"use strict";var Zc=Object.create;var mo=Object.defineProperty,Yc=Object.defineProperties,Jc=Object.getOwnPropertyDescriptor,Qc=Object.getOwnPropertyDescriptors,es=Object.getOwnPropertyNames,Do=Object.getOwnPropertySymbols,os=Object.getPrototypeOf,Rn=Object.prototype.hasOwnProperty,Zt=Object.prototype.propertyIsEnumerable;var Xt=(e,o,n)=>o in e?mo(e,o,{enumerable:!0,configurable:!0,writable:!0,value:n}):e[o]=n,f=(e,o)=>{for(var n in o||(o={}))Rn.call(o,n)&&Xt(e,n,o[n]);if(Do)for(var n of Do(o))Zt.call(o,n)&&Xt(e,n,o[n]);return e},v=(e,o)=>Yc(e,Qc(o));var ye=(e,o)=>{var n={};for(var r in e)Rn.call(e,r)&&o.indexOf(r)<0&&(n[r]=e[r]);if(e!=null&&Do)for(var r of Do(e))o.indexOf(r)<0&&Zt.call(e,r)&&(n[r]=e[r]);return n};var b=(e,o)=>()=>(e&&(o=e(e=0)),o);var H=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Ae=(e,o)=>{for(var n in o)mo(e,n,{get:o[n],enumerable:!0})},Yt=(e,o,n,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let s of es(o))!Rn.call(e,s)&&s!==n&&mo(e,s,{get:()=>o[s],enumerable:!(r=Jc(o,s))||r.enumerable});return e};var T=(e,o,n)=>(n=e!=null?Zc(os(e)):{},Yt(o||!e||!e.__esModule?mo(n,"default",{value:e,enumerable:!0}):n,e)),lo=e=>Yt(mo({},"__esModule",{value:!0}),e);var S=(e,o,n)=>new Promise((r,s)=>{var p=l=>{try{_(n.next(l))}catch(u){s(u)}},m=l=>{try{_(n.throw(l))}catch(u){s(u)}},_=l=>l.done?r(l.value):Promise.resolve(l.value).then(p,m);_((n=n.apply(e,o)).next())});var er=H(Ro=>{"use strict";t();Ro.byteLength=ts;Ro.toByteArray=as;Ro.fromByteArray=ss;var ee=[],X=[],ns=typeof Uint8Array!="undefined"?Uint8Array:Array,Mn="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(Ie=0,Jt=Mn.length;Ie<Jt;++Ie)ee[Ie]=Mn[Ie],X[Mn.charCodeAt(Ie)]=Ie;var Ie,Jt;X["-".charCodeAt(0)]=62;X["_".charCodeAt(0)]=63;function Qt(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var n=e.indexOf("=");n===-1&&(n=o);var r=n===o?0:4-n%4;return[n,r]}function ts(e){var o=Qt(e),n=o[0],r=o[1];return(n+r)*3/4-r}function rs(e,o,n){return(o+n)*3/4-n}function as(e){var o,n=Qt(e),r=n[0],s=n[1],p=new ns(rs(e,r,s)),m=0,_=s>0?r-4:r,l;for(l=0;l<_;l+=4)o=X[e.charCodeAt(l)]<<18|X[e.charCodeAt(l+1)]<<12|X[e.charCodeAt(l+2)]<<6|X[e.charCodeAt(l+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return s===2&&(o=X[e.charCodeAt(l)]<<2|X[e.charCodeAt(l+1)]>>4,p[m++]=o&255),s===1&&(o=X[e.charCodeAt(l)]<<10|X[e.charCodeAt(l+1)]<<4|X[e.charCodeAt(l+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function is(e){return ee[e>>18&63]+ee[e>>12&63]+ee[e>>6&63]+ee[e&63]}function cs(e,o,n){for(var r,s=[],p=o;p<n;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),s.push(is(r));return s.join("")}function ss(e){for(var o,n=e.length,r=n%3,s=[],p=16383,m=0,_=n-r;m<_;m+=p)s.push(cs(e,m,m+p>_?_:m+p));return r===1?(o=e[n-1],s.push(ee[o>>2]+ee[o<<4&63]+"==")):r===2&&(o=(e[n-2]<<8)+e[n-1],s.push(ee[o>>10]+ee[o>>4&63]+ee[o<<2&63]+"=")),s.join("")}});var or=H(On=>{t();On.read=function(e,o,n,r,s){var p,m,_=s*8-r-1,l=(1<<_)-1,u=l>>1,y=-7,x=n?s-1:0,w=n?-1:1,E=e[o+x];for(x+=w,p=E&(1<<-y)-1,E>>=-y,y+=_;y>0;p=p*256+e[o+x],x+=w,y-=8);for(m=p&(1<<-y)-1,p>>=-y,y+=r;y>0;m=m*256+e[o+x],x+=w,y-=8);if(p===0)p=1-u;else{if(p===l)return m?NaN:(E?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-u}return(E?-1:1)*m*Math.pow(2,p-r)};On.write=function(e,o,n,r,s,p){var m,_,l,u=p*8-s-1,y=(1<<u)-1,x=y>>1,w=s===23?Math.pow(2,-24)-Math.pow(2,-77):0,E=r?0:p-1,B=r?1:-1,G=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(_=isNaN(o)?1:0,m=y):(m=Math.floor(Math.log(o)/Math.LN2),o*(l=Math.pow(2,-m))<1&&(m--,l*=2),m+x>=1?o+=w/l:o+=w*Math.pow(2,1-x),o*l>=2&&(m++,l/=2),m+x>=y?(_=0,m=y):m+x>=1?(_=(o*l-1)*Math.pow(2,s),m=m+x):(_=o*Math.pow(2,x-1)*Math.pow(2,s),m=0));s>=8;e[n+E]=_&255,E+=B,_/=256,s-=8);for(m=m<<s|_,u+=s;u>0;e[n+E]=m&255,E+=B,m/=256,u-=8);e[n+E-B]|=G*128}});var gr=H(Ve=>{"use strict";t();var Fn=er(),$e=or(),nr=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;Ve.Buffer=d;Ve.SlowBuffer=gs;Ve.INSPECT_MAX_BYTES=50;var Mo=2147483647;Ve.kMaxLength=Mo;d.TYPED_ARRAY_SUPPORT=ps();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function ps(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(n){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function pe(e){if(e>Mo)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,n){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Gn(e)}return ar(e,o,n)}d.poolSize=8192;function ar(e,o,n){if(typeof e=="string")return ds(e,o);if(ArrayBuffer.isView(e))return ls(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(oe(e,ArrayBuffer)||e&&oe(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(oe(e,SharedArrayBuffer)||e&&oe(e.buffer,SharedArrayBuffer)))return Un(e,o,n);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,o,n);var s=_s(e);if(s)return s;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,n);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,n){return ar(e,o,n)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function ir(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function ms(e,o,n){return ir(e),e<=0?pe(e):o!==void 0?typeof n=="string"?pe(e).fill(o,n):pe(e).fill(o):pe(e)}d.alloc=function(e,o,n){return ms(e,o,n)};function Gn(e){return ir(e),pe(e<0?0:$n(e)|0)}d.allocUnsafe=function(e){return Gn(e)};d.allocUnsafeSlow=function(e){return Gn(e)};function ds(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var n=cr(e,o)|0,r=pe(n),s=r.write(e,o);return s!==n&&(r=r.slice(0,s)),r}function Bn(e){for(var o=e.length<0?0:$n(e.length)|0,n=pe(o),r=0;r<o;r+=1)n[r]=e[r]&255;return n}function ls(e){if(oe(e,Uint8Array)){var o=new Uint8Array(e);return Un(o.buffer,o.byteOffset,o.byteLength)}return Bn(e)}function Un(e,o,n){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(n||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&n===void 0?r=new Uint8Array(e):n===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,n),Object.setPrototypeOf(r,d.prototype),r}function _s(e){if(d.isBuffer(e)){var o=$n(e.length)|0,n=pe(o);return n.length===0||e.copy(n,0,0,o),n}if(e.length!==void 0)return typeof e.length!="number"||Vn(e.length)?pe(0):Bn(e);if(e.type==="Buffer"&&Array.isArray(e.data))return Bn(e.data)}function $n(e){if(e>=Mo)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+Mo.toString(16)+" bytes");return e|0}function gs(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,n){if(oe(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),oe(n,Uint8Array)&&(n=d.from(n,n.offset,n.byteLength)),!d.isBuffer(o)||!d.isBuffer(n))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===n)return 0;for(var r=o.length,s=n.length,p=0,m=Math.min(r,s);p<m;++p)if(o[p]!==n[p]){r=o[p],s=n[p];break}return r<s?-1:s<r?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,n){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var r;if(n===void 0)for(n=0,r=0;r<o.length;++r)n+=o[r].length;var s=d.allocUnsafe(n),p=0;for(r=0;r<o.length;++r){var m=o[r];if(oe(m,Uint8Array))p+m.length>s.length?d.from(m).copy(s,p):Uint8Array.prototype.set.call(s,m,p);else if(d.isBuffer(m))m.copy(s,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return s};function cr(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||oe(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var n=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&n===0)return 0;for(var s=!1;;)switch(o){case"ascii":case"latin1":case"binary":return n;case"utf8":case"utf-8":return Wn(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return n*2;case"hex":return n>>>1;case"base64":return _r(e).length;default:if(s)return r?-1:Wn(e).length;o=(""+o).toLowerCase(),s=!0}}d.byteLength=cr;function us(e,o,n){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((n===void 0||n>this.length)&&(n=this.length),n<=0)||(n>>>=0,o>>>=0,n<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return ks(this,o,n);case"utf8":case"utf-8":return pr(this,o,n);case"ascii":return ws(this,o,n);case"latin1":case"binary":return vs(this,o,n);case"base64":return bs(this,o,n);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return As(this,o,n);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function Le(e,o,n){var r=e[o];e[o]=e[n],e[n]=r}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var n=0;n<o;n+=2)Le(this,n,n+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var n=0;n<o;n+=4)Le(this,n,n+3),Le(this,n+1,n+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var n=0;n<o;n+=8)Le(this,n,n+7),Le(this,n+1,n+6),Le(this,n+2,n+5),Le(this,n+3,n+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?pr(this,0,o):us.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",n=Ve.INSPECT_MAX_BYTES;return o=this.toString("hex",0,n).replace(/(.{2})/g,"$1 ").trim(),this.length>n&&(o+=" ... "),"<Buffer "+o+">"};nr&&(d.prototype[nr]=d.prototype.inspect);d.prototype.compare=function(o,n,r,s,p){if(oe(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(n===void 0&&(n=0),r===void 0&&(r=o?o.length:0),s===void 0&&(s=0),p===void 0&&(p=this.length),n<0||r>o.length||s<0||p>this.length)throw new RangeError("out of range index");if(s>=p&&n>=r)return 0;if(s>=p)return-1;if(n>=r)return 1;if(n>>>=0,r>>>=0,s>>>=0,p>>>=0,this===o)return 0;for(var m=p-s,_=r-n,l=Math.min(m,_),u=this.slice(s,p),y=o.slice(n,r),x=0;x<l;++x)if(u[x]!==y[x]){m=u[x],_=y[x];break}return m<_?-1:_<m?1:0};function sr(e,o,n,r,s){if(e.length===0)return-1;if(typeof n=="string"?(r=n,n=0):n>2147483647?n=2147483647:n<-2147483648&&(n=-2147483648),n=+n,Vn(n)&&(n=s?0:e.length-1),n<0&&(n=e.length+n),n>=e.length){if(s)return-1;n=e.length-1}else if(n<0)if(s)n=0;else return-1;if(typeof o=="string"&&(o=d.from(o,r)),d.isBuffer(o))return o.length===0?-1:tr(e,o,n,r,s);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?s?Uint8Array.prototype.indexOf.call(e,o,n):Uint8Array.prototype.lastIndexOf.call(e,o,n):tr(e,[o],n,r,s);throw new TypeError("val must be string, number or Buffer")}function tr(e,o,n,r,s){var p=1,m=e.length,_=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,_/=2,n/=2}function l(E,B){return p===1?E[B]:E.readUInt16BE(B*p)}var u;if(s){var y=-1;for(u=n;u<m;u++)if(l(e,u)===l(o,y===-1?0:u-y)){if(y===-1&&(y=u),u-y+1===_)return y*p}else y!==-1&&(u-=u-y),y=-1}else for(n+_>m&&(n=m-_),u=n;u>=0;u--){for(var x=!0,w=0;w<_;w++)if(l(e,u+w)!==l(o,w)){x=!1;break}if(x)return u}return-1}d.prototype.includes=function(o,n,r){return this.indexOf(o,n,r)!==-1};d.prototype.indexOf=function(o,n,r){return sr(this,o,n,r,!0)};d.prototype.lastIndexOf=function(o,n,r){return sr(this,o,n,r,!1)};function fs(e,o,n,r){n=Number(n)||0;var s=e.length-n;r?(r=Number(r),r>s&&(r=s)):r=s;var p=o.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var _=parseInt(o.substr(m*2,2),16);if(Vn(_))return m;e[n+m]=_}return m}function Ts(e,o,n,r){return Oo(Wn(o,e.length-n),e,n,r)}function hs(e,o,n,r){return Oo(Es(o),e,n,r)}function xs(e,o,n,r){return Oo(_r(o),e,n,r)}function ys(e,o,n,r){return Oo(Ns(o,e.length-n),e,n,r)}d.prototype.write=function(o,n,r,s){if(n===void 0)s="utf8",r=this.length,n=0;else if(r===void 0&&typeof n=="string")s=n,r=this.length,n=0;else if(isFinite(n))n=n>>>0,isFinite(r)?(r=r>>>0,s===void 0&&(s="utf8")):(s=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-n;if((r===void 0||r>p)&&(r=p),o.length>0&&(r<0||n<0)||n>this.length)throw new RangeError("Attempt to write outside buffer bounds");s||(s="utf8");for(var m=!1;;)switch(s){case"hex":return fs(this,o,n,r);case"utf8":case"utf-8":return Ts(this,o,n,r);case"ascii":case"latin1":case"binary":return hs(this,o,n,r);case"base64":return xs(this,o,n,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return ys(this,o,n,r);default:if(m)throw new TypeError("Unknown encoding: "+s);s=(""+s).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function bs(e,o,n){return o===0&&n===e.length?Fn.fromByteArray(e):Fn.fromByteArray(e.slice(o,n))}function pr(e,o,n){n=Math.min(e.length,n);for(var r=[],s=o;s<n;){var p=e[s],m=null,_=p>239?4:p>223?3:p>191?2:1;if(s+_<=n){var l,u,y,x;switch(_){case 1:p<128&&(m=p);break;case 2:l=e[s+1],(l&192)===128&&(x=(p&31)<<6|l&63,x>127&&(m=x));break;case 3:l=e[s+1],u=e[s+2],(l&192)===128&&(u&192)===128&&(x=(p&15)<<12|(l&63)<<6|u&63,x>2047&&(x<55296||x>57343)&&(m=x));break;case 4:l=e[s+1],u=e[s+2],y=e[s+3],(l&192)===128&&(u&192)===128&&(y&192)===128&&(x=(p&15)<<18|(l&63)<<12|(u&63)<<6|y&63,x>65535&&x<1114112&&(m=x))}}m===null?(m=65533,_=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),s+=_}return Ss(r)}var rr=4096;function Ss(e){var o=e.length;if(o<=rr)return String.fromCharCode.apply(String,e);for(var n="",r=0;r<o;)n+=String.fromCharCode.apply(String,e.slice(r,r+=rr));return n}function ws(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]&127);return r}function vs(e,o,n){var r="";n=Math.min(e.length,n);for(var s=o;s<n;++s)r+=String.fromCharCode(e[s]);return r}function ks(e,o,n){var r=e.length;(!o||o<0)&&(o=0),(!n||n<0||n>r)&&(n=r);for(var s="",p=o;p<n;++p)s+=Cs[e[p]];return s}function As(e,o,n){for(var r=e.slice(o,n),s="",p=0;p<r.length-1;p+=2)s+=String.fromCharCode(r[p]+r[p+1]*256);return s}d.prototype.slice=function(o,n){var r=this.length;o=~~o,n=n===void 0?r:~~n,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),n<0?(n+=r,n<0&&(n=0)):n>r&&(n=r),n<o&&(n=o);var s=this.subarray(o,n);return Object.setPrototypeOf(s,d.prototype),s};function R(e,o,n){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>n)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||R(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return s};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||R(o,n,this.length);for(var s=this[o+--n],p=1;n>0&&(p*=256);)s+=this[o+--n]*p;return s};d.prototype.readUint8=d.prototype.readUInt8=function(o,n){return o=o>>>0,n||R(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,n){return o=o>>>0,n||R(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,n){return o=o>>>0,n||R(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,n){return o=o>>>0,n||R(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,n){return o=o>>>0,n||R(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,n,r){o=o>>>0,n=n>>>0,r||R(o,n,this.length);for(var s=this[o],p=1,m=0;++m<n&&(p*=256);)s+=this[o+m]*p;return p*=128,s>=p&&(s-=Math.pow(2,8*n)),s};d.prototype.readIntBE=function(o,n,r){o=o>>>0,n=n>>>0,r||R(o,n,this.length);for(var s=n,p=1,m=this[o+--s];s>0&&(p*=256);)m+=this[o+--s]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*n)),m};d.prototype.readInt8=function(o,n){return o=o>>>0,n||R(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,n){o=o>>>0,n||R(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(o,n){o=o>>>0,n||R(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(o,n){return o=o>>>0,n||R(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,n){return o=o>>>0,n||R(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,n){return o=o>>>0,n||R(o,4,this.length),$e.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,n){return o=o>>>0,n||R(o,4,this.length),$e.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,n){return o=o>>>0,n||R(o,8,this.length),$e.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,n){return o=o>>>0,n||R(o,8,this.length),$e.read(this,o,!1,52,8)};function $(e,o,n,r,s,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>s||o<p)throw new RangeError('"value" argument is out of bounds');if(n+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;$(this,o,n,r,p,0)}var m=1,_=0;for(this[n]=o&255;++_<r&&(m*=256);)this[n+_]=o/m&255;return n+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;$(this,o,n,r,p,0)}var m=r-1,_=1;for(this[n+m]=o&255;--m>=0&&(_*=256);)this[n+m]=o/_&255;return n+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,1,255,0),this[n]=o&255,n+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,2,65535,0),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,2,65535,0),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,4,4294967295,0),this[n+3]=o>>>24,this[n+2]=o>>>16,this[n+1]=o>>>8,this[n]=o&255,n+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,4,4294967295,0),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};d.prototype.writeIntLE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);$(this,o,n,r,p-1,-p)}var m=0,_=1,l=0;for(this[n]=o&255;++m<r&&(_*=256);)o<0&&l===0&&this[n+m-1]!==0&&(l=1),this[n+m]=(o/_>>0)-l&255;return n+r};d.prototype.writeIntBE=function(o,n,r,s){if(o=+o,n=n>>>0,!s){var p=Math.pow(2,8*r-1);$(this,o,n,r,p-1,-p)}var m=r-1,_=1,l=0;for(this[n+m]=o&255;--m>=0&&(_*=256);)o<0&&l===0&&this[n+m+1]!==0&&(l=1),this[n+m]=(o/_>>0)-l&255;return n+r};d.prototype.writeInt8=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,1,127,-128),o<0&&(o=255+o+1),this[n]=o&255,n+1};d.prototype.writeInt16LE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,2,32767,-32768),this[n]=o&255,this[n+1]=o>>>8,n+2};d.prototype.writeInt16BE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,2,32767,-32768),this[n]=o>>>8,this[n+1]=o&255,n+2};d.prototype.writeInt32LE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,4,2147483647,-2147483648),this[n]=o&255,this[n+1]=o>>>8,this[n+2]=o>>>16,this[n+3]=o>>>24,n+4};d.prototype.writeInt32BE=function(o,n,r){return o=+o,n=n>>>0,r||$(this,o,n,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[n]=o>>>24,this[n+1]=o>>>16,this[n+2]=o>>>8,this[n+3]=o&255,n+4};function mr(e,o,n,r,s,p){if(n+r>e.length)throw new RangeError("Index out of range");if(n<0)throw new RangeError("Index out of range")}function dr(e,o,n,r,s){return o=+o,n=n>>>0,s||mr(e,o,n,4,34028234663852886e22,-34028234663852886e22),$e.write(e,o,n,r,23,4),n+4}d.prototype.writeFloatLE=function(o,n,r){return dr(this,o,n,!0,r)};d.prototype.writeFloatBE=function(o,n,r){return dr(this,o,n,!1,r)};function lr(e,o,n,r,s){return o=+o,n=n>>>0,s||mr(e,o,n,8,17976931348623157e292,-17976931348623157e292),$e.write(e,o,n,r,52,8),n+8}d.prototype.writeDoubleLE=function(o,n,r){return lr(this,o,n,!0,r)};d.prototype.writeDoubleBE=function(o,n,r){return lr(this,o,n,!1,r)};d.prototype.copy=function(o,n,r,s){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!s&&s!==0&&(s=this.length),n>=o.length&&(n=o.length),n||(n=0),s>0&&s<r&&(s=r),s===r||o.length===0||this.length===0)return 0;if(n<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(s<0)throw new RangeError("sourceEnd out of bounds");s>this.length&&(s=this.length),o.length-n<s-r&&(s=o.length-n+r);var p=s-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(n,r,s):Uint8Array.prototype.set.call(o,this.subarray(r,s),n),p};d.prototype.fill=function(o,n,r,s){if(typeof o=="string"){if(typeof n=="string"?(s=n,n=0,r=this.length):typeof r=="string"&&(s=r,r=this.length),s!==void 0&&typeof s!="string")throw new TypeError("encoding must be a string");if(typeof s=="string"&&!d.isEncoding(s))throw new TypeError("Unknown encoding: "+s);if(o.length===1){var p=o.charCodeAt(0);(s==="utf8"&&p<128||s==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(n<0||this.length<n||this.length<r)throw new RangeError("Out of range index");if(r<=n)return this;n=n>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=n;m<r;++m)this[m]=o;else{var _=d.isBuffer(o)?o:d.from(o,s),l=_.length;if(l===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-n;++m)this[m+n]=_[m%l]}return this};var Is=/[^+/0-9A-Za-z-_]/g;function Ls(e){if(e=e.split("=")[0],e=e.trim().replace(Is,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Wn(e,o){o=o||1/0;for(var n,r=e.length,s=null,p=[],m=0;m<r;++m){if(n=e.charCodeAt(m),n>55295&&n<57344){if(!s){if(n>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&p.push(239,191,189);continue}s=n;continue}if(n<56320){(o-=3)>-1&&p.push(239,191,189),s=n;continue}n=(s-55296<<10|n-56320)+65536}else s&&(o-=3)>-1&&p.push(239,191,189);if(s=null,n<128){if((o-=1)<0)break;p.push(n)}else if(n<2048){if((o-=2)<0)break;p.push(n>>6|192,n&63|128)}else if(n<65536){if((o-=3)<0)break;p.push(n>>12|224,n>>6&63|128,n&63|128)}else if(n<1114112){if((o-=4)<0)break;p.push(n>>18|240,n>>12&63|128,n>>6&63|128,n&63|128)}else throw new Error("Invalid code point")}return p}function Es(e){for(var o=[],n=0;n<e.length;++n)o.push(e.charCodeAt(n)&255);return o}function Ns(e,o){for(var n,r,s,p=[],m=0;m<e.length&&!((o-=2)<0);++m)n=e.charCodeAt(m),r=n>>8,s=n%256,p.push(s),p.push(r);return p}function _r(e){return Fn.toByteArray(Ls(e))}function Oo(e,o,n,r){for(var s=0;s<r&&!(s+n>=o.length||s>=e.length);++s)o[s+n]=e[s];return s}function oe(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Vn(e){return e!==e}var Cs=function(){for(var e="0123456789abcdef",o=new Array(256),n=0;n<16;++n)for(var r=n*16,s=0;s<16;++s)o[r+s]=e[n]+e[s];return o}()});var xr=H((Nl,hr)=>{t();var P=hr.exports={},ne,te;function Hn(){throw new Error("setTimeout has not been defined")}function jn(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?ne=setTimeout:ne=Hn}catch(e){ne=Hn}try{typeof clearTimeout=="function"?te=clearTimeout:te=jn}catch(e){te=jn}})();function ur(e){if(ne===setTimeout)return setTimeout(e,0);if((ne===Hn||!ne)&&setTimeout)return ne=setTimeout,setTimeout(e,0);try{return ne(e,0)}catch(o){try{return ne.call(null,e,0)}catch(n){return ne.call(this,e,0)}}}function Ps(e){if(te===clearTimeout)return clearTimeout(e);if((te===jn||!te)&&clearTimeout)return te=clearTimeout,clearTimeout(e);try{return te(e)}catch(o){try{return te.call(null,e)}catch(n){return te.call(this,e)}}}var me=[],He=!1,Ee,Fo=-1;function Ds(){!He||!Ee||(He=!1,Ee.length?me=Ee.concat(me):Fo=-1,me.length&&fr())}function fr(){if(!He){var e=ur(Ds);He=!0;for(var o=me.length;o;){for(Ee=me,me=[];++Fo<o;)Ee&&Ee[Fo].run();Fo=-1,o=me.length}Ee=null,He=!1,Ps(e)}}P.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var n=1;n<arguments.length;n++)o[n-1]=arguments[n];me.push(new Tr(e,o)),me.length===1&&!He&&ur(fr)};function Tr(e,o){this.fun=e,this.array=o}Tr.prototype.run=function(){this.fun.apply(null,this.array)};P.title="browser";P.browser=!0;P.env={};P.argv=[];P.version="";P.versions={};function de(){}P.on=de;P.addListener=de;P.once=de;P.off=de;P.removeListener=de;P.removeAllListeners=de;P.emit=de;P.prependListener=de;P.prependOnceListener=de;P.listeners=function(e){return[]};P.binding=function(e){throw new Error("process.binding is not supported")};P.cwd=function(){return"/"};P.chdir=function(e){throw new Error("process.chdir is not supported")};P.umask=function(){return 0}});var i,c,Rs,a,t=b(()=>{i=T(gr()),c=T(xr()),Rs=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var n=__magic__;return n}(Object),a=Rs});var U=b(()=>{"use strict";t()});var Ne=b(()=>{"use strict";t()});var wr=b(()=>{"use strict";t()});var vr=b(()=>{"use strict";t()});var kr=b(()=>{"use strict";t()});var C=b(()=>{"use strict";t()});var Ar=b(()=>{"use strict";t()});var Ir=b(()=>{"use strict";t()});var Lr=b(()=>{"use strict";t()});var Er=b(()=>{"use strict";t()});var M=b(()=>{"use strict";t();wr();vr();kr();C();Ar();Ir();U();Lr();Er()});var Kn,Nr,s_,Cr,p_,Pr,Dr,m_,d_,Rr=b(()=>{"use strict";t();M();Kn={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",apiAddress:"https://devnet-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",apiAddress:"https://testnet-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",apiAddress:"https://api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",apiTimeout:"4000"}},{chainId:Nr,egldLabel:s_}=Kn["devnet"],{chainId:Cr,egldLabel:p_}=Kn["testnet"],{chainId:Pr,egldLabel:Dr}=Kn["mainnet"],m_={["devnet"]:Nr,["testnet"]:Cr,["mainnet"]:Pr},d_={[Nr]:"devnet",[Cr]:"testnet",[Pr]:"mainnet"}});var Xn=b(()=>{"use strict";t()});var Mr=b(()=>{"use strict";t()});var Go,Zn=b(()=>{"use strict";t();Go="transaction-status-toast-id"});var re,Or=b(()=>{"use strict";t();re=require("@multiversx/sdk-web-wallet-provider")});var Fr=b(()=>{"use strict";t()});var _o,$o,W,je,Br,Vo,Ur,J,Wr,A=b(()=>{"use strict";t();Ne();Rr();Xn();Mr();Zn();Or();Fr();_o=5e4,$o=1e9,W=18,je=4,Br=1,Vo="logout",Ur="login",J="0",Wr="..."});var go,Ho=b(()=>{"use strict";t();go=()=>Date.now()/1e3});var zr=b(()=>{"use strict";t()});var qr=b(()=>{"use strict";t()});var Jn=b(()=>{"use strict";t();Ho();zr();qr()});var et={};Ae(et,{getItem:()=>Us,localStorageKeys:()=>be,removeItem:()=>Ws,setItem:()=>Bs});var be,Qn,Bs,Us,Ws,ze=b(()=>{"use strict";t();Jn();be={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event"},Qn=typeof localStorage!="undefined",Bs=({key:e,data:o,expires:n})=>{!Qn||localStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Us=e=>{if(!Qn)return;let o=localStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:go()>=n.expires?(localStorage.removeItem(String(e)),null):n.data},Ws=e=>{!Qn||localStorage.removeItem(String(e))}});var ot={};Ae(ot,{clear:()=>Yr,getItem:()=>Xr,removeItem:()=>Zr,setItem:()=>Kr,storage:()=>Gs});var Kr,Xr,Zr,Yr,Gs,Jr=b(()=>{"use strict";t();Kr=({key:e,data:o,expires:n})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:n,data:o}))},Xr=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let n=JSON.parse(o);return!n||!n.hasOwnProperty("expires")||!n.hasOwnProperty("data")?null:Date.now()>=n.expires?(sessionStorage.removeItem(String(e)),null):n.data},Zr=e=>sessionStorage.removeItem(String(e)),Yr=()=>sessionStorage.clear(),Gs={setItem:Kr,getItem:Xr,removeItem:Zr,clear:Yr}});var Ce,qe=b(()=>{"use strict";t();ze();Jr();Ce={session:ot,local:et}});var nt,O,ae,F=b(()=>{"use strict";t();nt=require("@reduxjs/toolkit");A();O=(0,nt.createAction)(Vo),ae=(0,nt.createAction)(Ur,e=>({payload:e}))});var rt,Qr,ea,jo,tt,oa,at,ig,it,cg,$s,Vs,sg,pg,mg,Hs,zo,qo=b(()=>{"use strict";t();rt=require("@multiversx/sdk-core"),Qr=require("@reduxjs/toolkit"),ea=require("redux-persist");A();qe();ze();F();jo={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:J},tt={address:"",websocketEvent:null,accounts:{"":jo},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},oa=(0,Qr.createSlice)({name:"accountInfoSlice",initialState:tt,reducers:{setAddress:(e,o)=>{let n=o.payload;e.address=n,e.publicKey=new rt.Address(n).hex()},setAccount:(e,o)=>{let n=e.address===o.payload.address;e.accounts={[e.address]:n?o.payload:jo},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:n}=e;e.accounts[n].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(O,()=>(Ce.local.removeItem(be.loginExpiresAt),tt)),e.addCase(ae,(o,n)=>{let{address:r}=n.payload;o.address=r,o.publicKey=new rt.Address(r).hex()}),e.addCase(ea.REHYDRATE,(o,n)=>{var u;if(!((u=n.payload)!=null&&u.account))return;let{account:r}=n.payload,{address:s,shard:p,accounts:m,publicKey:_}=r;o.address=s,o.shard=p;let l=m&&s in m;o.accounts=l?m:tt.accounts,o.publicKey=_})}}),{setAccount:at,setAddress:ig,setAccountNonce:it,setAccountShard:cg,setLedgerAccount:$s,updateLedgerAccount:Vs,setWalletConnectAccount:sg,setIsAccountLoading:pg,setAccountLoadingError:mg,setWebsocketEvent:Hs}=oa.actions,zo=oa.reducer});function uo(){return new Date().setHours(new Date().getHours()+24)}function fo(e){Ce.local.setItem({key:be.loginExpiresAt,data:e,expires:e})}var ct=b(()=>{"use strict";t();qe();ze()});var ta,na,ra,yg,aa,js,zs,bg,qs,ia,Sg,Ko,Xo=b(()=>{"use strict";t();ta=require("@reduxjs/toolkit");ct();C();F();na={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,operaLogin:null,isLoginSessionInvalid:!1},ra=(0,ta.createSlice)({name:"loginInfoSlice",initialState:na,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload}},extraReducers:e=>{e.addCase(O,()=>na),e.addCase(ae,(o,n)=>{o.loginMethod=n.payload.loginMethod,fo(uo())})}}),{setLoginMethod:yg,setWalletConnectLogin:aa,setLedgerLogin:js,setTokenLogin:zs,setTokenLoginSignature:bg,setWalletLogin:qs,invalidateLoginSession:ia,setLogoutRoute:Sg}=ra.actions,Ko=ra.reducer});var sa,ca,pa,Ag,Ks,Ig,Xs,Zo,Yo=b(()=>{"use strict";t();sa=require("@reduxjs/toolkit");F();ca={},pa=(0,sa.createSlice)({name:"modalsSlice",initialState:ca,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(O,()=>ca)}}),{setTxSubmittedModal:Ag,setNotificationModal:Ks,clearTxSubmittedModal:Ig,clearNotificationModal:Xs}=pa.actions,Zo=pa.reducer});var To=b(()=>{"use strict";t()});var da=b(()=>{"use strict";t();To()});function st(e){return e[Math.floor(Math.random()*e.length)]}var la=b(()=>{"use strict";t()});var pt=b(()=>{"use strict";t();Xn()});var Pe=b(()=>{"use strict";t();da();la();pt()});var _a,ga,mt,Zs,ua,$g,Vg,Jo,Qo=b(()=>{"use strict";t();_a=require("@reduxjs/toolkit"),ga=T(require("lodash.omit"));Pe();mt={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},Zs={network:mt,chainID:"-1"},ua=(0,_a.createSlice)({name:"appConfig",initialState:Zs,reducers:{initializeNetworkConfig:(e,o)=>{let n=st(o.payload.walletConnectBridgeAddresses),r=st(o.payload.walletConnectV2RelayAddresses),s=(0,ga.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=v(f(f({},e.network),s),{walletConnectBridgeAddress:n,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:$g,setChainID:Vg}=ua.actions,Jo=ua.reducer});var fa,dt,Ta,Xg,Zg,Yg,en,on=b(()=>{"use strict";t();fa=require("@reduxjs/toolkit");M();F();dt={isSigning:!1,signedSessions:{}},Ta=(0,fa.createSlice)({name:"signedMessageInfoSliceState",initialState:dt,reducers:{setSignSession:(e,o)=>{let{sessionId:n,signedSession:r,errorMessage:s}=o.payload;s&&(e.errorMessage=s),e.isSigning=r.status==="pending",e.signedSessions[n]=r},setSignSessionState:(e,o)=>f(f({},e),o.payload),clearSignedMessageInfo:()=>dt},extraReducers:e=>{e.addCase(O,()=>dt)}}),{setSignSession:Xg,clearSignedMessageInfo:Zg,setSignSessionState:Yg}=Ta.actions,en=Ta.reducer});var xa,ya,ha,ba,Ys,lt,au,iu,cu,su,nn,tn=b(()=>{"use strict";t();xa=require("@reduxjs/toolkit"),ya=require("redux-persist");Zn();M();Ho();F();ha={customToasts:[],transactionToasts:[],failTransactionToast:null},ba=(0,xa.createSlice)({name:"toastsSlice",initialState:ha,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(v(f({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(n=>n.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:go(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(n=>n.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=v(f({},o.payload),{toastId:Go})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(O,()=>ha),e.addCase(ya.REHYDRATE,(o,n)=>{var s,p;let r=(p=(s=n.customToasts)==null?void 0:s.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=r})}}),{addCustomToast:Ys,removeCustomToast:lt,addTransactionToast:au,removeTransactionToast:iu,addFailTransactionToast:cu,removeFailTransactionToast:su}=ba.actions,nn=ba.reducer});var Sa,gt,ut,ft,Js,_t,wa,lu,Qs,Tt,rn,an=b(()=>{"use strict";t();Sa=require("@reduxjs/toolkit");F();gt="Transaction failed",ut="Transaction successful",ft="Processing transaction",Js="Transaction submitted",_t={},wa=(0,Sa.createSlice)({name:"transactionsInfo",initialState:_t,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:n,transactionsDisplayInfo:r}=o.payload;n!=null&&(e[n]={errorMessage:(r==null?void 0:r.errorMessage)||gt,successMessage:(r==null?void 0:r.successMessage)||ut,processingMessage:(r==null?void 0:r.processingMessage)||ft,submittedMessage:(r==null?void 0:r.submittedMessage)||Js,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>_t},extraReducers:e=>{e.addCase(O,()=>_t)}}),{clearTransactionsInfo:lu,setTransactionsDisplayInfo:Qs,clearTransactionsInfoForSessionId:Tt}=wa.actions,rn=wa.reducer});function Xe(e){return e!=null&&(cp(e)||dp(e))}function Ze(e){return e!=null&&(sp(e)||lp(e))}function Ye(e){return e!=null&&(pp(e)||_p(e))}function ht(e){return e!=null&&mp(e)}function cp(e){return e!=null&&ep.includes(e)}function sp(e){return e!=null&&op.includes(e)}function pp(e){return e!=null&&np.includes(e)}function mp(e){return e!=null&&tp.includes(e)}function dp(e){return e!=null&&rp.includes(e)}function lp(e){return e!=null&&ap.includes(e)}function _p(e){return e!=null&&ip.includes(e)}var ep,op,np,tp,rp,ap,ip,Je=b(()=>{"use strict";t();C();ep=["sent"],op=["success"],np=["fail","cancelled","timedOut"],tp=["timedOut"],rp=["pending"],ap=["success"],ip=["fail","invalid"]});var va,ka,ho,gp,Aa,Ia,La,up,cn,fp,Tp,bu,hp,sn,xt,pn,mn=b(()=>{"use strict";t();va=require("@reduxjs/toolkit"),ka=require("redux-persist");C();Je();F();ho={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},gp={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Aa=(0,va.createSlice)({name:"transactionsSlice",initialState:ho,reducers:{moveTransactionsToSignedState:(e,o)=>{var l,u;let{sessionId:n,transactions:r,errorMessage:s,status:p,redirectRoute:m}=o.payload,_=((l=e.customTransactionInformationForSessionId)==null?void 0:l[n])||gp;e.signedTransactions[n]={transactions:r,status:p,errorMessage:s,redirectRoute:m,customTransactionInformation:_},((u=e==null?void 0:e.transactionsToSign)==null?void 0:u.sessionId)===n&&(e.transactionsToSign=ho.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=f(f({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:n,status:r,errorMessage:s,transactions:p}=o.payload;e.signedTransactions[n]!=null&&(e.signedTransactions[n].status=r,s!=null&&(e.signedTransactions[n].errorMessage=s),p!=null&&(e.signedTransactions[n].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var _,l,u,y,x,w;let{sessionId:n,status:r,errorMessage:s,transactionHash:p}=o.payload,m=(l=(_=e.signedTransactions)==null?void 0:_[n])==null?void 0:l.transactions;if(m!=null){e.signedTransactions[n].transactions=m.map(G=>G.hash===p?v(f({},G),{status:r,errorMessage:s}):G);let E=(y=(u=e.signedTransactions[n])==null?void 0:u.transactions)==null?void 0:y.every(G=>Ze(G.status)),B=(w=(x=e.signedTransactions[n])==null?void 0:x.transactions)==null?void 0:w.every(G=>Ye(G.status));E&&(e.signedTransactions[n].status="success"),B&&(e.signedTransactions[n].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:n,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[n]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=ho.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=ho.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(O,()=>ho),e.addCase(ka.REHYDRATE,(o,n)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=n.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:s}=n.payload.transactions,p=Object.entries(r).reduce((_,[l,u])=>{let y=new Date(l),x=new Date;return x.setHours(x.getHours()+5),x-y>0||(_[l]=u),_},{});s!=null&&(o.customTransactionInformationForSessionId=s),r!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:Ia,updateSignedTransactions:La,setTransactionsToSign:up,clearAllTransactionsToSign:cn,clearAllSignedTransactions:fp,clearSignedTransaction:Tp,clearTransactionToSign:bu,setSignTransactionsError:hp,setSignTransactionsCancelMessage:sn,moveTransactionsToSignedState:xt}=Aa.actions,pn=Aa.reducer});var L=b(()=>{"use strict";t();qo();Xo();Yo();Qo();on();tn();an();mn()});var yt=b(()=>{"use strict";t()});var Da,Pa,Ra,ju,zu,qu,Ku,Xu,Zu,Ma,Oa=b(()=>{"use strict";t();Da=require("@reduxjs/toolkit");F();Pa={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Ra=(0,Da.createSlice)({name:"dappModalsSlice",initialState:Pa,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=f({},o.payload)}},extraReducers:e=>{e.addCase(O,()=>Pa)}}),{setDappModalVisibility:ju,setDappModalHeaderVisibility:zu,setDappModalFooterVisibility:qu,setDappModalHeaderText:Ku,setDappModalFooterText:Xu,setDappModalConfig:Zu}=Ra.actions,Ma=Ra.reducer});var Fa,Sp,wp,Re,gn=b(()=>{"use strict";t();Fa=require("@reduxjs/toolkit");yt();qo();Oa();Xo();Yo();Qo();on();tn();an();mn();Sp={["account"]:zo,["dappModal"]:Ma,["loginInfo"]:Ko,["modals"]:Zo,["networkConfig"]:Jo,["signedMessageInfo"]:en,["toasts"]:nn,["transactionsInfo"]:rn,["transactions"]:pn},wp=(e={})=>(0,Fa.combineReducers)(f(f({},Sp),e)),Re=wp});var Ga={};Ae(Ga,{default:()=>Bp,sessionStorageReducers:()=>bt});function Se(e,o=[]){return{key:e,version:1,storage:Ua.default,blacklist:o}}var z,Ba,Ua,vp,yo,kp,Ap,Ip,Lp,Ep,Np,Cp,Pp,Dp,Wa,Rp,bt,Mp,Op,Fp,Bp,$a=b(()=>{"use strict";t();z=require("redux-persist"),Ba=T(require("redux-persist/lib/storage")),Ua=T(require("redux-persist/lib/storage/session"));gn();L();qo();Xo();Yo();Qo();on();tn();an();mn();yt();vp={persistReducersStorageType:"localStorage"},yo={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},kp=Se(yo["account"]),Ap=Se(yo["loginInfo"]),Ip=Se(yo["modals"]),Lp=Se(yo["networkConfig"]),Ep={2:e=>v(f({},e),{networkConfig:mt})};Np=Se("sdk-dapp-transactionsInfo"),Cp=Se("sdk-dapp-transactions",["transactionsToSign"]),Pp=Se("sdk-dapp-toasts"),Dp=Se("sdk-dapp-signedMessageInfo"),Wa={key:"sdk-dapp-store",version:2,storage:Ba.default,whitelist:Object.keys(yo),migrate:(0,z.createMigrate)(Ep,{debug:!1})},Rp=v(f({},Wa),{whitelist:[]}),bt={["toasts"]:(0,z.persistReducer)(Pp,nn),["transactions"]:(0,z.persistReducer)(Cp,pn),["transactionsInfo"]:(0,z.persistReducer)(Np,rn),["signedMessageInfo"]:(0,z.persistReducer)(Dp,en)},Mp=v(f({},bt),{["account"]:(0,z.persistReducer)(kp,zo),["loginInfo"]:(0,z.persistReducer)(Ap,Ko),["modals"]:(0,z.persistReducer)(Ip,Zo),["networkConfig"]:(0,z.persistReducer)(Lp,Jo)}),Op=vp.persistReducersStorageType==="localStorage",Fp=Op?(0,z.persistReducer)(Wa,Re(bt)):(0,z.persistReducer)(Rp,Re(Mp)),Bp=Fp});var Va={};Ae(Va,{default:()=>Up});var Up,Ha=b(()=>{"use strict";t();gn();Up=Re()});var ja={};Ae(ja,{default:()=>Gp});var Z,Wp,Gp,za=b(()=>{"use strict";t();Z=require("redux-persist"),Wp=[Z.FLUSH,Z.REHYDRATE,Z.PAUSE,Z.PERSIST,Z.PURGE,Z.REGISTER],Gp=Wp});var Xa={};Ae(Xa,{default:()=>Ka});function Ka(e){return(0,qa.persistStore)(e)}var qa,Za=b(()=>{"use strict";t();qa=require("redux-persist")});var rc=H((hL,tc)=>{t();var Ld=typeof a=="object"&&a&&a.Object===Object&&a;tc.exports=Ld});var ic=H((xL,ac)=>{t();var Ed=rc(),Nd=typeof self=="object"&&self&&self.Object===Object&&self,Cd=Ed||Nd||Function("return this")();ac.exports=Cd});var Wt=H((yL,cc)=>{t();var Pd=ic(),Dd=Pd.Symbol;cc.exports=Dd});var dc=H((bL,mc)=>{t();var sc=Wt(),pc=Object.prototype,Rd=pc.hasOwnProperty,Md=pc.toString,Io=sc?sc.toStringTag:void 0;function Od(e){var o=Rd.call(e,Io),n=e[Io];try{e[Io]=void 0;var r=!0}catch(p){}var s=Md.call(e);return r&&(o?e[Io]=n:delete e[Io]),s}mc.exports=Od});var _c=H((SL,lc)=>{t();var Fd=Object.prototype,Bd=Fd.toString;function Ud(e){return Bd.call(e)}lc.exports=Ud});var Tc=H((wL,fc)=>{t();var gc=Wt(),Wd=dc(),Gd=_c(),$d="[object Null]",Vd="[object Undefined]",uc=gc?gc.toStringTag:void 0;function Hd(e){return e==null?e===void 0?Vd:$d:uc&&uc in Object(e)?Wd(e):Gd(e)}fc.exports=Hd});var xc=H((vL,hc)=>{t();var jd=Array.isArray;hc.exports=jd});var bc=H((kL,yc)=>{t();function zd(e){return e!=null&&typeof e=="object"}yc.exports=zd});var wc=H((AL,Sc)=>{t();var qd=Tc(),Kd=xc(),Xd=bc(),Zd="[object String]";function Yd(e){return typeof e=="string"||!Kd(e)&&Xd(e)&&qd(e)==Zd}Sc.exports=Yd});var vl={};Ae(vl,{TransactionInfoMethod:()=>wl});module.exports=lo(vl);t();var so=T(require("react"));U();t();U();var zn=e=>{var r,s,p,m,_,l;let o="Transaction";return((r=e.action)==null?void 0:r.name)&&((s=e.action)==null?void 0:s.category)&&(((p=e.action)==null?void 0:p.category)==="esdtNft"&&((m=e.action)==null?void 0:m.name)==="transfer"?o="Transaction":e.action&&(o=e.action.name),(l=(_=e.action)==null?void 0:_.arguments)!=null&&l.functionName&&(o=e.action.arguments.functionName)),o};t();var Uo=T(require("react")),Wo=T(require("classnames"));t();var Ms=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Ms));var g={"link-style":"dapp-core-component__main__link-style",linkStyle:"dapp-core-component__main__link-style","link-second-style":"dapp-core-component__main__link-second-style",linkSecondStyle:"dapp-core-component__main__link-second-style",wrapper:"dapp-core-component__main__wrapper","opacity-6":"dapp-core-component__main__opacity-6",opacity6:"dapp-core-component__main__opacity-6","qr-code-svg-container":"dapp-core-component__main__qr-code-svg-container",qrCodeSvgContainer:"dapp-core-component__main__qr-code-svg-container","pair-list":"dapp-core-component__main__pair-list",pairList:"dapp-core-component__main__pair-list","pair-loader":"dapp-core-component__main__pair-loader",pairLoader:"dapp-core-component__main__pair-loader","pair-image":"dapp-core-component__main__pair-image",pairImage:"dapp-core-component__main__pair-image","pair-remove":"dapp-core-component__main__pair-remove",pairRemove:"dapp-core-component__main__pair-remove","dapp-icon":"dapp-core-component__main__dapp-icon",dappIcon:"dapp-core-component__main__dapp-icon","icon-medium":"dapp-core-component__main__icon-medium",iconMedium:"dapp-core-component__main__icon-medium",centering:"dapp-core-component__main__centering","border-n":"dapp-core-component__main__border-n",borderN:"dapp-core-component__main__border-n","t-shadow":"dapp-core-component__main__t-shadow",tShadow:"dapp-core-component__main__t-shadow",btn:"dapp-core-component__main__btn",media:"dapp-core-component__main__media","media-body":"dapp-core-component__main__media-body",mediaBody:"dapp-core-component__main__media-body","connect-btns":"dapp-core-component__main__connect-btns",connectBtns:"dapp-core-component__main__connect-btns",info:"dapp-core-component__main__info","trust-badge":"dapp-core-component__main__trust-badge",trustBadge:"dapp-core-component__main__trust-badge",navbar:"dapp-core-component__main__navbar","dapp-name":"dapp-core-component__main__dapp-name",dappName:"dapp-core-component__main__dapp-name","action-btn":"dapp-core-component__main__action-btn",actionBtn:"dapp-core-component__main__action-btn",transactions:"dapp-core-component__main__transactions","transaction-icon":"dapp-core-component__main__transaction-icon",transactionIcon:"dapp-core-component__main__transaction-icon",table:"dapp-core-component__main__table","page-state":"dapp-core-component__main__page-state",pageState:"dapp-core-component__main__page-state","lds-ellipsis":"dapp-core-component__main__lds-ellipsis",ldsEllipsis:"dapp-core-component__main__lds-ellipsis","bg-blue":"dapp-core-component__main__bg-blue",bgBlue:"dapp-core-component__main__bg-blue","lds-ellipsis1":"dapp-core-component__main__lds-ellipsis1",ldsEllipsis1:"dapp-core-component__main__lds-ellipsis1","lds-ellipsis2":"dapp-core-component__main__lds-ellipsis2",ldsEllipsis2:"dapp-core-component__main__lds-ellipsis2","lds-ellipsis3":"dapp-core-component__main__lds-ellipsis3",ldsEllipsis3:"dapp-core-component__main__lds-ellipsis3",focus:"dapp-core-component__main__focus",disabled:"dapp-core-component__main__disabled","btn-primary":"dapp-core-component__main__btn-primary",btnPrimary:"dapp-core-component__main__btn-primary","btn-secondary":"dapp-core-component__main__btn-secondary",btnSecondary:"dapp-core-component__main__btn-secondary","btn-warning":"dapp-core-component__main__btn-warning",btnWarning:"dapp-core-component__main__btn-warning","btn-dark":"dapp-core-component__main__btn-dark",btnDark:"dapp-core-component__main__btn-dark","btn-link":"dapp-core-component__main__btn-link",btnLink:"dapp-core-component__main__btn-link","btn-light":"dapp-core-component__main__btn-light",btnLight:"dapp-core-component__main__btn-light",badge:"dapp-core-component__main__badge","badge-pill":"dapp-core-component__main__badge-pill",badgePill:"dapp-core-component__main__badge-pill","badge-primary":"dapp-core-component__main__badge-primary",badgePrimary:"dapp-core-component__main__badge-primary","badge-secondary":"dapp-core-component__main__badge-secondary",badgeSecondary:"dapp-core-component__main__badge-secondary","badge-success":"dapp-core-component__main__badge-success",badgeSuccess:"dapp-core-component__main__badge-success","badge-info":"dapp-core-component__main__badge-info",badgeInfo:"dapp-core-component__main__badge-info","badge-warning":"dapp-core-component__main__badge-warning",badgeWarning:"dapp-core-component__main__badge-warning","badge-danger":"dapp-core-component__main__badge-danger",badgeDanger:"dapp-core-component__main__badge-danger","badge-light":"dapp-core-component__main__badge-light",badgeLight:"dapp-core-component__main__badge-light","badge-dark":"dapp-core-component__main__badge-dark",badgeDark:"dapp-core-component__main__badge-dark",card:"dapp-core-component__main__card","list-group":"dapp-core-component__main__list-group",listGroup:"dapp-core-component__main__list-group","card-header":"dapp-core-component__main__card-header",cardHeader:"dapp-core-component__main__card-header","card-footer":"dapp-core-component__main__card-footer",cardFooter:"dapp-core-component__main__card-footer","card-header-item":"dapp-core-component__main__card-header-item",cardHeaderItem:"dapp-core-component__main__card-header-item",h6:"dapp-core-component__main__h6",compact:"dapp-core-component__main__compact","card-body":"dapp-core-component__main__card-body",cardBody:"dapp-core-component__main__card-body","card-title":"dapp-core-component__main__card-title",cardTitle:"dapp-core-component__main__card-title","card-subtitle":"dapp-core-component__main__card-subtitle",cardSubtitle:"dapp-core-component__main__card-subtitle","card-text":"dapp-core-component__main__card-text",cardText:"dapp-core-component__main__card-text","card-link":"dapp-core-component__main__card-link",cardLink:"dapp-core-component__main__card-link",text:"dapp-core-component__main__text","text-white":"dapp-core-component__main__text-white",textWhite:"dapp-core-component__main__text-white","text-monospace":"dapp-core-component__main__text-monospace",textMonospace:"dapp-core-component__main__text-monospace","trim-size-xl":"dapp-core-component__main__trim-size-xl",trimSizeXl:"dapp-core-component__main__trim-size-xl",trim:"dapp-core-component__main__trim","trim-only-sm":"dapp-core-component__main__trim-only-sm",trimOnlySm:"dapp-core-component__main__trim-only-sm","text-justify":"dapp-core-component__main__text-justify",textJustify:"dapp-core-component__main__text-justify","text-wrap":"dapp-core-component__main__text-wrap",textWrap:"dapp-core-component__main__text-wrap","text-nowrap":"dapp-core-component__main__text-nowrap",textNowrap:"dapp-core-component__main__text-nowrap","text-truncate":"dapp-core-component__main__text-truncate",textTruncate:"dapp-core-component__main__text-truncate","text-left":"dapp-core-component__main__text-left",textLeft:"dapp-core-component__main__text-left","text-right":"dapp-core-component__main__text-right",textRight:"dapp-core-component__main__text-right","text-center":"dapp-core-component__main__text-center",textCenter:"dapp-core-component__main__text-center","text-primary":"dapp-core-component__main__text-primary",textPrimary:"dapp-core-component__main__text-primary","text-secondary":"dapp-core-component__main__text-secondary",textSecondary:"dapp-core-component__main__text-secondary","text-primary-highlight":"dapp-core-component__main__text-primary-highlight",textPrimaryHighlight:"dapp-core-component__main__text-primary-highlight","text-muted":"dapp-core-component__main__text-muted",textMuted:"dapp-core-component__main__text-muted","text-success":"dapp-core-component__main__text-success",textSuccess:"dapp-core-component__main__text-success","text-body":"dapp-core-component__main__text-body",textBody:"dapp-core-component__main__text-body","text-black-50":"dapp-core-component__main__text-black-50",textBlack50:"dapp-core-component__main__text-black-50","text-white-50":"dapp-core-component__main__text-white-50",textWhite50:"dapp-core-component__main__text-white-50","text-lg-right":"dapp-core-component__main__text-lg-right",textLgRight:"dapp-core-component__main__text-lg-right","text-hide":"dapp-core-component__main__text-hide",textHide:"dapp-core-component__main__text-hide","text-decoration-none":"dapp-core-component__main__text-decoration-none",textDecorationNone:"dapp-core-component__main__text-decoration-none","text-break":"dapp-core-component__main__text-break",textBreak:"dapp-core-component__main__text-break","text-reset":"dapp-core-component__main__text-reset",textReset:"dapp-core-component__main__text-reset","text-lowercase":"dapp-core-component__main__text-lowercase",textLowercase:"dapp-core-component__main__text-lowercase","text-uppercase":"dapp-core-component__main__text-uppercase",textUppercase:"dapp-core-component__main__text-uppercase","text-capitalize":"dapp-core-component__main__text-capitalize",textCapitalize:"dapp-core-component__main__text-capitalize","text-danger":"dapp-core-component__main__text-danger",textDanger:"dapp-core-component__main__text-danger","text-warning":"dapp-core-component__main__text-warning",textWarning:"dapp-core-component__main__text-warning","trim-fs-sm":"dapp-core-component__main__trim-fs-sm",trimFsSm:"dapp-core-component__main__trim-fs-sm","mx-auto":"dapp-core-component__main__mx-auto",mxAuto:"dapp-core-component__main__mx-auto","m-0":"dapp-core-component__main__m-0",m0:"dapp-core-component__main__m-0","mt-0":"dapp-core-component__main__mt-0",mt0:"dapp-core-component__main__mt-0","my-0":"dapp-core-component__main__my-0",my0:"dapp-core-component__main__my-0","mr-0":"dapp-core-component__main__mr-0",mr0:"dapp-core-component__main__mr-0","mx-0":"dapp-core-component__main__mx-0",mx0:"dapp-core-component__main__mx-0","mb-0":"dapp-core-component__main__mb-0",mb0:"dapp-core-component__main__mb-0","ml-0":"dapp-core-component__main__ml-0",ml0:"dapp-core-component__main__ml-0","m-1":"dapp-core-component__main__m-1",m1:"dapp-core-component__main__m-1","mt-1":"dapp-core-component__main__mt-1",mt1:"dapp-core-component__main__mt-1","my-1":"dapp-core-component__main__my-1",my1:"dapp-core-component__main__my-1","mr-1":"dapp-core-component__main__mr-1",mr1:"dapp-core-component__main__mr-1","mx-1":"dapp-core-component__main__mx-1",mx1:"dapp-core-component__main__mx-1","mb-1":"dapp-core-component__main__mb-1",mb1:"dapp-core-component__main__mb-1","ml-1":"dapp-core-component__main__ml-1",ml1:"dapp-core-component__main__ml-1","m-2":"dapp-core-component__main__m-2",m2:"dapp-core-component__main__m-2","mt-2":"dapp-core-component__main__mt-2",mt2:"dapp-core-component__main__mt-2","my-2":"dapp-core-component__main__my-2",my2:"dapp-core-component__main__my-2","mr-2":"dapp-core-component__main__mr-2",mr2:"dapp-core-component__main__mr-2","mx-2":"dapp-core-component__main__mx-2",mx2:"dapp-core-component__main__mx-2","mb-2":"dapp-core-component__main__mb-2",mb2:"dapp-core-component__main__mb-2","ml-2":"dapp-core-component__main__ml-2",ml2:"dapp-core-component__main__ml-2","m-3":"dapp-core-component__main__m-3",m3:"dapp-core-component__main__m-3","mt-3":"dapp-core-component__main__mt-3",mt3:"dapp-core-component__main__mt-3","my-3":"dapp-core-component__main__my-3",my3:"dapp-core-component__main__my-3","mr-3":"dapp-core-component__main__mr-3",mr3:"dapp-core-component__main__mr-3","mx-3":"dapp-core-component__main__mx-3",mx3:"dapp-core-component__main__mx-3","mb-3":"dapp-core-component__main__mb-3",mb3:"dapp-core-component__main__mb-3","ml-3":"dapp-core-component__main__ml-3",ml3:"dapp-core-component__main__ml-3","m-4":"dapp-core-component__main__m-4",m4:"dapp-core-component__main__m-4","mt-4":"dapp-core-component__main__mt-4",mt4:"dapp-core-component__main__mt-4","my-4":"dapp-core-component__main__my-4",my4:"dapp-core-component__main__my-4","mr-4":"dapp-core-component__main__mr-4",mr4:"dapp-core-component__main__mr-4","mx-4":"dapp-core-component__main__mx-4",mx4:"dapp-core-component__main__mx-4","mb-4":"dapp-core-component__main__mb-4",mb4:"dapp-core-component__main__mb-4","ml-4":"dapp-core-component__main__ml-4",ml4:"dapp-core-component__main__ml-4","m-5":"dapp-core-component__main__m-5",m5:"dapp-core-component__main__m-5","mt-5":"dapp-core-component__main__mt-5",mt5:"dapp-core-component__main__mt-5","my-5":"dapp-core-component__main__my-5",my5:"dapp-core-component__main__my-5","mr-5":"dapp-core-component__main__mr-5",mr5:"dapp-core-component__main__mr-5","mx-5":"dapp-core-component__main__mx-5",mx5:"dapp-core-component__main__mx-5","mb-5":"dapp-core-component__main__mb-5",mb5:"dapp-core-component__main__mb-5","ml-5":"dapp-core-component__main__ml-5",ml5:"dapp-core-component__main__ml-5","p-0":"dapp-core-component__main__p-0",p0:"dapp-core-component__main__p-0","pt-0":"dapp-core-component__main__pt-0",pt0:"dapp-core-component__main__pt-0","py-0":"dapp-core-component__main__py-0",py0:"dapp-core-component__main__py-0","pr-0":"dapp-core-component__main__pr-0",pr0:"dapp-core-component__main__pr-0","px-0":"dapp-core-component__main__px-0",px0:"dapp-core-component__main__px-0","pb-0":"dapp-core-component__main__pb-0",pb0:"dapp-core-component__main__pb-0","pl-0":"dapp-core-component__main__pl-0",pl0:"dapp-core-component__main__pl-0","p-1":"dapp-core-component__main__p-1",p1:"dapp-core-component__main__p-1","pt-1":"dapp-core-component__main__pt-1",pt1:"dapp-core-component__main__pt-1","py-1":"dapp-core-component__main__py-1",py1:"dapp-core-component__main__py-1","pr-1":"dapp-core-component__main__pr-1",pr1:"dapp-core-component__main__pr-1","px-1":"dapp-core-component__main__px-1",px1:"dapp-core-component__main__px-1","pb-1":"dapp-core-component__main__pb-1",pb1:"dapp-core-component__main__pb-1","pl-1":"dapp-core-component__main__pl-1",pl1:"dapp-core-component__main__pl-1","p-2":"dapp-core-component__main__p-2",p2:"dapp-core-component__main__p-2","pt-2":"dapp-core-component__main__pt-2",pt2:"dapp-core-component__main__pt-2","py-2":"dapp-core-component__main__py-2",py2:"dapp-core-component__main__py-2","pr-2":"dapp-core-component__main__pr-2",pr2:"dapp-core-component__main__pr-2","px-2":"dapp-core-component__main__px-2",px2:"dapp-core-component__main__px-2","pb-2":"dapp-core-component__main__pb-2",pb2:"dapp-core-component__main__pb-2","pl-2":"dapp-core-component__main__pl-2",pl2:"dapp-core-component__main__pl-2","p-3":"dapp-core-component__main__p-3",p3:"dapp-core-component__main__p-3","pt-3":"dapp-core-component__main__pt-3",pt3:"dapp-core-component__main__pt-3","py-3":"dapp-core-component__main__py-3",py3:"dapp-core-component__main__py-3","pr-3":"dapp-core-component__main__pr-3",pr3:"dapp-core-component__main__pr-3","px-3":"dapp-core-component__main__px-3",px3:"dapp-core-component__main__px-3","pb-3":"dapp-core-component__main__pb-3",pb3:"dapp-core-component__main__pb-3","pl-3":"dapp-core-component__main__pl-3",pl3:"dapp-core-component__main__pl-3","p-4":"dapp-core-component__main__p-4",p4:"dapp-core-component__main__p-4","pt-4":"dapp-core-component__main__pt-4",pt4:"dapp-core-component__main__pt-4","py-4":"dapp-core-component__main__py-4",py4:"dapp-core-component__main__py-4","pr-4":"dapp-core-component__main__pr-4",pr4:"dapp-core-component__main__pr-4","px-4":"dapp-core-component__main__px-4",px4:"dapp-core-component__main__px-4","pb-4":"dapp-core-component__main__pb-4",pb4:"dapp-core-component__main__pb-4","pl-4":"dapp-core-component__main__pl-4",pl4:"dapp-core-component__main__pl-4","p-5":"dapp-core-component__main__p-5",p5:"dapp-core-component__main__p-5","pt-5":"dapp-core-component__main__pt-5",pt5:"dapp-core-component__main__pt-5","py-5":"dapp-core-component__main__py-5",py5:"dapp-core-component__main__py-5","pr-5":"dapp-core-component__main__pr-5",pr5:"dapp-core-component__main__pr-5","px-5":"dapp-core-component__main__px-5",px5:"dapp-core-component__main__px-5","pb-5":"dapp-core-component__main__pb-5",pb5:"dapp-core-component__main__pb-5","pl-5":"dapp-core-component__main__pl-5",pl5:"dapp-core-component__main__pl-5","m-n1":"dapp-core-component__main__m-n1",mN1:"dapp-core-component__main__m-n1","mt-n1":"dapp-core-component__main__mt-n1",mtN1:"dapp-core-component__main__mt-n1","my-n1":"dapp-core-component__main__my-n1",myN1:"dapp-core-component__main__my-n1","mr-n1":"dapp-core-component__main__mr-n1",mrN1:"dapp-core-component__main__mr-n1","mx-n1":"dapp-core-component__main__mx-n1",mxN1:"dapp-core-component__main__mx-n1","mb-n1":"dapp-core-component__main__mb-n1",mbN1:"dapp-core-component__main__mb-n1","ml-n1":"dapp-core-component__main__ml-n1",mlN1:"dapp-core-component__main__ml-n1","m-n2":"dapp-core-component__main__m-n2",mN2:"dapp-core-component__main__m-n2","mt-n2":"dapp-core-component__main__mt-n2",mtN2:"dapp-core-component__main__mt-n2","my-n2":"dapp-core-component__main__my-n2",myN2:"dapp-core-component__main__my-n2","mr-n2":"dapp-core-component__main__mr-n2",mrN2:"dapp-core-component__main__mr-n2","mx-n2":"dapp-core-component__main__mx-n2",mxN2:"dapp-core-component__main__mx-n2","mb-n2":"dapp-core-component__main__mb-n2",mbN2:"dapp-core-component__main__mb-n2","ml-n2":"dapp-core-component__main__ml-n2",mlN2:"dapp-core-component__main__ml-n2","m-n3":"dapp-core-component__main__m-n3",mN3:"dapp-core-component__main__m-n3","mt-n3":"dapp-core-component__main__mt-n3",mtN3:"dapp-core-component__main__mt-n3","my-n3":"dapp-core-component__main__my-n3",myN3:"dapp-core-component__main__my-n3","mr-n3":"dapp-core-component__main__mr-n3",mrN3:"dapp-core-component__main__mr-n3","mx-n3":"dapp-core-component__main__mx-n3",mxN3:"dapp-core-component__main__mx-n3","mb-n3":"dapp-core-component__main__mb-n3",mbN3:"dapp-core-component__main__mb-n3","ml-n3":"dapp-core-component__main__ml-n3",mlN3:"dapp-core-component__main__ml-n3","m-n4":"dapp-core-component__main__m-n4",mN4:"dapp-core-component__main__m-n4","mt-n4":"dapp-core-component__main__mt-n4",mtN4:"dapp-core-component__main__mt-n4","my-n4":"dapp-core-component__main__my-n4",myN4:"dapp-core-component__main__my-n4","mr-n4":"dapp-core-component__main__mr-n4",mrN4:"dapp-core-component__main__mr-n4","mx-n4":"dapp-core-component__main__mx-n4",mxN4:"dapp-core-component__main__mx-n4","mb-n4":"dapp-core-component__main__mb-n4",mbN4:"dapp-core-component__main__mb-n4","ml-n4":"dapp-core-component__main__ml-n4",mlN4:"dapp-core-component__main__ml-n4","m-n5":"dapp-core-component__main__m-n5",mN5:"dapp-core-component__main__m-n5","mt-n5":"dapp-core-component__main__mt-n5",mtN5:"dapp-core-component__main__mt-n5","my-n5":"dapp-core-component__main__my-n5",myN5:"dapp-core-component__main__my-n5","mr-n5":"dapp-core-component__main__mr-n5",mrN5:"dapp-core-component__main__mr-n5","mx-n5":"dapp-core-component__main__mx-n5",mxN5:"dapp-core-component__main__mx-n5","mb-n5":"dapp-core-component__main__mb-n5",mbN5:"dapp-core-component__main__mb-n5","ml-n5":"dapp-core-component__main__ml-n5",mlN5:"dapp-core-component__main__ml-n5","m-auto":"dapp-core-component__main__m-auto",mAuto:"dapp-core-component__main__m-auto","mt-auto":"dapp-core-component__main__mt-auto",mtAuto:"dapp-core-component__main__mt-auto","my-auto":"dapp-core-component__main__my-auto",myAuto:"dapp-core-component__main__my-auto","mr-auto":"dapp-core-component__main__mr-auto",mrAuto:"dapp-core-component__main__mr-auto","mb-auto":"dapp-core-component__main__mb-auto",mbAuto:"dapp-core-component__main__mb-auto","ml-auto":"dapp-core-component__main__ml-auto",mlAuto:"dapp-core-component__main__ml-auto","mr-lg-1":"dapp-core-component__main__mr-lg-1",mrLg1:"dapp-core-component__main__mr-lg-1","mr-lg-4":"dapp-core-component__main__mr-lg-4",mrLg4:"dapp-core-component__main__mr-lg-4","mx-lg-4":"dapp-core-component__main__mx-lg-4",mxLg4:"dapp-core-component__main__mx-lg-4","mb-xl-2":"dapp-core-component__main__mb-xl-2",mbXl2:"dapp-core-component__main__mb-xl-2","pr-sm-2":"dapp-core-component__main__pr-sm-2",prSm2:"dapp-core-component__main__pr-sm-2","px-sm-2":"dapp-core-component__main__px-sm-2",pxSm2:"dapp-core-component__main__px-sm-2","pr-xl-0":"dapp-core-component__main__pr-xl-0",prXl0:"dapp-core-component__main__pr-xl-0","px-xl-0":"dapp-core-component__main__px-xl-0",pxXl0:"dapp-core-component__main__px-xl-0",col:"dapp-core-component__main__col",row:"dapp-core-component__main__row","col-xl":"dapp-core-component__main__col-xl",colXl:"dapp-core-component__main__col-xl","col-xl-auto":"dapp-core-component__main__col-xl-auto",colXlAuto:"dapp-core-component__main__col-xl-auto","col-xl-12":"dapp-core-component__main__col-xl-12",colXl12:"dapp-core-component__main__col-xl-12","col-xl-11":"dapp-core-component__main__col-xl-11",colXl11:"dapp-core-component__main__col-xl-11","col-xl-10":"dapp-core-component__main__col-xl-10",colXl10:"dapp-core-component__main__col-xl-10","col-xl-9":"dapp-core-component__main__col-xl-9",colXl9:"dapp-core-component__main__col-xl-9","col-xl-8":"dapp-core-component__main__col-xl-8",colXl8:"dapp-core-component__main__col-xl-8","col-xl-7":"dapp-core-component__main__col-xl-7",colXl7:"dapp-core-component__main__col-xl-7","col-xl-6":"dapp-core-component__main__col-xl-6",colXl6:"dapp-core-component__main__col-xl-6","col-xl-5":"dapp-core-component__main__col-xl-5",colXl5:"dapp-core-component__main__col-xl-5","col-xl-4":"dapp-core-component__main__col-xl-4",colXl4:"dapp-core-component__main__col-xl-4","col-xl-3":"dapp-core-component__main__col-xl-3",colXl3:"dapp-core-component__main__col-xl-3","col-xl-2":"dapp-core-component__main__col-xl-2",colXl2:"dapp-core-component__main__col-xl-2","col-xl-1":"dapp-core-component__main__col-xl-1",colXl1:"dapp-core-component__main__col-xl-1","col-lg":"dapp-core-component__main__col-lg",colLg:"dapp-core-component__main__col-lg","col-lg-auto":"dapp-core-component__main__col-lg-auto",colLgAuto:"dapp-core-component__main__col-lg-auto","col-lg-12":"dapp-core-component__main__col-lg-12",colLg12:"dapp-core-component__main__col-lg-12","col-lg-11":"dapp-core-component__main__col-lg-11",colLg11:"dapp-core-component__main__col-lg-11","col-lg-10":"dapp-core-component__main__col-lg-10",colLg10:"dapp-core-component__main__col-lg-10","col-lg-9":"dapp-core-component__main__col-lg-9",colLg9:"dapp-core-component__main__col-lg-9","col-lg-8":"dapp-core-component__main__col-lg-8",colLg8:"dapp-core-component__main__col-lg-8","col-lg-7":"dapp-core-component__main__col-lg-7",colLg7:"dapp-core-component__main__col-lg-7","col-lg-6":"dapp-core-component__main__col-lg-6",colLg6:"dapp-core-component__main__col-lg-6","col-lg-5":"dapp-core-component__main__col-lg-5",colLg5:"dapp-core-component__main__col-lg-5","col-lg-4":"dapp-core-component__main__col-lg-4",colLg4:"dapp-core-component__main__col-lg-4","col-lg-3":"dapp-core-component__main__col-lg-3",colLg3:"dapp-core-component__main__col-lg-3","col-lg-2":"dapp-core-component__main__col-lg-2",colLg2:"dapp-core-component__main__col-lg-2","col-lg-1":"dapp-core-component__main__col-lg-1",colLg1:"dapp-core-component__main__col-lg-1","col-md":"dapp-core-component__main__col-md",colMd:"dapp-core-component__main__col-md","col-md-auto":"dapp-core-component__main__col-md-auto",colMdAuto:"dapp-core-component__main__col-md-auto","col-md-12":"dapp-core-component__main__col-md-12",colMd12:"dapp-core-component__main__col-md-12","col-md-11":"dapp-core-component__main__col-md-11",colMd11:"dapp-core-component__main__col-md-11","col-md-10":"dapp-core-component__main__col-md-10",colMd10:"dapp-core-component__main__col-md-10","col-md-9":"dapp-core-component__main__col-md-9",colMd9:"dapp-core-component__main__col-md-9","col-md-8":"dapp-core-component__main__col-md-8",colMd8:"dapp-core-component__main__col-md-8","col-md-7":"dapp-core-component__main__col-md-7",colMd7:"dapp-core-component__main__col-md-7","col-md-6":"dapp-core-component__main__col-md-6",colMd6:"dapp-core-component__main__col-md-6","col-md-5":"dapp-core-component__main__col-md-5",colMd5:"dapp-core-component__main__col-md-5","col-md-4":"dapp-core-component__main__col-md-4",colMd4:"dapp-core-component__main__col-md-4","col-md-3":"dapp-core-component__main__col-md-3",colMd3:"dapp-core-component__main__col-md-3","col-md-2":"dapp-core-component__main__col-md-2",colMd2:"dapp-core-component__main__col-md-2","col-md-1":"dapp-core-component__main__col-md-1",colMd1:"dapp-core-component__main__col-md-1","col-sm":"dapp-core-component__main__col-sm",colSm:"dapp-core-component__main__col-sm","col-sm-auto":"dapp-core-component__main__col-sm-auto",colSmAuto:"dapp-core-component__main__col-sm-auto","col-sm-12":"dapp-core-component__main__col-sm-12",colSm12:"dapp-core-component__main__col-sm-12","col-sm-11":"dapp-core-component__main__col-sm-11",colSm11:"dapp-core-component__main__col-sm-11","col-sm-10":"dapp-core-component__main__col-sm-10",colSm10:"dapp-core-component__main__col-sm-10","col-sm-9":"dapp-core-component__main__col-sm-9",colSm9:"dapp-core-component__main__col-sm-9","col-sm-8":"dapp-core-component__main__col-sm-8",colSm8:"dapp-core-component__main__col-sm-8","col-sm-7":"dapp-core-component__main__col-sm-7",colSm7:"dapp-core-component__main__col-sm-7","col-sm-6":"dapp-core-component__main__col-sm-6",colSm6:"dapp-core-component__main__col-sm-6","col-sm-5":"dapp-core-component__main__col-sm-5",colSm5:"dapp-core-component__main__col-sm-5","col-sm-4":"dapp-core-component__main__col-sm-4",colSm4:"dapp-core-component__main__col-sm-4","col-sm-3":"dapp-core-component__main__col-sm-3",colSm3:"dapp-core-component__main__col-sm-3","col-sm-2":"dapp-core-component__main__col-sm-2",colSm2:"dapp-core-component__main__col-sm-2","col-sm-1":"dapp-core-component__main__col-sm-1",colSm1:"dapp-core-component__main__col-sm-1","col-auto":"dapp-core-component__main__col-auto",colAuto:"dapp-core-component__main__col-auto","col-12":"dapp-core-component__main__col-12",col12:"dapp-core-component__main__col-12","col-11":"dapp-core-component__main__col-11",col11:"dapp-core-component__main__col-11","col-10":"dapp-core-component__main__col-10",col10:"dapp-core-component__main__col-10","col-9":"dapp-core-component__main__col-9",col9:"dapp-core-component__main__col-9","col-8":"dapp-core-component__main__col-8",col8:"dapp-core-component__main__col-8","col-7":"dapp-core-component__main__col-7",col7:"dapp-core-component__main__col-7","col-6":"dapp-core-component__main__col-6",col6:"dapp-core-component__main__col-6","col-5":"dapp-core-component__main__col-5",col5:"dapp-core-component__main__col-5","col-4":"dapp-core-component__main__col-4",col4:"dapp-core-component__main__col-4","col-3":"dapp-core-component__main__col-3",col3:"dapp-core-component__main__col-3","col-2":"dapp-core-component__main__col-2",col2:"dapp-core-component__main__col-2","col-1":"dapp-core-component__main__col-1",col1:"dapp-core-component__main__col-1","flex-row":"dapp-core-component__main__flex-row",flexRow:"dapp-core-component__main__flex-row","flex-column":"dapp-core-component__main__flex-column",flexColumn:"dapp-core-component__main__flex-column","flex-row-reverse":"dapp-core-component__main__flex-row-reverse",flexRowReverse:"dapp-core-component__main__flex-row-reverse","flex-column-reverse":"dapp-core-component__main__flex-column-reverse",flexColumnReverse:"dapp-core-component__main__flex-column-reverse","flex-wrap":"dapp-core-component__main__flex-wrap",flexWrap:"dapp-core-component__main__flex-wrap","flex-nowrap":"dapp-core-component__main__flex-nowrap",flexNowrap:"dapp-core-component__main__flex-nowrap","flex-wrap-reverse":"dapp-core-component__main__flex-wrap-reverse",flexWrapReverse:"dapp-core-component__main__flex-wrap-reverse","flex-fill":"dapp-core-component__main__flex-fill",flexFill:"dapp-core-component__main__flex-fill","flex-grow-0":"dapp-core-component__main__flex-grow-0",flexGrow0:"dapp-core-component__main__flex-grow-0","flex-grow-1":"dapp-core-component__main__flex-grow-1",flexGrow1:"dapp-core-component__main__flex-grow-1","flex-shrink-0":"dapp-core-component__main__flex-shrink-0",flexShrink0:"dapp-core-component__main__flex-shrink-0","flex-shrink-1":"dapp-core-component__main__flex-shrink-1",flexShrink1:"dapp-core-component__main__flex-shrink-1","justify-content-start":"dapp-core-component__main__justify-content-start",justifyContentStart:"dapp-core-component__main__justify-content-start","justify-content-end":"dapp-core-component__main__justify-content-end",justifyContentEnd:"dapp-core-component__main__justify-content-end","justify-content-center":"dapp-core-component__main__justify-content-center",justifyContentCenter:"dapp-core-component__main__justify-content-center","justify-content-between":"dapp-core-component__main__justify-content-between",justifyContentBetween:"dapp-core-component__main__justify-content-between","justify-content-around":"dapp-core-component__main__justify-content-around",justifyContentAround:"dapp-core-component__main__justify-content-around","align-items-start":"dapp-core-component__main__align-items-start",alignItemsStart:"dapp-core-component__main__align-items-start","align-items-end":"dapp-core-component__main__align-items-end",alignItemsEnd:"dapp-core-component__main__align-items-end","align-items-center":"dapp-core-component__main__align-items-center",alignItemsCenter:"dapp-core-component__main__align-items-center","align-items-baseline":"dapp-core-component__main__align-items-baseline",alignItemsBaseline:"dapp-core-component__main__align-items-baseline","align-items-stretch":"dapp-core-component__main__align-items-stretch",alignItemsStretch:"dapp-core-component__main__align-items-stretch","align-content-start":"dapp-core-component__main__align-content-start",alignContentStart:"dapp-core-component__main__align-content-start","align-content-end":"dapp-core-component__main__align-content-end",alignContentEnd:"dapp-core-component__main__align-content-end","align-content-center":"dapp-core-component__main__align-content-center",alignContentCenter:"dapp-core-component__main__align-content-center","align-content-between":"dapp-core-component__main__align-content-between",alignContentBetween:"dapp-core-component__main__align-content-between","align-content-around":"dapp-core-component__main__align-content-around",alignContentAround:"dapp-core-component__main__align-content-around","align-content-stretch":"dapp-core-component__main__align-content-stretch",alignContentStretch:"dapp-core-component__main__align-content-stretch","align-self-auto":"dapp-core-component__main__align-self-auto",alignSelfAuto:"dapp-core-component__main__align-self-auto","align-self-start":"dapp-core-component__main__align-self-start",alignSelfStart:"dapp-core-component__main__align-self-start","align-self-end":"dapp-core-component__main__align-self-end",alignSelfEnd:"dapp-core-component__main__align-self-end","align-self-center":"dapp-core-component__main__align-self-center",alignSelfCenter:"dapp-core-component__main__align-self-center","align-self-baseline":"dapp-core-component__main__align-self-baseline",alignSelfBaseline:"dapp-core-component__main__align-self-baseline","align-self-stretch":"dapp-core-component__main__align-self-stretch",alignSelfStretch:"dapp-core-component__main__align-self-stretch","flex-sm-row":"dapp-core-component__main__flex-sm-row",flexSmRow:"dapp-core-component__main__flex-sm-row","flex-lg-row":"dapp-core-component__main__flex-lg-row",flexLgRow:"dapp-core-component__main__flex-lg-row","flex-lg-wrap":"dapp-core-component__main__flex-lg-wrap",flexLgWrap:"dapp-core-component__main__flex-lg-wrap","form-check":"dapp-core-component__main__form-check",formCheck:"dapp-core-component__main__form-check","form-check-label":"dapp-core-component__main__form-check-label",formCheckLabel:"dapp-core-component__main__form-check-label","form-check-input":"dapp-core-component__main__form-check-input",formCheckInput:"dapp-core-component__main__form-check-input","form-group":"dapp-core-component__main__form-group",formGroup:"dapp-core-component__main__form-group","form-control":"dapp-core-component__main__form-control",formControl:"dapp-core-component__main__form-control","form-label":"dapp-core-component__main__form-label",formLabel:"dapp-core-component__main__form-label","form-text":"dapp-core-component__main__form-text",formText:"dapp-core-component__main__form-text",border:"dapp-core-component__main__border","border-top":"dapp-core-component__main__border-top",borderTop:"dapp-core-component__main__border-top","border-right":"dapp-core-component__main__border-right",borderRight:"dapp-core-component__main__border-right","border-bottom":"dapp-core-component__main__border-bottom",borderBottom:"dapp-core-component__main__border-bottom","border-left":"dapp-core-component__main__border-left",borderLeft:"dapp-core-component__main__border-left","border-0":"dapp-core-component__main__border-0",border0:"dapp-core-component__main__border-0","border-top-0":"dapp-core-component__main__border-top-0",borderTop0:"dapp-core-component__main__border-top-0","border-right-0":"dapp-core-component__main__border-right-0",borderRight0:"dapp-core-component__main__border-right-0","border-bottom-0":"dapp-core-component__main__border-bottom-0",borderBottom0:"dapp-core-component__main__border-bottom-0","border-left-0":"dapp-core-component__main__border-left-0",borderLeft0:"dapp-core-component__main__border-left-0","rounded-sm":"dapp-core-component__main__rounded-sm",roundedSm:"dapp-core-component__main__rounded-sm",rounded:"dapp-core-component__main__rounded","rounded-top":"dapp-core-component__main__rounded-top",roundedTop:"dapp-core-component__main__rounded-top","rounded-right":"dapp-core-component__main__rounded-right",roundedRight:"dapp-core-component__main__rounded-right","rounded-bottom":"dapp-core-component__main__rounded-bottom",roundedBottom:"dapp-core-component__main__rounded-bottom","rounded-left":"dapp-core-component__main__rounded-left",roundedLeft:"dapp-core-component__main__rounded-left","rounded-lg":"dapp-core-component__main__rounded-lg",roundedLg:"dapp-core-component__main__rounded-lg","rounded-circle":"dapp-core-component__main__rounded-circle",roundedCircle:"dapp-core-component__main__rounded-circle","rounded-pill":"dapp-core-component__main__rounded-pill",roundedPill:"dapp-core-component__main__rounded-pill","rounded-0":"dapp-core-component__main__rounded-0",rounded0:"dapp-core-component__main__rounded-0",h1:"dapp-core-component__main__h1",h2:"dapp-core-component__main__h2",h3:"dapp-core-component__main__h3",h4:"dapp-core-component__main__h4",h5:"dapp-core-component__main__h5",lead:"dapp-core-component__main__lead","bg-primary":"dapp-core-component__main__bg-primary",bgPrimary:"dapp-core-component__main__bg-primary","bg-warning":"dapp-core-component__main__bg-warning",bgWarning:"dapp-core-component__main__bg-warning",warning:"dapp-core-component__main__warning","bg-danger":"dapp-core-component__main__bg-danger",bgDanger:"dapp-core-component__main__bg-danger",danger:"dapp-core-component__main__danger","bg-success":"dapp-core-component__main__bg-success",bgSuccess:"dapp-core-component__main__bg-success",success:"dapp-core-component__main__success","position-relative":"dapp-core-component__main__position-relative",positionRelative:"dapp-core-component__main__position-relative","position-absolute":"dapp-core-component__main__position-absolute",positionAbsolute:"dapp-core-component__main__position-absolute","w-25":"dapp-core-component__main__w-25",w25:"dapp-core-component__main__w-25","w-50":"dapp-core-component__main__w-50",w50:"dapp-core-component__main__w-50","w-75":"dapp-core-component__main__w-75",w75:"dapp-core-component__main__w-75","w-100":"dapp-core-component__main__w-100",w100:"dapp-core-component__main__w-100","w-auto":"dapp-core-component__main__w-auto",wAuto:"dapp-core-component__main__w-auto","h-25":"dapp-core-component__main__h-25",h25:"dapp-core-component__main__h-25","h-50":"dapp-core-component__main__h-50",h50:"dapp-core-component__main__h-50","h-75":"dapp-core-component__main__h-75",h75:"dapp-core-component__main__h-75","h-100":"dapp-core-component__main__h-100",h100:"dapp-core-component__main__h-100","h-auto":"dapp-core-component__main__h-auto",hAuto:"dapp-core-component__main__h-auto","mw-100":"dapp-core-component__main__mw-100",mw100:"dapp-core-component__main__mw-100","mh-100":"dapp-core-component__main__mh-100",mh100:"dapp-core-component__main__mh-100","min-vw-100":"dapp-core-component__main__min-vw-100",minVw100:"dapp-core-component__main__min-vw-100","min-vh-100":"dapp-core-component__main__min-vh-100",minVh100:"dapp-core-component__main__min-vh-100","vw-100":"dapp-core-component__main__vw-100",vw100:"dapp-core-component__main__vw-100","vh-100":"dapp-core-component__main__vh-100",vh100:"dapp-core-component__main__vh-100","w-md-auto":"dapp-core-component__main__w-md-auto",wMdAuto:"dapp-core-component__main__w-md-auto","locked-icon":"dapp-core-component__main__locked-icon",lockedIcon:"dapp-core-component__main__locked-icon","shadow-sm":"dapp-core-component__main__shadow-sm",shadowSm:"dapp-core-component__main__shadow-sm",shadow:"dapp-core-component__main__shadow","shadow-lg":"dapp-core-component__main__shadow-lg",shadowLg:"dapp-core-component__main__shadow-lg","shadow-none":"dapp-core-component__main__shadow-none",shadowNone:"dapp-core-component__main__shadow-none","d-none":"dapp-core-component__main__d-none",dNone:"dapp-core-component__main__d-none","d-inline":"dapp-core-component__main__d-inline",dInline:"dapp-core-component__main__d-inline","d-inline-block":"dapp-core-component__main__d-inline-block",dInlineBlock:"dapp-core-component__main__d-inline-block","d-block":"dapp-core-component__main__d-block",dBlock:"dapp-core-component__main__d-block","d-table":"dapp-core-component__main__d-table",dTable:"dapp-core-component__main__d-table","d-table-row":"dapp-core-component__main__d-table-row",dTableRow:"dapp-core-component__main__d-table-row","d-table-cell":"dapp-core-component__main__d-table-cell",dTableCell:"dapp-core-component__main__d-table-cell","d-flex":"dapp-core-component__main__d-flex",dFlex:"dapp-core-component__main__d-flex","d-inline-flex":"dapp-core-component__main__d-inline-flex",dInlineFlex:"dapp-core-component__main__d-inline-flex","d-sm-flex":"dapp-core-component__main__d-sm-flex",dSmFlex:"dapp-core-component__main__d-sm-flex","table-sm":"dapp-core-component__main__table-sm",tableSm:"dapp-core-component__main__table-sm","table-bordered":"dapp-core-component__main__table-bordered",tableBordered:"dapp-core-component__main__table-bordered","table-borderless":"dapp-core-component__main__table-borderless",tableBorderless:"dapp-core-component__main__table-borderless","table-striped":"dapp-core-component__main__table-striped",tableStriped:"dapp-core-component__main__table-striped","thead-dark":"dapp-core-component__main__thead-dark",theadDark:"dapp-core-component__main__thead-dark","thead-light":"dapp-core-component__main__thead-light",theadLight:"dapp-core-component__main__thead-light","table-dark":"dapp-core-component__main__table-dark",tableDark:"dapp-core-component__main__table-dark","table-hover":"dapp-core-component__main__table-hover",tableHover:"dapp-core-component__main__table-hover","table-responsive":"dapp-core-component__main__table-responsive",tableResponsive:"dapp-core-component__main__table-responsive","transaction-hash":"dapp-core-component__main__transaction-hash",transactionHash:"dapp-core-component__main__transaction-hash","font-weight-light":"dapp-core-component__main__font-weight-light",fontWeightLight:"dapp-core-component__main__font-weight-light","font-weight-lighter":"dapp-core-component__main__font-weight-lighter",fontWeightLighter:"dapp-core-component__main__font-weight-lighter","font-weight-normal":"dapp-core-component__main__font-weight-normal",fontWeightNormal:"dapp-core-component__main__font-weight-normal","font-weight-medium":"dapp-core-component__main__font-weight-medium",fontWeightMedium:"dapp-core-component__main__font-weight-medium","font-weight-bold":"dapp-core-component__main__font-weight-bold",fontWeightBold:"dapp-core-component__main__font-weight-bold","font-weight-bolder":"dapp-core-component__main__font-weight-bolder",fontWeightBolder:"dapp-core-component__main__font-weight-bolder","font-italic":"dapp-core-component__main__font-italic",fontItalic:"dapp-core-component__main__font-italic","icon-state":"dapp-core-component__main__icon-state",iconState:"dapp-core-component__main__icon-state",half:"dapp-core-component__main__half","side-icon":"dapp-core-component__main__side-icon",sideIcon:"dapp-core-component__main__side-icon","toast-messages":"dapp-core-component__main__toast-messages",toastMessages:"dapp-core-component__main__toast-messages",toast:"dapp-core-component__main__toast",clickable:"dapp-core-component__main__clickable","toast-visible":"dapp-core-component__main__toast-visible",toastVisible:"dapp-core-component__main__toast-visible",progress:"dapp-core-component__main__progress","progress-bar":"dapp-core-component__main__progress-bar",progressBar:"dapp-core-component__main__progress-bar","content-height":"dapp-core-component__main__content-height",contentHeight:"dapp-core-component__main__content-height","tx-description":"dapp-core-component__main__tx-description",txDescription:"dapp-core-component__main__tx-description",close:"dapp-core-component__main__close","data-field":"dapp-core-component__main__data-field",dataField:"dapp-core-component__main__data-field",textarea:"dapp-core-component__main__textarea",highlighted:"dapp-core-component__main__highlighted","sc-call":"dapp-core-component__main__sc-call",scCall:"dapp-core-component__main__sc-call","small-font":"dapp-core-component__main__small-font",smallFont:"dapp-core-component__main__small-font","progress-steps":"dapp-core-component__main__progress-steps",progressSteps:"dapp-core-component__main__progress-steps",steps:"dapp-core-component__main__steps",dot:"dapp-core-component__main__dot","side-link":"dapp-core-component__main__side-link",sideLink:"dapp-core-component__main__side-link"};t();var Os=`.dapp-core-component__detailItem-module__detail-item:last-child {
  border-bottom: 0;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Os));var br={"detail-item":"dapp-core-component__detailItem-module__detail-item",detailItem:"dapp-core-component__detailItem-module__detail-item"};var qn=({children:e,title:o,className:n="",colWidth:r="2",noBorder:s=!1})=>Uo.default.createElement("div",{className:(0,Wo.default)(g.row,br.detailItem,n,{[g.pt3]:s,[g.pb1]:s,[g.borderBottom]:!s,[g.py3]:!s})},Uo.default.createElement("div",{className:(0,Wo.default)(g.textSecondary,g.textLgRight,g.fontWeightMedium,g[`colLg${r}`])},o),Uo.default.createElement("div",{className:(0,Wo.default)(g.fontWeightMedium,g[`colLg${12-Number(r)}`])},e));t();t();var co=T(require("react"));t();U();t();U();var Sr=e=>{var o,n;switch(e.name){case"transfer":return["Transfer",{token:(o=e.arguments)==null?void 0:o.transfers},"to",{address:(n=e.arguments)==null?void 0:n.receiver}];default:return[]}};t();var Gr=T(require("bignumber.js"));A();U();var $r=e=>{var o,n,r,s,p,m,_,l,u,y,x,w;switch(e.name){case"claimLockedAssets":return["Claim locked assets"];case"enterFarm":case"enterFarmProxy":return["Enter farm with",{token:(o=e.arguments)==null?void 0:o.transfers}];case"enterFarmAndLockRewards":case"enterFarmAndLockRewardsProxy":return["Enter farm and lock rewards with",{token:(n=e.arguments)==null?void 0:n.transfers}];case"exitFarm":case"exitFarmProxy":return["Exit farm with",{token:(r=e.arguments)==null?void 0:r.transfers}];case"claimRewards":case"claimRewardsProxy":return["Claim rewards",{token:(s=e.arguments)==null?void 0:s.transfers}];case"compoundRewards":case"compoundRewardsProxy":return["Reinvest rewards",{token:(p=e.arguments)==null?void 0:p.transfers}];case"swapTokensFixedInput":case"swap":return e.description?[e.description]:[];case"swapTokensFixedOutput":return e.description?[e.description]:[];case"addLiquidity":case"addLiquidityProxy":return["Added liquidity for",{token:[(m=e.arguments)==null?void 0:m.transfers[0]]},"and",{token:[(_=e.arguments)==null?void 0:_.transfers[1]]}];case"removeLiquidity":case"removeLiquidityProxy":return["Removed liquidity with ",{token:(l=e.arguments)==null?void 0:l.transfers}];case"mergeLockedAssetTokens":let E=J;if((u=e.arguments)!=null&&u.transfers){let B=e.arguments.transfers.map(({value:G})=>G);E=Gr.default.sum.apply(null,B).toString(10)}return[`Merge ${(y=e.arguments)==null?void 0:y.transfers.length}`,{tokenNoLink:[(x=e.arguments)==null?void 0:x.transfers[0]]},"positions into a single",{tokenNoLink:[(w=e.arguments)==null?void 0:w.transfers[0]]},"position of value",{value:E}];case"wrapEgld":case"unwrapEgld":default:return e.description?[e.description]:[]}};t();U();var Vr=e=>{var o,n,r,s,p,m,_,l,u,y,x,w;switch(e.name){case"delegate":case"stake":return["Delegate",{egldValue:(o=e.arguments)==null?void 0:o.value},"to staking provider",{providerName:(n=e.arguments)==null?void 0:n.providerName,providerAvatar:(r=e.arguments)==null?void 0:r.providerAvatar}];case"unDelegate":return["Undelegate",{egldValue:(s=e.arguments)==null?void 0:s.value},"from staking provider",{providerName:(p=e.arguments)==null?void 0:p.providerName,providerAvatar:(m=e.arguments)==null?void 0:m.providerAvatar}];case"claimRewards":return["Claim rewards from staking provider",{providerName:(_=e.arguments)==null?void 0:_.providerName,providerAvatar:(l=e.arguments)==null?void 0:l.providerAvatar}];case"reDelegateRewards":return["Redelegate rewards from staking provider",{providerName:(u=e.arguments)==null?void 0:u.providerName,providerAvatar:(y=e.arguments)==null?void 0:y.providerAvatar}];case"withdraw":return["Withdraw from staking provider",{providerName:(x=e.arguments)==null?void 0:x.providerName,providerAvatar:(w=e.arguments)==null?void 0:w.providerAvatar}];default:return[]}};var Hr=e=>{if(!e.arguments)return e.description?[e.description]:[e.name];switch(e.category){case"esdtNft":return Sr(e);case"mex":return $r(e);case"stake":return Vr(e);default:return e.description?[e.description]:[]}};t();var N=T(require("react")),Ge=T(require("classnames"));t();var Yn=T(require("react")),Fs=e=>Yn.createElement("svg",f({"aria-hidden":"true",focusable:"false","data-prefix":"far","data-icon":"globe",role:"img",xmlns:"http://www.w3.org/2000/svg",viewBox:"0 0 496 512",className:"svg-inline--fa fa-globe fa-w-16 fa-5x"},e),Yn.createElement("path",{fill:"#6c757d",d:"M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm179.3 160h-67.2c-6.7-36.5-17.5-68.8-31.2-94.7 42.9 19 77.7 52.7 98.4 94.7zM248 56c18.6 0 48.6 41.2 63.2 112H184.8C199.4 97.2 229.4 56 248 56zM48 256c0-13.7 1.4-27.1 4-40h77.7c-1 13.1-1.7 26.3-1.7 40s.7 26.9 1.7 40H52c-2.6-12.9-4-26.3-4-40zm20.7 88h67.2c6.7 36.5 17.5 68.8 31.2 94.7-42.9-19-77.7-52.7-98.4-94.7zm67.2-176H68.7c20.7-42 55.5-75.7 98.4-94.7-13.7 25.9-24.5 58.2-31.2 94.7zM248 456c-18.6 0-48.6-41.2-63.2-112h126.5c-14.7 70.8-44.7 112-63.3 112zm70.1-160H177.9c-1.1-12.8-1.9-26-1.9-40s.8-27.2 1.9-40h140.3c1.1 12.8 1.9 26 1.9 40s-.9 27.2-2 40zm10.8 142.7c13.7-25.9 24.4-58.2 31.2-94.7h67.2c-20.7 42-55.5 75.7-98.4 94.7zM366.3 296c1-13.1 1.7-26.3 1.7-40s-.7-26.9-1.7-40H444c2.6 12.9 4 26.3 4 40s-1.4 27.1-4 40h-77.7z",className:""})),jr=Fs;t();t();var Dt=T(require("react")),Oi=require("@fortawesome/free-solid-svg-icons"),Fi=require("@fortawesome/react-fontawesome"),Bi=T(require("classnames"));t();t();var ni=require("react"),Qe=require("react-redux");t();var St=require("@reduxjs/toolkit"),ei=require("react-redux/lib/utils/Subscription");L();t();var Na=T(require("lodash.throttle"));A();t();var Ea=T(require("lodash.isequal")),dn=require("reselect");M();var h=(0,dn.createSelectorCreator)(dn.defaultMemoize,Ea.default),xo=(e,o)=>e!=""&&Boolean(o);L();ct();qe();ze();var xp=[Vo],ln=!1,yp=(0,Na.default)(()=>{fo(uo())},5e3),Ca=e=>o=>n=>{var l;if(xp.includes(n.type))return o(n);let r=e.getState(),s=Ce.local.getItem(be.loginExpiresAt);if(!xo((l=r==null?void 0:r.loginInfo)==null?void 0:l.loginMethod,r==null?void 0:r.account.address))return o(n);if(s==null)return fo(uo());let m=Date.now();return s-m<0&&!ln?setTimeout(()=>{ln=!0,e.dispatch(ia())},1e3):(ln&&(ln=!1),yp()),o(n)};t();t();function _n(){return typeof sessionStorage!="undefined"}var Ya=_n()?($a(),lo(Ga)).default:(Ha(),lo(Va)).default,Ja=_n()?(za(),lo(ja)).default:[],Qa=_n()?(Za(),lo(Xa)).default:e=>e;gn();var k=(0,St.configureStore)({reducer:Ya,middleware:e=>e({serializableCheck:{ignoredActions:[...Ja,it.type,at.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(Ca)}),oi=(0,ei.createSubscription)(k),Lf=Qa(k),Ef=(0,St.configureStore)({reducer:Re});var $p={store:k,subscription:oi},wt=(0,ni.createContext)($p),Df=(0,Qe.createStoreHook)(wt),V=(0,Qe.createDispatchHook)(wt),I=(0,Qe.createSelectorHook)(wt);t();t();L();var le=e=>e.account,eo=h(le,e=>e.address),bo=h(le,eo,(e,o)=>o in e.accounts?e.accounts[o]:jo),Vp=h(le,bo,(e,o)=>{let s=e,{accounts:n}=s,r=ye(s,["accounts"]);return v(f({},r),{account:o})}),Hp=h(bo,e=>e.balance),jp=h(bo,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Ff=h(le,e=>e.shard),zp=h(le,e=>e.ledgerAccount),Bf=h(le,e=>e.walletConnectAccount),Uf=h(le,e=>e.isAccountLoading),Wf=h(le,e=>e.accountLoadingError),ti=h(le,e=>e.websocketEvent);t();var Q=e=>e.loginInfo,Hf=h(Q,e=>e.loginMethod),vt=h(Q,eo,(e,o)=>xo(e.loginMethod,o)),jf=h(Q,e=>e.walletConnectLogin),zf=h(Q,e=>e.ledgerLogin),qp=h(Q,e=>e.walletLogin),qf=h(Q,e=>e.isLoginSessionInvalid),Kp=h(Q,e=>e.tokenLogin),Kf=h(Q,e=>e.logoutRoute);t();var ri=e=>e.modals,Yf=h(ri,e=>e.txSubmittedModal),Xp=h(ri,e=>e.notificationModal);t();var ie=e=>e.networkConfig,un=h(ie,e=>e.chainID),Zp=h(ie,e=>e.network.walletConnectBridgeAddress),Yp=h(ie,e=>e.network.walletConnectV2RelayAddress),Jp=h(ie,e=>e.network.walletConnectV2ProjectId),Qp=h(ie,e=>e.network.walletConnectV2Options),ai=h(ie,e=>e.network.walletConnectDeepLink),we=h(ie,e=>e.network),ii=h(we,e=>e.apiAddress),ci=h(we,e=>e.explorerAddress),si=h(we,e=>e.egldLabel);t();var fn=e=>e.signedMessageInfo,nT=h(fn,e=>e.isSigning),tT=h(fn,e=>e.errorMessage),rT=h(fn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return e.signedSessions[o[n-1]]}),aT=h(fn,e=>{let o=Object.keys(e.signedSessions),n=o.length;return o.length>0?o[n-1]:""});t();var kt=e=>e.toasts,sT=h(kt,e=>e.customToasts),pT=h(kt,e=>e.transactionToasts),mT=h(kt,e=>e.failTransactionToast);t();L();var pi={errorMessage:gt,successMessage:ut,processingMessage:ft},em=e=>e.transactionsInfo,om=h(em,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||pi);t();t();var q=require("@multiversx/sdk-core");A();t();function Tn(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function hn(e){var s,p,m;let{data:o}=e,n=Tn(o!=null?o:"")?q.TransactionPayload.fromEncoded(o):new q.TransactionPayload(o),r=new q.Transaction(f({value:e.value.valueOf(),data:n,nonce:e.nonce.valueOf(),receiver:new q.Address(e.receiver),sender:new q.Address(e.sender),gasLimit:(s=e.gasLimit.valueOf())!=null?s:_o,gasPrice:(p=e.gasPrice.valueOf())!=null?p:$o,chainID:e.chainID.valueOf(),version:new q.TransactionVersion((m=e.version)!=null?m:Br)},e.options?{options:new q.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new q.Address(e.sender)),r}t();t();t();t();t();var tm=require("@multiversx/sdk-core/out");t();A();t();var So=require("@multiversx/sdk-core");M();t();var mi=require("@multiversx/sdk-core");function rm(e){try{let o=new mi.Address(e);return Boolean(o.bech32())}catch(o){return!1}}function Me(e){return(e==null?void 0:e.startsWith("erd"))&&e.length===62&&/^\w+$/.test(e)&&rm(e)}var am=["ESDTNFTTransfer","ESDTNFTBurn","ESDTNFTAddQuantity","ESDTNFTCreate","MultiESDTNFTTransfer","ESDTTransfer","ESDTBurn","ESDTLocalMint","ESDTLocalBurn","ESDTWipe","ESDTFreeze"];function xn(e,o,n=""){if(!Me(e))return!1;if(new So.Address(e).isContractAddress())return!0;let p=di({receiver:e,data:n});return p?new So.Address(p).isContractAddress()||sm(e,o,n):!1}var im=e=>e.toLowerCase().match(/[0-9a-f]/g),cm=e=>e.length%2===0;function sm(e,o,n){let r=n==null?void 0:n.split("@");if(r==null)return!1;let[s,...p]=r,m=o!=null&&e!=null&&e===o,_=am.includes(s),l=p.every(u=>im(u)&&cm(u));return m&&_&&l}function di({receiver:e,data:o}){try{if(!o)return e;let n=Tn(o)?So.TransactionPayload.fromEncoded(o).toString():o,r=pm(n),s=n.split("@");return r>-1?s[r]:e}catch(n){console.log(n);return}}function pm(e){return e.includes("MultiESDTNFTTransfer")?1:e.includes("ESDTNFTTransfer")?4:-1}t();t();t();var dm=require("@multiversx/sdk-core"),lm=T(require("bignumber.js"));C();t();t();t();t();t();var ce="accounts";var yn="blocks",gi="code",At="collections";var ui="contracts";var fi="identities";var Ti="locked-accounts",hi="logs",xi="miniblocks";var It="nfts",yi="nodes",Lt="providers",bi="roles",Et="sc-results";var Oe="tokens";var _e="transactions";var se={shard:e=>`/${yn}?shard=${e}`,receiverShard:e=>`/${_e}?receivershard=${e}`,senderShard:e=>`/${_e}?sendershard=${e}`,transactionDetails:e=>`/${_e}/${e}`,transactionDetailsScResults:e=>`/${_e}/${e}/${Et}`,transactionDetailsLogs:e=>`/${_e}/${e}/${hi}`,nodeDetails:e=>`/${yi}/${e}`,accountDetails:e=>`/${ce}/${e}`,accountDetailsContractCode:e=>`/${ce}/${e}/${gi}`,accountDetailsTokens:e=>`/${ce}/${e}/${Oe}`,accountDetailsNfts:e=>`/${ce}/${e}/${It}`,accountDetailsScResults:e=>`/${ce}/${e}/${Et}`,accountDetailsContracts:e=>`/${ce}/${e}/${ui}`,identityDetails:e=>`/${fi}/${e}`,tokenDetails:e=>`/${Oe}/${e}`,tokenDetailsAccounts:e=>`/${Oe}/${e}/${ce}`,tokenDetailsLockedAccounts:e=>`/${Oe}/${e}/${Ti}`,tokenDetailsRoles:e=>`/${Oe}/${e}/${bi}`,collectionDetails:e=>`/${At}/${e}`,nftDetails:e=>`/${It}/${e}`,providerDetails:e=>`/${Lt}/${e}`,providerDetailsTransactions:e=>`/${Lt}/${e}/${_e}`,miniblockDetails:e=>`/${xi}/${e}`};t();t();t();t();var Si=!1;function gm(e){Si||(console.error(e),Si=!0)}function Nt({explorerAddress:e,to:o}){try{return new URL(o).href}catch(n){return o.startsWith("/")||(gm(`Link not prepended by / : ${o}`),o=`/${o}`),e?`${e}${o}`:o}}t();U();t();t();t();var um=T(require("bignumber.js"));A();t();var ki=require("@multiversx/sdk-core"),ve=T(require("bignumber.js"));A();t();var vi=T(require("bignumber.js")),wo=(e,o=!0)=>{let n=String(e);if(!n.match(/^[-]?\d+$/))return!1;let r=new vi.default(n),s=o?0:-1;return r.toString(10)===n&&r.comparedTo(0)>=s};t();function ge(e){return{if:function(o){return o?{then:n=>n instanceof Function?ge(n(e)):ge(n)}:{then:()=>ge(e)}},then:o=>o instanceof Function?ge(o(e)):ge(o),valueOf:function(){return e}}}ve.default.config({ROUNDING_MODE:ve.default.ROUND_FLOOR});function ue({input:e,decimals:o=W,digits:n=je,showLastNonZeroDecimal:r=!0,showIsLessThanDecimalsLabel:s=!1,addCommas:p=!1}){if(!wo(e,!1))throw new Error("Invalid input");let m=new ve.default(e).isNegative(),_=e;return m&&(_=e.substring(1)),ge(_).then(()=>ki.TokenPayment.fungibleFromBigInteger("",_,o).toRationalNumber()).then(l=>{let u=new ve.default(l);if(u.isZero())return J;let y=u.toString(10),[x,w]=y.split("."),E=new ve.default(w||0),B=ge(0).if(Boolean(w&&r)).then(()=>Math.max(w.length,n)).if(E.isZero()&&!r).then(0).if(Boolean(w&&!r)).then(()=>Math.min(w.length,n)).valueOf(),G=w&&n>=1&&n<=w.length&&E.isGreaterThan(0)&&new ve.default(w.substring(0,n)).isZero(),qc=u.toFormat(B);return ge(y).if(p).then(qc).if(Boolean(G)).then(No=>{let po=new ve.default(x).isZero(),[Y,Co]=No.split("."),Po=new Array(n-1).fill(0),Kc=[...Po,0].join(""),Xc=[...Po,1].join("");return po?s?`<${Y}.${Xc}`:r?`${Y}.${Co}`:Y:`${Y}.${Kc}`}).if(Boolean(!G&&w)).then(No=>{let[po]=No.split("."),Y=w.substring(0,B);if(r){let Co=n-Y.length;if(Co>0){let Po=Array(Co).fill(0).join("");return Y=`${Y}${Po}`,`${po}.${Y}`}return No}return Y?`${po}.${Y}`:po}).valueOf()}).if(m).then(l=>`-${l}`).valueOf()}t();t();t();A();t();t();t();t();A();t();A();t();var fm=require("@multiversx/sdk-core");A();t();var vo=require("@multiversx/sdk-core"),xm=require("@multiversx/sdk-network-providers");A();t();t();var Tm=T(require("bignumber.js"));A();t();A();t();t();t();t();function oo({token:e,noValue:o,showLastNonZeroDecimal:n}){var u;let r=!o&&e.value&&e.type!=="NonFungibleESDT",s=e.decimals!=null,p="";e.type==="NonFungibleESDT"&&(p="NFT"),e.type==="SemiFungibleESDT"&&(p="SFT"),e.type==="MetaESDT"&&(p="Meta-ESDT");let m="";r&&s&&(m=s?ue({input:e.value,decimals:e.decimals,digits:2,showLastNonZeroDecimal:n}):Number(e.value).toLocaleString("en"));let _=se.nftDetails(String(e.identifier)),l=e.ticker===e.collection&&(u=e.identifier)!=null?u:e.ticker;return{badgeText:p,tokenFormattedAmount:r?m:null,tokenExplorerLink:_,tokenLinkText:l,token:e,noValue:o,showLastNonZeroDecimal:n}}t();A();function bn({token:e,noValue:o,showLastNonZeroDecimal:n}){var _;let r=Boolean(!o&&e.value),s=r?ue({input:e.value,decimals:(_=e.decimals)!=null?_:W,digits:2,showLastNonZeroDecimal:n,addCommas:!0}):null,p=se.tokenDetails(String(e.token)),m=e.ticker;return{tokenExplorerLink:p,tokenFormattedAmount:s,showFormattedAmount:r,tokenLinkText:m,token:e,showLastNonZeroDecimal:n}}t();U();var ym=["reDelegateRewards","claimRewards","unBond"],bm=["wrapEgld","unwrapEgld"],Sm=["unStake"],wm=["unDelegate"];t();t();t();A();t();var km=T(require("bignumber.js"));t();t();t();t();U();t();var Am=T(require("bignumber.js"));t();t();t();var Em=T(require("bignumber.js"));M();t();t();t();t();C();t();t();t();t();U();t();t();t();var Nm=T(require("linkifyjs"));t();A();t();var Cm=T(require("bignumber.js"));t();M();t();t();C();t();C();t();t();t();var Ei=require("react");M();t();t();U();t();t();var Pm=require("@multiversx/sdk-core/out"),Dm=T(require("bignumber.js"));U();t();M();t();t();M();var n0=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];t();var Ni=require("react");M();t();var Mm=require("react");U();var p0=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];Je();var wn=e=>e.transactions,no=h(wn,e=>e.signedTransactions),Om=h(wn,e=>e.signTransactionsError),Ci=h(wn,e=>e.signTransactionsCancelMessage),vn=e=>o=>Object.entries(o).reduce((n,[r,s])=>(e(s.status)&&(n[r]=s),n),{}),Pi=h(no,vn(Xe)),Di=h(no,vn(Ze)),Ri=h(no,vn(Ye)),Fm=h(no,vn(ht)),Mi=h(wn,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:v(f({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(n=>hn(n)))||[]})}),Bm=h(no,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});var kn=()=>I(ie);t();var Um=`.dapp-core-component__explorerLinkStyles__link .dapp-core-component__explorerLinkStyles__search {
  color: #6c757d;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Um));var Pt={link:"dapp-core-component__explorerLinkStyles__link",search:"dapp-core-component__explorerLinkStyles__search"};var fe=p=>{var m=p,{page:e,text:o,className:n="dapp-explorer-link",children:r}=m,s=ye(m,["page","text","className","children"]);let{network:{explorerAddress:_}}=kn(),l=o!=null?o:Dt.default.createElement(Fi.FontAwesomeIcon,{icon:Oi.faExternalLinkSquare,className:Pt.search}),u=Nt({explorerAddress:String(_),to:e});return Dt.default.createElement("a",f({href:u,target:"_blank",className:(0,Bi.default)(Pt.link,g.ml2,n),rel:"noreferrer"},s),r!=null?r:l)};t();t();var ke=T(require("react")),Ui=T(require("classnames"));A();t();var Wm=`.dapp-core-component__formatAmountStyles__int-amount {
  color: inherit;
}

.dapp-core-component__formatAmountStyles__decimals {
  color: inherit;
}

.dapp-core-component__formatAmountStyles__symbol {
  color: inherit;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Wm));var ko={"int-amount":"dapp-core-component__formatAmountStyles__int-amount",intAmount:"dapp-core-component__formatAmountStyles__int-amount",decimals:"dapp-core-component__formatAmountStyles__decimals",symbol:"dapp-core-component__formatAmountStyles__symbol"};var Gm=e=>ke.default.createElement("span",{"data-testid":e["data-testid"]?e["data-testid"]:"formatAmountComponent",className:e.className},ke.default.createElement("span",{className:ko["int-amount"],"data-testid":"formatAmountInt"},"...")),$m=(e,o)=>{var x;let{value:n,showLastNonZeroDecimal:r=!1,showLabel:s=!0}=e,p=e.digits!=null?e.digits:je,m=e.decimals!=null?e.decimals:W,_=ue({input:n,decimals:m,digits:p,showLastNonZeroDecimal:r,addCommas:!0}),l=_.split("."),u=l.length===1,y=_!==J;if(p>0&&u&&y){let w="";for(let E=1;E<=p;E++)w=w+J;l.push(w)}return ke.default.createElement("span",{"data-testid":e["data-testid"]?e["data-testid"]:"formatAmountComponent",className:e.className},ke.default.createElement("span",{className:ko["int-amount"],"data-testid":"formatAmountInt"},l[0]),l.length>1&&ke.default.createElement("span",{className:ko.decimals,"data-testid":"formatAmountDecimals"},".",l[1]),s&&ke.default.createElement("span",{className:(0,Ui.default)(ko.symbol,e.token&&g.textMuted),"data-testid":"formatAmountSymbol"},` ${(x=e.token)!=null?x:o}`))},Vm=e=>{let{value:o}=e;return wo(o)?$m(e,e.egldLabel||""):Gm(e)},Fe=e=>{let o=e.egldLabel||Dr,n=v(f({},e),{egldLabel:o});return ke.default.createElement(Vm,f({},n))};t();t();var jt=T(require("react")),Pc=T(require("classnames"));t();var D=T(require("react")),Nc=T(require("classnames"));A();t();t();var Wi=require("react");t();L();t();t();t();t();t();t();t();t();var zm=require("@multiversx/sdk-extension-provider"),qm=require("@multiversx/sdk-hw-provider"),Km=require("@multiversx/sdk-opera-provider"),Xm=require("@multiversx/sdk-web-wallet-provider");A();C();t();var Te=e=>`Unable to perform ${e}, Provider not initialized`,An=class{init(){return Promise.resolve(!1)}login(o){throw new Error(Te(`login with options: ${o}`))}logout(o){throw new Error(Te(`logout with options: ${o}`))}getAddress(){throw new Error(Te("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,n){throw new Error(Te(`sendTransaction with transactions: ${o} options: ${n}`))}signTransaction(o,n){throw new Error(Te(`signTransaction with transactions: ${o} options: ${n}`))}signTransactions(o,n){throw new Error(Te(`signTransactions with transactions: ${o} options: ${n}`))}signMessage(o,n){throw new Error(Te(`signTransactions with ${o} and options ${n}`))}sendCustomMessage({method:o,params:n}){throw new Error(Te(`sendCustomMessage with method: ${o} params: ${n}`))}sendCustomRequest(o){throw new Error(Te(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},Zm=new An;L();t();t();var Ym=T(require("axios"));t();t();t();C();t();t();t();Je();t();t();t();var Jm=T(require("axios"));t();var Qm=T(require("axios"));t();var od=T(require("axios"));t();t();var nd=T(require("axios"));t();var td=T(require("axios"));t();L();C();t();t();L();t();t();var Hi=require("@multiversx/sdk-core");t();F();M();t();Je();t();L();C();t();L();M();t();t();t();M();t();t();t();t();var id=T(require("axios")),cd=T(require("swr"));t();t();var ji=require("react");t();var md=require("react"),Ki=require("@multiversx/sdk-web-wallet-provider"),dd=T(require("qs"));A();L();C();t();t();var zi,R1={search:(zi=window==null?void 0:window.location)==null?void 0:zi.search,removeParams:[]};t();var Xi=require("react");pt();t();var Qi=require("react"),yd=require("@multiversx/sdk-extension-provider");A();t();L();C();t();t();t();t();var Ji=require("react"),fd=require("@multiversx/sdk-extension-provider");A();t();t();t();t();t();t();t();t();var Ao=T(require("react"));var kk=(0,Ao.createContext)({}),Ak=k.getState();t();var Yi=T(require("react"));t();var Ft=T(require("react")),_d=T(require("axios"));t();t();L();t();t();var bd=T(require("axios"));L();M();To();t();Ne();L();t();t();var ol=require("react");t();t();t();L();t();t();var vd=T(require("bignumber.js"));A();L();M();t();var wd=T(require("bignumber.js"));A();t();var kd=require("@multiversx/sdk-core"),Ad=T(require("bignumber.js"));A();t();t();t();t();var nc=T(require("axios"));t();t();t();function ec(e){return S(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}t();var oc=(e,o,n,r=0)=>S(void 0,null,function*(){try{return yield e(...n)}catch(s){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield ec(o.delay)),yield oc(e,o,n,r+1)):null}}),Ut=(e,o={retries:5,delay:500})=>(...n)=>S(void 0,null,function*(){return yield oc(e,o,n)});var Id=4;var fL=Ut((e,o)=>S(void 0,null,function*(){let{data:n}=yield nc.default.get(`${e}/${yn}?from=${Id}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[r]=n;return r}));t();t();Ho();t();t();var Jd=T(wc());t();var vc,BL={origin:typeof window!="undefined"?(vc=window==null?void 0:window.location)==null?void 0:vc.origin:"",apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};t();C();Je();t();var Ln=require("react"),kc=(e,o)=>{let[n,r]=(0,Ln.useState)(e);return(0,Ln.useEffect)(()=>{let p=setTimeout(()=>r(e),o);return()=>clearTimeout(p)},[e]),n};t();t();var tl=require("react"),rl=require("@multiversx/sdk-extension-provider");Ne();F();C();t();Pe();t();var nl=require("react"),Ac=require("@multiversx/sdk-core");L();t();var al=require("react"),il=require("@multiversx/sdk-opera-provider");Ne();F();C();Pe();t();var pl=require("react");Ne();L();t();t();Jn();t();t();var cl=T(require("platform"));t();F();M();To();qe();ze();t();t();t();t();t();t();t();t();t();C();t();To();qe();t();t();t();L();t();t();var sl=T(require("axios"));t();t();var Ic=require("react"),dl=require("@multiversx/sdk-hw-provider");Ne();F();L();C();Pe();t();var ml=require("react");t();var $t=require("react");F();L();C();Pe();t();var Vt=require("react");F();L();C();Pe();t();t();var Lc=require("react");t();var fl=require("react");t();var Ht=require("react"),ul=require("socket.io-client");L();t();t();t();t();var Ec=e=>e.dappModal,Tl=h(Ec,e=>e.visible),Cn=h(Ec,e=>e.modal),xC=h(Cn,e=>e.showHeader),yC=h(Cn,e=>e.showFooter),bC=h(Cn,e=>e.headerText),SC=h(Cn,e=>e.footerText);t();var hl=`.dapp-core-component__trim-styles__trim {
  display: flex;
  max-width: 100%;
  overflow: hidden;
  position: relative;
  white-space: nowrap;
  /* IE fix */
  /* SAFARI fix */
}
.dapp-core-component__trim-styles__trim.dapp-core-component__trim-styles__overflow .dapp-core-component__trim-styles__ellipsis {
  display: block;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: left;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  direction: rtl;
  text-align: right;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span {
  font-size: 0.875rem;
  pointer-events: none;
  user-select: none;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  flex-shrink: 0;
  display: none;
  pointer-events: none;
  user-select: none;
}
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
@supports (-webkit-hyphens: none) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right,
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
    letter-spacing: -0.001em;
  }
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__hidden-text-ref {
  position: absolute;
  display: block;
  color: transparent;
}
@media (max-width: 1199.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 26rem;
  }
}
@media (max-width: 991.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 12rem;
  }
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 8rem;
  }
}

.dapp-core-component__trim-styles__trim-wrapper {
  display: flex;
  max-width: 100%;
  overflow: hidden;
}

a:hover > .dapp-core-component__trim-styles__trim span {
  color: #0056b3;
}
a:hover > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent;
}

a > .dapp-core-component__trim-styles__trim span,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span {
  color: #1b46c2;
}
a > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim {
  max-width: 10rem;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
  max-width: none;
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}

.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  font-size: 0.875rem;
}

@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-size-xl .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(hl));var ro={trim:"dapp-core-component__trim-styles__trim",overflow:"dapp-core-component__trim-styles__overflow",ellipsis:"dapp-core-component__trim-styles__ellipsis",left:"dapp-core-component__trim-styles__left",right:"dapp-core-component__trim-styles__right","hidden-text-ref":"dapp-core-component__trim-styles__hidden-text-ref",hiddenTextRef:"dapp-core-component__trim-styles__hidden-text-ref","trim-wrapper":"dapp-core-component__trim-styles__trim-wrapper",trimWrapper:"dapp-core-component__trim-styles__trim-wrapper","text-primary":"dapp-core-component__trim-styles__text-primary",textPrimary:"dapp-core-component__trim-styles__text-primary",table:"dapp-core-component__trim-styles__table","trim-only-sm":"dapp-core-component__trim-styles__trim-only-sm",trimOnlySm:"dapp-core-component__trim-styles__trim-only-sm","trim-fs-sm":"dapp-core-component__trim-styles__trim-fs-sm",trimFsSm:"dapp-core-component__trim-styles__trim-fs-sm","trim-size-xl":"dapp-core-component__trim-styles__trim-size-xl",trimSizeXl:"dapp-core-component__trim-styles__trim-size-xl"};var Cc=({text:e,className:o="dapp-trim",dataTestId:n="trim-text-component",color:r})=>{let[s,p]=(0,D.useState)(0),[m,_]=(0,D.useState)(!1),l=(0,D.useRef)(document==null?void 0:document.createElement("span")),u=(0,D.useRef)(document==null?void 0:document.createElement("span")),y=kc(s,300),x=()=>{if(l.current&&u.current){let B=u.current.offsetWidth-l.current.offsetWidth;_(B>1)}},w=()=>{p(s+1)};return(0,D.useEffect)(()=>(window==null||window.addEventListener("resize",w),()=>{window==null||window.removeEventListener("resize",w)})),(0,D.useEffect)(()=>{x()},[y]),D.default.createElement("span",{ref:l,className:(0,Nc.default)(ro.trim,r,o,{overflow:m}),"data-testid":n},D.default.createElement("span",{ref:u,className:ro.hiddenTextRef},e),m?D.default.createElement(D.default.Fragment,null,D.default.createElement("span",{className:ro.left},D.default.createElement("span",null,String(e).substring(0,Math.floor(e.length/2)))),D.default.createElement("span",{className:ro.ellipsis},Wr),D.default.createElement("span",{className:ro.right},D.default.createElement("span",null,String(e).substring(Math.ceil(e.length/2))))):D.default.createElement("span",null,e))};var Dc=s=>{var p=s,{address:e,assets:o,color:n}=p,r=ye(p,["address","assets","color"]);if(o&&o.name){let m=o.name.replace(new RegExp("\\p{Emoji}","gu"),""),_=`${m} (${e})`;return jt.default.createElement("span",v(f({className:(0,Pc.default)(g.textTruncate,{[`text-${n}`]:n})},r),{title:_}),m)}return jt.default.createElement(Cc,f({text:e,color:n},r))};t();var Rc=T(require("react")),Mc=require("@fortawesome/free-solid-svg-icons/faFileAlt"),Oc=require("@fortawesome/react-fontawesome"),Fc=T(require("classnames"));var Bc=({initiator:e,secondInitiator:o})=>xn(e)||xn(o!=null?o:"")?Rc.default.createElement(Oc.FontAwesomeIcon,{title:"Smart Contract",icon:Mc.faFileAlt,className:(0,Fc.default)(g.mr1,g.textSecondary),"data-testid":"scIcon"}):null;t();var K=T(require("react"));t();var xl=`.dapp-core-component__styles__group {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(xl));var zt={group:"dapp-core-component__styles__group"};t();t();var Eo=T(require("react")),Pn=T(require("classnames"));t();var yl=`.dapp-core-component__styles__transaction-action-collection {
  color: #1f43f4;
  display: inline-flex;
  margin-left: 0;
}
.dapp-core-component__styles__transaction-action-collection:hover {
  text-decoration: none;
}

.dapp-core-component__styles__transaction-action-nft {
  display: contents;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer {
  margin-left: 0;
  color: #1f43f4;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer:hover {
  text-decoration: none;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__explorer .dapp-core-component__styles__data {
  margin: 0;
  margin-right: 0.25rem;
}
.dapp-core-component__styles__transaction-action-nft .dapp-core-component__styles__truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(yl));var Ue={"transaction-action-collection":"dapp-core-component__styles__transaction-action-collection",transactionActionCollection:"dapp-core-component__styles__transaction-action-collection","transaction-action-nft":"dapp-core-component__styles__transaction-action-nft",transactionActionNft:"dapp-core-component__styles__transaction-action-nft",explorer:"dapp-core-component__styles__explorer",data:"dapp-core-component__styles__data",truncate:"dapp-core-component__styles__truncate"};var Uc=({token:e})=>e.collection?Eo.default.createElement(fe,{page:se.collectionDetails(e.collection),className:(0,Pn.default)(Ue.transactionActionCollection)},Eo.default.createElement("div",{className:(0,Pn.default)(g.dFlex,g.alignItemsCenter)},e.svgUrl&&Eo.default.createElement("img",{src:e.svgUrl,alt:e.name,className:(0,Pn.default)(g.sideIcon,g.mr1)}),Eo.default.createElement("span",null,e.ticker))):null;t();var he=T(require("react")),ao=T(require("classnames"));t();var Wc=T(require("react")),Gc=T(require("classnames"));var $c=({text:e,className:o,"data-testid":n="nftBadge"})=>Wc.default.createElement("div",{"data-testid":n,className:(0,Gc.default)(g.badge,g.badgeSecondary,g.badgePill,g.fontWeightLight,o)},e);var Vc=({badgeText:e,tokenFormattedAmount:o,tokenExplorerLink:n,tokenLinkText:r,token:s,showLastNonZeroDecimal:p})=>s.identifier?he.default.createElement("div",{className:Ue.transactionActionNft},e!=null&&he.default.createElement($c,{text:e,className:(0,ao.default)(g.mr1,g.myAuto)}),o!=null&&he.default.createElement("div",{className:(0,ao.default)(g.mr1,{[g.textTruncate]:s.svgUrl})},he.default.createElement(Fe,{value:s.value,digits:2,showLabel:!1,showLastNonZeroDecimal:p,decimals:s.decimals,"data-testid":"nftFormattedAmount"})),he.default.createElement(fe,{page:n,"data-testid":"nftExplorerLink",className:(0,ao.default)(Ue.explorer,{[g.sideLink]:s.svgUrl,[g.dFlex]:s.svgUrl,[g.textTruncate]:!s.svgUrl})},he.default.createElement("div",{className:Ue.data},s.svgUrl&&he.default.createElement("img",{src:s.svgUrl,alt:s.name,className:(0,ao.default)(g.sideIcon,g.mr1)}),he.default.createElement("span",{className:(0,ao.default)({[Ue.truncate]:s.ticker===s.collection})},r)))):null;t();var xe=T(require("react")),Dn=T(require("classnames"));A();var Hc=({tokenExplorerLink:e,showFormattedAmount:o,tokenLinkText:n,token:r,showLastNonZeroDecimal:s})=>{var p;return r.token?xe.default.createElement(xe.default.Fragment,null,o&&xe.default.createElement("div",{className:g.textTruncate},xe.default.createElement(Fe,{value:r.value,digits:2,showLabel:!1,decimals:(p=r.decimals)!=null?p:W,showLastNonZeroDecimal:s,"data-testid":"tokenFormattedAmount"})),xe.default.createElement(fe,{page:e,"data-testid":"tokenExplorerLink",className:(0,Dn.default)(g.dFlex,{[g.sideLink]:r.svgUrl})},xe.default.createElement("div",{className:(0,Dn.default)(g.dFlex,g.alignItemsCenter)},r.svgUrl&&xe.default.createElement("img",{src:r.svgUrl,alt:r.name,className:(0,Dn.default)(g.sideIcon,g.mr1)}),xe.default.createElement("span",null,n)))):null};var We={Collection:Uc,Nft:Vc,Token:Hc};var qt=({token:e,noValue:o,showLastNonZeroDecimal:n})=>{if(["MetaESDT".toString(),"SemiFungibleESDT","NonFungibleESDT"].includes(e.type))switch(e.type){case"SemiFungibleESDT":{let s=oo({token:e,showLastNonZeroDecimal:n});return K.default.createElement("div",{className:zt.group},K.default.createElement("span",{className:g.mr1},"SFT quantity"),K.default.createElement(We.Nft,v(f({},s),{badgeText:null})),K.default.createElement("span",{className:g.mr1},"of collection"),K.default.createElement(We.Collection,{token:e}))}case"NonFungibleESDT":{let s=oo({token:e,noValue:o,showLastNonZeroDecimal:n});return K.default.createElement("div",{className:zt.group},K.default.createElement("span",{className:g.mr1},"NFT"),K.default.createElement(We.Nft,v(f({},s),{badgeText:null})),K.default.createElement("span",{className:g.mr1},"of collection"),K.default.createElement(We.Collection,{token:e}))}case"MetaESDT":{let s=oo({token:e,noValue:o,showLastNonZeroDecimal:n});return K.default.createElement(We.Nft,v(f({},s),{badgeText:null}))}default:return null}else{let s=bn({token:e,noValue:o,showLastNonZeroDecimal:n});return K.default.createElement(We.Token,v(f({},s),{showLastNonZeroDecimal:!0}))}};t();var bl=`.dapp-core-component__styles__address {
  display: flex;
  align-items: center;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer {
  margin-left: 0;
  position: relative;
  display: flex;
  max-width: 100%;
  overflow: hidden;
  color: #1f43f4;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer:hover {
  text-decoration: none;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span {
  max-width: none;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span [class*=left] span,
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span [class*=right] span {
  font-size: 1rem;
  color: #1f43f4;
}
.dapp-core-component__styles__address .dapp-core-component__styles__explorer span:last-child {
  color: #1f43f4;
}

.dapp-core-component__styles__token {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__token.dapp-core-component__styles__spaced {
    margin-right: 0.25rem;
  }
}
.dapp-core-component__styles__token .dapp-core-component__styles__comma {
  display: none;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__token .dapp-core-component__styles__comma {
    display: block;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(bl));var io={address:"dapp-core-component__styles__address",explorer:"dapp-core-component__styles__explorer",token:"dapp-core-component__styles__token",spaced:"dapp-core-component__styles__spaced",comma:"dapp-core-component__styles__comma"};var jc=({entry:e,transaction:o})=>{if(typeof e=="string")return N.default.createElement("span",{className:g.mr1},e.replace("eGLD","EGLD"));if(Boolean(e.address)){let n;return e.address===o.sender&&(n=o.senderAssets),e.address===o.receiver&&(n=o.receiverAssets),Me(e.address)?N.default.createElement("div",{className:io.address},N.default.createElement(Bc,{initiator:e.address}),N.default.createElement(fe,{page:se.accountDetails(e.address),"data-testid":"receiverLink",className:io.explorer},N.default.createElement(Dc,{address:e.address,assets:n}))):""}if(Boolean(e.token&&e.token.length>0))return e.token.map((n,r)=>N.default.createElement("div",{key:`tx-${n.identifier}-${r}`,className:(0,Ge.default)(io.token,{[io.spaced]:e.token.length>1})},N.default.createElement(qt,{token:n,showLastNonZeroDecimal:!0}),r<e.token.length-1&&N.default.createElement("span",{className:io.comma},",")));if(Boolean(e.tokenNoValue&&e.tokenNoValue.length>0))return e.tokenNoValue.map((n,r)=>N.default.createElement("div",{key:`tx-${n.token}-${r}`},N.default.createElement(qt,{token:n,noValue:!0,showLastNonZeroDecimal:!0}),r<e.tokenNoValue.length-1&&N.default.createElement("span",{className:(0,Ge.default)(g.mlN1,g.mr1,g.dNone,g.dSmFlex)},",")));if(Boolean(e.tokenNoLink&&e.tokenNoLink.length>0))return e.tokenNoLink.map((n,r)=>N.default.createElement("div",{key:`tx-${n.token}-${r}`},N.default.createElement("span",{className:g.mr1},n.ticker),r<e.tokenNoLink.length-1&&N.default.createElement("span",{className:(0,Ge.default)(g.mlN1,g.mr1,g.dNone,g.dSmFlex)},",")));if(Boolean(e.value))return N.default.createElement("span",null,N.default.createElement(Fe,{value:e.value,showLabel:!1,showLastNonZeroDecimal:!0}));if(Boolean(e.egldValue))return N.default.createElement("span",null,N.default.createElement(Fe,{value:e.egldValue,showLastNonZeroDecimal:!0}));if(Boolean(e.providerName)){let n=(0,Ge.default)(g.sideIcon,g.mr1,g.mrLg1,g.roundedCircle);return N.default.createElement("span",{className:(0,Ge.default)(g.dFlex,g.mr1)},N.default.createElement(fe,{page:se.providerDetails(o.receiver),className:(0,Ge.default)(g.dFlex,g.alignSelfCenter)},e.providerAvatar?N.default.createElement("img",{src:e.providerAvatar,className:n,alt:" "}):N.default.createElement(jr,{className:n}),e.providerName))}return null};t();var Sl=`.dapp-core-component__styles__action {
  display: flex;
  flex-direction: column;
}
@media (min-width: 992px) {
  .dapp-core-component__styles__action {
    flex-wrap: wrap;
    flex-direction: row;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__styles__action .dapp-core-component__styles__result {
    flex-direction: row;
    align-items: center;
    display: contents;
    margin-right: 0.25rem;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Sl));var Kt={action:"dapp-core-component__styles__action",result:"dapp-core-component__styles__result"};var zc=({transaction:e})=>{let o=(0,co.useMemo)(()=>e.action?Hr(e.action):[],[e.action]);return co.default.createElement("div",{"data-testid":"transactionAction",className:Kt.action},o.map((n,r)=>co.default.createElement("div",{key:JSON.stringify(o)+r,className:Kt.result},co.default.createElement(jc,{transaction:e,entry:n}))))};var wl=({className:e,transaction:o})=>{var s;let n=o.action&&o.action.category,r=((s=o.action)==null?void 0:s.category)!=="scCall";return n?so.default.createElement(so.default.Fragment,null,so.default.createElement(qn,{className:e,title:"Method"},zn(o)),r&&so.default.createElement(qn,{className:e,title:"Transaction Action"},so.default.createElement(zc,{transaction:o}))):null};0&&(module.exports={TransactionInfoMethod});
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
