// @multiversx/sdk-dapp/__commonjs/UI/TransactionInfo/components/ScResultsList/components/ResultHash/index.js
"use strict";var mi=Object.create;var Re=Object.defineProperty,di=Object.defineProperties,_i=Object.getOwnPropertyDescriptor,li=Object.getOwnPropertyDescriptors,gi=Object.getOwnPropertyNames,Ye=Object.getOwnPropertySymbols,ui=Object.getPrototypeOf,Ho=Object.prototype.hasOwnProperty,Yn=Object.prototype.propertyIsEnumerable;var Xn=(e,o,t)=>o in e?Re(e,o,{enumerable:!0,configurable:!0,writable:!0,value:t}):e[o]=t,u=(e,o)=>{for(var t in o||(o={}))Ho.call(o,t)&&Xn(e,t,o[t]);if(Ye)for(var t of Ye(o))Yn.call(o,t)&&Xn(e,t,o[t]);return e},w=(e,o)=>di(e,li(o));var Me=(e,o)=>{var t={};for(var r in e)Ho.call(e,r)&&o.indexOf(r)<0&&(t[r]=e[r]);if(e!=null&&Ye)for(var r of Ye(e))o.indexOf(r)<0&&Yn.call(e,r)&&(t[r]=e[r]);return t};var f=(e,o)=>()=>(e&&(o=e(e=0)),o);var B=(e,o)=>()=>(o||e((o={exports:{}}).exports,o),o.exports),Te=(e,o)=>{for(var t in o)Re(e,t,{get:o[t],enumerable:!0})},Zn=(e,o,t,r)=>{if(o&&typeof o=="object"||typeof o=="function")for(let s of gi(o))!Ho.call(e,s)&&s!==t&&Re(e,s,{get:()=>o[s],enumerable:!(r=_i(o,s))||r.enumerable});return e};var h=(e,o,t)=>(t=e!=null?mi(ui(e)):{},Zn(o||!e||!e.__esModule?Re(t,"default",{value:e,enumerable:!0}):t,e)),Oe=e=>Zn(Re({},"__esModule",{value:!0}),e);var y=(e,o,t)=>new Promise((r,s)=>{var p=_=>{try{l(t.next(_))}catch(T){s(T)}},m=_=>{try{l(t.throw(_))}catch(T){s(T)}},l=_=>_.done?r(_.value):Promise.resolve(_.value).then(p,m);l((t=t.apply(e,o)).next())});var et=B(Ze=>{"use strict";n();Ze.byteLength=hi;Ze.toByteArray=xi;Ze.fromByteArray=Si;var j=[],$=[],fi=typeof Uint8Array!="undefined"?Uint8Array:Array,Vo="ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";for(pe=0,Jn=Vo.length;pe<Jn;++pe)j[pe]=Vo[pe],$[Vo.charCodeAt(pe)]=pe;var pe,Jn;$["-".charCodeAt(0)]=62;$["_".charCodeAt(0)]=63;function Qn(e){var o=e.length;if(o%4>0)throw new Error("Invalid string. Length must be a multiple of 4");var t=e.indexOf("=");t===-1&&(t=o);var r=t===o?0:4-t%4;return[t,r]}function hi(e){var o=Qn(e),t=o[0],r=o[1];return(t+r)*3/4-r}function Ti(e,o,t){return(o+t)*3/4-t}function xi(e){var o,t=Qn(e),r=t[0],s=t[1],p=new fi(Ti(e,r,s)),m=0,l=s>0?r-4:r,_;for(_=0;_<l;_+=4)o=$[e.charCodeAt(_)]<<18|$[e.charCodeAt(_+1)]<<12|$[e.charCodeAt(_+2)]<<6|$[e.charCodeAt(_+3)],p[m++]=o>>16&255,p[m++]=o>>8&255,p[m++]=o&255;return s===2&&(o=$[e.charCodeAt(_)]<<2|$[e.charCodeAt(_+1)]>>4,p[m++]=o&255),s===1&&(o=$[e.charCodeAt(_)]<<10|$[e.charCodeAt(_+1)]<<4|$[e.charCodeAt(_+2)]>>2,p[m++]=o>>8&255,p[m++]=o&255),p}function yi(e){return j[e>>18&63]+j[e>>12&63]+j[e>>6&63]+j[e&63]}function bi(e,o,t){for(var r,s=[],p=o;p<t;p+=3)r=(e[p]<<16&16711680)+(e[p+1]<<8&65280)+(e[p+2]&255),s.push(yi(r));return s.join("")}function Si(e){for(var o,t=e.length,r=t%3,s=[],p=16383,m=0,l=t-r;m<l;m+=p)s.push(bi(e,m,m+p>l?l:m+p));return r===1?(o=e[t-1],s.push(j[o>>2]+j[o<<4&63]+"==")):r===2&&(o=(e[t-2]<<8)+e[t-1],s.push(j[o>>10]+j[o>>4&63]+j[o<<2&63]+"=")),s.join("")}});var ot=B(jo=>{n();jo.read=function(e,o,t,r,s){var p,m,l=s*8-r-1,_=(1<<l)-1,T=_>>1,b=-7,x=t?s-1:0,D=t?-1:1,R=e[o+x];for(x+=D,p=R&(1<<-b)-1,R>>=-b,b+=l;b>0;p=p*256+e[o+x],x+=D,b-=8);for(m=p&(1<<-b)-1,p>>=-b,b+=r;b>0;m=m*256+e[o+x],x+=D,b-=8);if(p===0)p=1-T;else{if(p===_)return m?NaN:(R?-1:1)*(1/0);m=m+Math.pow(2,r),p=p-T}return(R?-1:1)*m*Math.pow(2,p-r)};jo.write=function(e,o,t,r,s,p){var m,l,_,T=p*8-s-1,b=(1<<T)-1,x=b>>1,D=s===23?Math.pow(2,-24)-Math.pow(2,-77):0,R=r?0:p-1,re=r?1:-1,Z=o<0||o===0&&1/o<0?1:0;for(o=Math.abs(o),isNaN(o)||o===1/0?(l=isNaN(o)?1:0,m=b):(m=Math.floor(Math.log(o)/Math.LN2),o*(_=Math.pow(2,-m))<1&&(m--,_*=2),m+x>=1?o+=D/_:o+=D*Math.pow(2,1-x),o*_>=2&&(m++,_/=2),m+x>=b?(l=0,m=b):m+x>=1?(l=(o*_-1)*Math.pow(2,s),m=m+x):(l=o*Math.pow(2,x-1)*Math.pow(2,s),m=0));s>=8;e[t+R]=l&255,R+=re,l/=256,s-=8);for(m=m<<s|l,T+=s;T>0;e[t+R]=m&255,R+=re,m/=256,T-=8);e[t+R-re]|=Z*128}});var gt=B(ye=>{"use strict";n();var zo=et(),xe=ot(),nt=typeof Symbol=="function"&&typeof Symbol.for=="function"?Symbol.for("nodejs.util.inspect.custom"):null;ye.Buffer=d;ye.SlowBuffer=Li;ye.INSPECT_MAX_BYTES=50;var Je=2147483647;ye.kMaxLength=Je;d.TYPED_ARRAY_SUPPORT=wi();!d.TYPED_ARRAY_SUPPORT&&typeof console!="undefined"&&typeof console.error=="function"&&console.error("This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.");function wi(){try{var e=new Uint8Array(1),o={foo:function(){return 42}};return Object.setPrototypeOf(o,Uint8Array.prototype),Object.setPrototypeOf(e,o),e.foo()===42}catch(t){return!1}}Object.defineProperty(d.prototype,"parent",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.buffer}});Object.defineProperty(d.prototype,"offset",{enumerable:!0,get:function(){if(!!d.isBuffer(this))return this.byteOffset}});function J(e){if(e>Je)throw new RangeError('The value "'+e+'" is invalid for option "size"');var o=new Uint8Array(e);return Object.setPrototypeOf(o,d.prototype),o}function d(e,o,t){if(typeof e=="number"){if(typeof o=="string")throw new TypeError('The "string" argument must be of type string. Received type number');return Yo(e)}return at(e,o,t)}d.poolSize=8192;function at(e,o,t){if(typeof e=="string")return ki(e,o);if(ArrayBuffer.isView(e))return Ii(e);if(e==null)throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e);if(z(e,ArrayBuffer)||e&&z(e.buffer,ArrayBuffer)||typeof SharedArrayBuffer!="undefined"&&(z(e,SharedArrayBuffer)||e&&z(e.buffer,SharedArrayBuffer)))return Ko(e,o,t);if(typeof e=="number")throw new TypeError('The "value" argument must not be of type number. Received type number');var r=e.valueOf&&e.valueOf();if(r!=null&&r!==e)return d.from(r,o,t);var s=Ai(e);if(s)return s;if(typeof Symbol!="undefined"&&Symbol.toPrimitive!=null&&typeof e[Symbol.toPrimitive]=="function")return d.from(e[Symbol.toPrimitive]("string"),o,t);throw new TypeError("The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type "+typeof e)}d.from=function(e,o,t){return at(e,o,t)};Object.setPrototypeOf(d.prototype,Uint8Array.prototype);Object.setPrototypeOf(d,Uint8Array);function it(e){if(typeof e!="number")throw new TypeError('"size" argument must be of type number');if(e<0)throw new RangeError('The value "'+e+'" is invalid for option "size"')}function vi(e,o,t){return it(e),e<=0?J(e):o!==void 0?typeof t=="string"?J(e).fill(o,t):J(e).fill(o):J(e)}d.alloc=function(e,o,t){return vi(e,o,t)};function Yo(e){return it(e),J(e<0?0:Zo(e)|0)}d.allocUnsafe=function(e){return Yo(e)};d.allocUnsafeSlow=function(e){return Yo(e)};function ki(e,o){if((typeof o!="string"||o==="")&&(o="utf8"),!d.isEncoding(o))throw new TypeError("Unknown encoding: "+o);var t=ct(e,o)|0,r=J(t),s=r.write(e,o);return s!==t&&(r=r.slice(0,s)),r}function qo(e){for(var o=e.length<0?0:Zo(e.length)|0,t=J(o),r=0;r<o;r+=1)t[r]=e[r]&255;return t}function Ii(e){if(z(e,Uint8Array)){var o=new Uint8Array(e);return Ko(o.buffer,o.byteOffset,o.byteLength)}return qo(e)}function Ko(e,o,t){if(o<0||e.byteLength<o)throw new RangeError('"offset" is outside of buffer bounds');if(e.byteLength<o+(t||0))throw new RangeError('"length" is outside of buffer bounds');var r;return o===void 0&&t===void 0?r=new Uint8Array(e):t===void 0?r=new Uint8Array(e,o):r=new Uint8Array(e,o,t),Object.setPrototypeOf(r,d.prototype),r}function Ai(e){if(d.isBuffer(e)){var o=Zo(e.length)|0,t=J(o);return t.length===0||e.copy(t,0,0,o),t}if(e.length!==void 0)return typeof e.length!="number"||Jo(e.length)?J(0):qo(e);if(e.type==="Buffer"&&Array.isArray(e.data))return qo(e.data)}function Zo(e){if(e>=Je)throw new RangeError("Attempt to allocate Buffer larger than maximum size: 0x"+Je.toString(16)+" bytes");return e|0}function Li(e){return+e!=e&&(e=0),d.alloc(+e)}d.isBuffer=function(o){return o!=null&&o._isBuffer===!0&&o!==d.prototype};d.compare=function(o,t){if(z(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),z(t,Uint8Array)&&(t=d.from(t,t.offset,t.byteLength)),!d.isBuffer(o)||!d.isBuffer(t))throw new TypeError('The "buf1", "buf2" arguments must be one of type Buffer or Uint8Array');if(o===t)return 0;for(var r=o.length,s=t.length,p=0,m=Math.min(r,s);p<m;++p)if(o[p]!==t[p]){r=o[p],s=t[p];break}return r<s?-1:s<r?1:0};d.isEncoding=function(o){switch(String(o).toLowerCase()){case"hex":case"utf8":case"utf-8":case"ascii":case"latin1":case"binary":case"base64":case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return!0;default:return!1}};d.concat=function(o,t){if(!Array.isArray(o))throw new TypeError('"list" argument must be an Array of Buffers');if(o.length===0)return d.alloc(0);var r;if(t===void 0)for(t=0,r=0;r<o.length;++r)t+=o[r].length;var s=d.allocUnsafe(t),p=0;for(r=0;r<o.length;++r){var m=o[r];if(z(m,Uint8Array))p+m.length>s.length?d.from(m).copy(s,p):Uint8Array.prototype.set.call(s,m,p);else if(d.isBuffer(m))m.copy(s,p);else throw new TypeError('"list" argument must be an Array of Buffers');p+=m.length}return s};function ct(e,o){if(d.isBuffer(e))return e.length;if(ArrayBuffer.isView(e)||z(e,ArrayBuffer))return e.byteLength;if(typeof e!="string")throw new TypeError('The "string" argument must be one of type string, Buffer, or ArrayBuffer. Received type '+typeof e);var t=e.length,r=arguments.length>2&&arguments[2]===!0;if(!r&&t===0)return 0;for(var s=!1;;)switch(o){case"ascii":case"latin1":case"binary":return t;case"utf8":case"utf-8":return Xo(e).length;case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return t*2;case"hex":return t>>>1;case"base64":return lt(e).length;default:if(s)return r?-1:Xo(e).length;o=(""+o).toLowerCase(),s=!0}}d.byteLength=ct;function Ei(e,o,t){var r=!1;if((o===void 0||o<0)&&(o=0),o>this.length||((t===void 0||t>this.length)&&(t=this.length),t<=0)||(t>>>=0,o>>>=0,t<=o))return"";for(e||(e="utf8");;)switch(e){case"hex":return Ui(this,o,t);case"utf8":case"utf-8":return pt(this,o,t);case"ascii":return Fi(this,o,t);case"latin1":case"binary":return Bi(this,o,t);case"base64":return Mi(this,o,t);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Wi(this,o,t);default:if(r)throw new TypeError("Unknown encoding: "+e);e=(e+"").toLowerCase(),r=!0}}d.prototype._isBuffer=!0;function me(e,o,t){var r=e[o];e[o]=e[t],e[t]=r}d.prototype.swap16=function(){var o=this.length;if(o%2!==0)throw new RangeError("Buffer size must be a multiple of 16-bits");for(var t=0;t<o;t+=2)me(this,t,t+1);return this};d.prototype.swap32=function(){var o=this.length;if(o%4!==0)throw new RangeError("Buffer size must be a multiple of 32-bits");for(var t=0;t<o;t+=4)me(this,t,t+3),me(this,t+1,t+2);return this};d.prototype.swap64=function(){var o=this.length;if(o%8!==0)throw new RangeError("Buffer size must be a multiple of 64-bits");for(var t=0;t<o;t+=8)me(this,t,t+7),me(this,t+1,t+6),me(this,t+2,t+5),me(this,t+3,t+4);return this};d.prototype.toString=function(){var o=this.length;return o===0?"":arguments.length===0?pt(this,0,o):Ei.apply(this,arguments)};d.prototype.toLocaleString=d.prototype.toString;d.prototype.equals=function(o){if(!d.isBuffer(o))throw new TypeError("Argument must be a Buffer");return this===o?!0:d.compare(this,o)===0};d.prototype.inspect=function(){var o="",t=ye.INSPECT_MAX_BYTES;return o=this.toString("hex",0,t).replace(/(.{2})/g,"$1 ").trim(),this.length>t&&(o+=" ... "),"<Buffer "+o+">"};nt&&(d.prototype[nt]=d.prototype.inspect);d.prototype.compare=function(o,t,r,s,p){if(z(o,Uint8Array)&&(o=d.from(o,o.offset,o.byteLength)),!d.isBuffer(o))throw new TypeError('The "target" argument must be one of type Buffer or Uint8Array. Received type '+typeof o);if(t===void 0&&(t=0),r===void 0&&(r=o?o.length:0),s===void 0&&(s=0),p===void 0&&(p=this.length),t<0||r>o.length||s<0||p>this.length)throw new RangeError("out of range index");if(s>=p&&t>=r)return 0;if(s>=p)return-1;if(t>=r)return 1;if(t>>>=0,r>>>=0,s>>>=0,p>>>=0,this===o)return 0;for(var m=p-s,l=r-t,_=Math.min(m,l),T=this.slice(s,p),b=o.slice(t,r),x=0;x<_;++x)if(T[x]!==b[x]){m=T[x],l=b[x];break}return m<l?-1:l<m?1:0};function st(e,o,t,r,s){if(e.length===0)return-1;if(typeof t=="string"?(r=t,t=0):t>2147483647?t=2147483647:t<-2147483648&&(t=-2147483648),t=+t,Jo(t)&&(t=s?0:e.length-1),t<0&&(t=e.length+t),t>=e.length){if(s)return-1;t=e.length-1}else if(t<0)if(s)t=0;else return-1;if(typeof o=="string"&&(o=d.from(o,r)),d.isBuffer(o))return o.length===0?-1:tt(e,o,t,r,s);if(typeof o=="number")return o=o&255,typeof Uint8Array.prototype.indexOf=="function"?s?Uint8Array.prototype.indexOf.call(e,o,t):Uint8Array.prototype.lastIndexOf.call(e,o,t):tt(e,[o],t,r,s);throw new TypeError("val must be string, number or Buffer")}function tt(e,o,t,r,s){var p=1,m=e.length,l=o.length;if(r!==void 0&&(r=String(r).toLowerCase(),r==="ucs2"||r==="ucs-2"||r==="utf16le"||r==="utf-16le")){if(e.length<2||o.length<2)return-1;p=2,m/=2,l/=2,t/=2}function _(R,re){return p===1?R[re]:R.readUInt16BE(re*p)}var T;if(s){var b=-1;for(T=t;T<m;T++)if(_(e,T)===_(o,b===-1?0:T-b)){if(b===-1&&(b=T),T-b+1===l)return b*p}else b!==-1&&(T-=T-b),b=-1}else for(t+l>m&&(t=m-l),T=t;T>=0;T--){for(var x=!0,D=0;D<l;D++)if(_(e,T+D)!==_(o,D)){x=!1;break}if(x)return T}return-1}d.prototype.includes=function(o,t,r){return this.indexOf(o,t,r)!==-1};d.prototype.indexOf=function(o,t,r){return st(this,o,t,r,!0)};d.prototype.lastIndexOf=function(o,t,r){return st(this,o,t,r,!1)};function Ci(e,o,t,r){t=Number(t)||0;var s=e.length-t;r?(r=Number(r),r>s&&(r=s)):r=s;var p=o.length;r>p/2&&(r=p/2);for(var m=0;m<r;++m){var l=parseInt(o.substr(m*2,2),16);if(Jo(l))return m;e[t+m]=l}return m}function Pi(e,o,t,r){return Qe(Xo(o,e.length-t),e,t,r)}function Ni(e,o,t,r){return Qe(Hi(o),e,t,r)}function Di(e,o,t,r){return Qe(lt(o),e,t,r)}function Ri(e,o,t,r){return Qe(Vi(o,e.length-t),e,t,r)}d.prototype.write=function(o,t,r,s){if(t===void 0)s="utf8",r=this.length,t=0;else if(r===void 0&&typeof t=="string")s=t,r=this.length,t=0;else if(isFinite(t))t=t>>>0,isFinite(r)?(r=r>>>0,s===void 0&&(s="utf8")):(s=r,r=void 0);else throw new Error("Buffer.write(string, encoding, offset[, length]) is no longer supported");var p=this.length-t;if((r===void 0||r>p)&&(r=p),o.length>0&&(r<0||t<0)||t>this.length)throw new RangeError("Attempt to write outside buffer bounds");s||(s="utf8");for(var m=!1;;)switch(s){case"hex":return Ci(this,o,t,r);case"utf8":case"utf-8":return Pi(this,o,t,r);case"ascii":case"latin1":case"binary":return Ni(this,o,t,r);case"base64":return Di(this,o,t,r);case"ucs2":case"ucs-2":case"utf16le":case"utf-16le":return Ri(this,o,t,r);default:if(m)throw new TypeError("Unknown encoding: "+s);s=(""+s).toLowerCase(),m=!0}};d.prototype.toJSON=function(){return{type:"Buffer",data:Array.prototype.slice.call(this._arr||this,0)}};function Mi(e,o,t){return o===0&&t===e.length?zo.fromByteArray(e):zo.fromByteArray(e.slice(o,t))}function pt(e,o,t){t=Math.min(e.length,t);for(var r=[],s=o;s<t;){var p=e[s],m=null,l=p>239?4:p>223?3:p>191?2:1;if(s+l<=t){var _,T,b,x;switch(l){case 1:p<128&&(m=p);break;case 2:_=e[s+1],(_&192)===128&&(x=(p&31)<<6|_&63,x>127&&(m=x));break;case 3:_=e[s+1],T=e[s+2],(_&192)===128&&(T&192)===128&&(x=(p&15)<<12|(_&63)<<6|T&63,x>2047&&(x<55296||x>57343)&&(m=x));break;case 4:_=e[s+1],T=e[s+2],b=e[s+3],(_&192)===128&&(T&192)===128&&(b&192)===128&&(x=(p&15)<<18|(_&63)<<12|(T&63)<<6|b&63,x>65535&&x<1114112&&(m=x))}}m===null?(m=65533,l=1):m>65535&&(m-=65536,r.push(m>>>10&1023|55296),m=56320|m&1023),r.push(m),s+=l}return Oi(r)}var rt=4096;function Oi(e){var o=e.length;if(o<=rt)return String.fromCharCode.apply(String,e);for(var t="",r=0;r<o;)t+=String.fromCharCode.apply(String,e.slice(r,r+=rt));return t}function Fi(e,o,t){var r="";t=Math.min(e.length,t);for(var s=o;s<t;++s)r+=String.fromCharCode(e[s]&127);return r}function Bi(e,o,t){var r="";t=Math.min(e.length,t);for(var s=o;s<t;++s)r+=String.fromCharCode(e[s]);return r}function Ui(e,o,t){var r=e.length;(!o||o<0)&&(o=0),(!t||t<0||t>r)&&(t=r);for(var s="",p=o;p<t;++p)s+=ji[e[p]];return s}function Wi(e,o,t){for(var r=e.slice(o,t),s="",p=0;p<r.length-1;p+=2)s+=String.fromCharCode(r[p]+r[p+1]*256);return s}d.prototype.slice=function(o,t){var r=this.length;o=~~o,t=t===void 0?r:~~t,o<0?(o+=r,o<0&&(o=0)):o>r&&(o=r),t<0?(t+=r,t<0&&(t=0)):t>r&&(t=r),t<o&&(t=o);var s=this.subarray(o,t);return Object.setPrototypeOf(s,d.prototype),s};function E(e,o,t){if(e%1!==0||e<0)throw new RangeError("offset is not uint");if(e+o>t)throw new RangeError("Trying to access beyond buffer length")}d.prototype.readUintLE=d.prototype.readUIntLE=function(o,t,r){o=o>>>0,t=t>>>0,r||E(o,t,this.length);for(var s=this[o],p=1,m=0;++m<t&&(p*=256);)s+=this[o+m]*p;return s};d.prototype.readUintBE=d.prototype.readUIntBE=function(o,t,r){o=o>>>0,t=t>>>0,r||E(o,t,this.length);for(var s=this[o+--t],p=1;t>0&&(p*=256);)s+=this[o+--t]*p;return s};d.prototype.readUint8=d.prototype.readUInt8=function(o,t){return o=o>>>0,t||E(o,1,this.length),this[o]};d.prototype.readUint16LE=d.prototype.readUInt16LE=function(o,t){return o=o>>>0,t||E(o,2,this.length),this[o]|this[o+1]<<8};d.prototype.readUint16BE=d.prototype.readUInt16BE=function(o,t){return o=o>>>0,t||E(o,2,this.length),this[o]<<8|this[o+1]};d.prototype.readUint32LE=d.prototype.readUInt32LE=function(o,t){return o=o>>>0,t||E(o,4,this.length),(this[o]|this[o+1]<<8|this[o+2]<<16)+this[o+3]*16777216};d.prototype.readUint32BE=d.prototype.readUInt32BE=function(o,t){return o=o>>>0,t||E(o,4,this.length),this[o]*16777216+(this[o+1]<<16|this[o+2]<<8|this[o+3])};d.prototype.readIntLE=function(o,t,r){o=o>>>0,t=t>>>0,r||E(o,t,this.length);for(var s=this[o],p=1,m=0;++m<t&&(p*=256);)s+=this[o+m]*p;return p*=128,s>=p&&(s-=Math.pow(2,8*t)),s};d.prototype.readIntBE=function(o,t,r){o=o>>>0,t=t>>>0,r||E(o,t,this.length);for(var s=t,p=1,m=this[o+--s];s>0&&(p*=256);)m+=this[o+--s]*p;return p*=128,m>=p&&(m-=Math.pow(2,8*t)),m};d.prototype.readInt8=function(o,t){return o=o>>>0,t||E(o,1,this.length),this[o]&128?(255-this[o]+1)*-1:this[o]};d.prototype.readInt16LE=function(o,t){o=o>>>0,t||E(o,2,this.length);var r=this[o]|this[o+1]<<8;return r&32768?r|4294901760:r};d.prototype.readInt16BE=function(o,t){o=o>>>0,t||E(o,2,this.length);var r=this[o+1]|this[o]<<8;return r&32768?r|4294901760:r};d.prototype.readInt32LE=function(o,t){return o=o>>>0,t||E(o,4,this.length),this[o]|this[o+1]<<8|this[o+2]<<16|this[o+3]<<24};d.prototype.readInt32BE=function(o,t){return o=o>>>0,t||E(o,4,this.length),this[o]<<24|this[o+1]<<16|this[o+2]<<8|this[o+3]};d.prototype.readFloatLE=function(o,t){return o=o>>>0,t||E(o,4,this.length),xe.read(this,o,!0,23,4)};d.prototype.readFloatBE=function(o,t){return o=o>>>0,t||E(o,4,this.length),xe.read(this,o,!1,23,4)};d.prototype.readDoubleLE=function(o,t){return o=o>>>0,t||E(o,8,this.length),xe.read(this,o,!0,52,8)};d.prototype.readDoubleBE=function(o,t){return o=o>>>0,t||E(o,8,this.length),xe.read(this,o,!1,52,8)};function M(e,o,t,r,s,p){if(!d.isBuffer(e))throw new TypeError('"buffer" argument must be a Buffer instance');if(o>s||o<p)throw new RangeError('"value" argument is out of bounds');if(t+r>e.length)throw new RangeError("Index out of range")}d.prototype.writeUintLE=d.prototype.writeUIntLE=function(o,t,r,s){if(o=+o,t=t>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;M(this,o,t,r,p,0)}var m=1,l=0;for(this[t]=o&255;++l<r&&(m*=256);)this[t+l]=o/m&255;return t+r};d.prototype.writeUintBE=d.prototype.writeUIntBE=function(o,t,r,s){if(o=+o,t=t>>>0,r=r>>>0,!s){var p=Math.pow(2,8*r)-1;M(this,o,t,r,p,0)}var m=r-1,l=1;for(this[t+m]=o&255;--m>=0&&(l*=256);)this[t+m]=o/l&255;return t+r};d.prototype.writeUint8=d.prototype.writeUInt8=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,1,255,0),this[t]=o&255,t+1};d.prototype.writeUint16LE=d.prototype.writeUInt16LE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,2,65535,0),this[t]=o&255,this[t+1]=o>>>8,t+2};d.prototype.writeUint16BE=d.prototype.writeUInt16BE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,2,65535,0),this[t]=o>>>8,this[t+1]=o&255,t+2};d.prototype.writeUint32LE=d.prototype.writeUInt32LE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,4,4294967295,0),this[t+3]=o>>>24,this[t+2]=o>>>16,this[t+1]=o>>>8,this[t]=o&255,t+4};d.prototype.writeUint32BE=d.prototype.writeUInt32BE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,4,4294967295,0),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};d.prototype.writeIntLE=function(o,t,r,s){if(o=+o,t=t>>>0,!s){var p=Math.pow(2,8*r-1);M(this,o,t,r,p-1,-p)}var m=0,l=1,_=0;for(this[t]=o&255;++m<r&&(l*=256);)o<0&&_===0&&this[t+m-1]!==0&&(_=1),this[t+m]=(o/l>>0)-_&255;return t+r};d.prototype.writeIntBE=function(o,t,r,s){if(o=+o,t=t>>>0,!s){var p=Math.pow(2,8*r-1);M(this,o,t,r,p-1,-p)}var m=r-1,l=1,_=0;for(this[t+m]=o&255;--m>=0&&(l*=256);)o<0&&_===0&&this[t+m+1]!==0&&(_=1),this[t+m]=(o/l>>0)-_&255;return t+r};d.prototype.writeInt8=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,1,127,-128),o<0&&(o=255+o+1),this[t]=o&255,t+1};d.prototype.writeInt16LE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,2,32767,-32768),this[t]=o&255,this[t+1]=o>>>8,t+2};d.prototype.writeInt16BE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,2,32767,-32768),this[t]=o>>>8,this[t+1]=o&255,t+2};d.prototype.writeInt32LE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,4,2147483647,-2147483648),this[t]=o&255,this[t+1]=o>>>8,this[t+2]=o>>>16,this[t+3]=o>>>24,t+4};d.prototype.writeInt32BE=function(o,t,r){return o=+o,t=t>>>0,r||M(this,o,t,4,2147483647,-2147483648),o<0&&(o=4294967295+o+1),this[t]=o>>>24,this[t+1]=o>>>16,this[t+2]=o>>>8,this[t+3]=o&255,t+4};function mt(e,o,t,r,s,p){if(t+r>e.length)throw new RangeError("Index out of range");if(t<0)throw new RangeError("Index out of range")}function dt(e,o,t,r,s){return o=+o,t=t>>>0,s||mt(e,o,t,4,34028234663852886e22,-34028234663852886e22),xe.write(e,o,t,r,23,4),t+4}d.prototype.writeFloatLE=function(o,t,r){return dt(this,o,t,!0,r)};d.prototype.writeFloatBE=function(o,t,r){return dt(this,o,t,!1,r)};function _t(e,o,t,r,s){return o=+o,t=t>>>0,s||mt(e,o,t,8,17976931348623157e292,-17976931348623157e292),xe.write(e,o,t,r,52,8),t+8}d.prototype.writeDoubleLE=function(o,t,r){return _t(this,o,t,!0,r)};d.prototype.writeDoubleBE=function(o,t,r){return _t(this,o,t,!1,r)};d.prototype.copy=function(o,t,r,s){if(!d.isBuffer(o))throw new TypeError("argument should be a Buffer");if(r||(r=0),!s&&s!==0&&(s=this.length),t>=o.length&&(t=o.length),t||(t=0),s>0&&s<r&&(s=r),s===r||o.length===0||this.length===0)return 0;if(t<0)throw new RangeError("targetStart out of bounds");if(r<0||r>=this.length)throw new RangeError("Index out of range");if(s<0)throw new RangeError("sourceEnd out of bounds");s>this.length&&(s=this.length),o.length-t<s-r&&(s=o.length-t+r);var p=s-r;return this===o&&typeof Uint8Array.prototype.copyWithin=="function"?this.copyWithin(t,r,s):Uint8Array.prototype.set.call(o,this.subarray(r,s),t),p};d.prototype.fill=function(o,t,r,s){if(typeof o=="string"){if(typeof t=="string"?(s=t,t=0,r=this.length):typeof r=="string"&&(s=r,r=this.length),s!==void 0&&typeof s!="string")throw new TypeError("encoding must be a string");if(typeof s=="string"&&!d.isEncoding(s))throw new TypeError("Unknown encoding: "+s);if(o.length===1){var p=o.charCodeAt(0);(s==="utf8"&&p<128||s==="latin1")&&(o=p)}}else typeof o=="number"?o=o&255:typeof o=="boolean"&&(o=Number(o));if(t<0||this.length<t||this.length<r)throw new RangeError("Out of range index");if(r<=t)return this;t=t>>>0,r=r===void 0?this.length:r>>>0,o||(o=0);var m;if(typeof o=="number")for(m=t;m<r;++m)this[m]=o;else{var l=d.isBuffer(o)?o:d.from(o,s),_=l.length;if(_===0)throw new TypeError('The value "'+o+'" is invalid for argument "value"');for(m=0;m<r-t;++m)this[m+t]=l[m%_]}return this};var Gi=/[^+/0-9A-Za-z-_]/g;function $i(e){if(e=e.split("=")[0],e=e.trim().replace(Gi,""),e.length<2)return"";for(;e.length%4!==0;)e=e+"=";return e}function Xo(e,o){o=o||1/0;for(var t,r=e.length,s=null,p=[],m=0;m<r;++m){if(t=e.charCodeAt(m),t>55295&&t<57344){if(!s){if(t>56319){(o-=3)>-1&&p.push(239,191,189);continue}else if(m+1===r){(o-=3)>-1&&p.push(239,191,189);continue}s=t;continue}if(t<56320){(o-=3)>-1&&p.push(239,191,189),s=t;continue}t=(s-55296<<10|t-56320)+65536}else s&&(o-=3)>-1&&p.push(239,191,189);if(s=null,t<128){if((o-=1)<0)break;p.push(t)}else if(t<2048){if((o-=2)<0)break;p.push(t>>6|192,t&63|128)}else if(t<65536){if((o-=3)<0)break;p.push(t>>12|224,t>>6&63|128,t&63|128)}else if(t<1114112){if((o-=4)<0)break;p.push(t>>18|240,t>>12&63|128,t>>6&63|128,t&63|128)}else throw new Error("Invalid code point")}return p}function Hi(e){for(var o=[],t=0;t<e.length;++t)o.push(e.charCodeAt(t)&255);return o}function Vi(e,o){for(var t,r,s,p=[],m=0;m<e.length&&!((o-=2)<0);++m)t=e.charCodeAt(m),r=t>>8,s=t%256,p.push(s),p.push(r);return p}function lt(e){return zo.toByteArray($i(e))}function Qe(e,o,t,r){for(var s=0;s<r&&!(s+t>=o.length||s>=e.length);++s)o[s+t]=e[s];return s}function z(e,o){return e instanceof o||e!=null&&e.constructor!=null&&e.constructor.name!=null&&e.constructor.name===o.name}function Jo(e){return e!==e}var ji=function(){for(var e="0123456789abcdef",o=new Array(256),t=0;t<16;++t)for(var r=t*16,s=0;s<16;++s)o[r+s]=e[t]+e[s];return o}()});var xt=B((Qm,Tt)=>{n();var L=Tt.exports={},q,K;function Qo(){throw new Error("setTimeout has not been defined")}function en(){throw new Error("clearTimeout has not been defined")}(function(){try{typeof setTimeout=="function"?q=setTimeout:q=Qo}catch(e){q=Qo}try{typeof clearTimeout=="function"?K=clearTimeout:K=en}catch(e){K=en}})();function ut(e){if(q===setTimeout)return setTimeout(e,0);if((q===Qo||!q)&&setTimeout)return q=setTimeout,setTimeout(e,0);try{return q(e,0)}catch(o){try{return q.call(null,e,0)}catch(t){return q.call(this,e,0)}}}function zi(e){if(K===clearTimeout)return clearTimeout(e);if((K===en||!K)&&clearTimeout)return K=clearTimeout,clearTimeout(e);try{return K(e)}catch(o){try{return K.call(null,e)}catch(t){return K.call(this,e)}}}var Q=[],be=!1,de,eo=-1;function qi(){!be||!de||(be=!1,de.length?Q=de.concat(Q):eo=-1,Q.length&&ft())}function ft(){if(!be){var e=ut(qi);be=!0;for(var o=Q.length;o;){for(de=Q,Q=[];++eo<o;)de&&de[eo].run();eo=-1,o=Q.length}de=null,be=!1,zi(e)}}L.nextTick=function(e){var o=new Array(arguments.length-1);if(arguments.length>1)for(var t=1;t<arguments.length;t++)o[t-1]=arguments[t];Q.push(new ht(e,o)),Q.length===1&&!be&&ut(ft)};function ht(e,o){this.fun=e,this.array=o}ht.prototype.run=function(){this.fun.apply(null,this.array)};L.title="browser";L.browser=!0;L.env={};L.argv=[];L.version="";L.versions={};function ee(){}L.on=ee;L.addListener=ee;L.once=ee;L.off=ee;L.removeListener=ee;L.removeAllListeners=ee;L.emit=ee;L.prependListener=ee;L.prependOnceListener=ee;L.listeners=function(e){return[]};L.binding=function(e){throw new Error("process.binding is not supported")};L.cwd=function(){return"/"};L.chdir=function(e){throw new Error("process.chdir is not supported")};L.umask=function(){return 0}});var i,c,Ki,a,n=f(()=>{i=h(gt()),c=h(xt()),Ki=function(e){function o(){var r=this||self;return delete e.prototype.__magic__,r}if(typeof globalThis=="object")return globalThis;if(this)return o();e.defineProperty(e.prototype,"__magic__",{configurable:!0,get:o});var t=__magic__;return t}(Object),a=Ki});var _e=f(()=>{"use strict";n()});var St=f(()=>{"use strict";n()});var wt=f(()=>{"use strict";n()});var vt=f(()=>{"use strict";n()});var A=f(()=>{"use strict";n()});var kt=f(()=>{"use strict";n()});var It=f(()=>{"use strict";n()});var O=f(()=>{"use strict";n()});var At=f(()=>{"use strict";n()});var Lt=f(()=>{"use strict";n()});var C=f(()=>{"use strict";n();St();wt();vt();A();kt();It();O();At();Lt()});var on,Et,Ed,Ct,Cd,Pt,Pd,Nd,Dd,Nt=f(()=>{"use strict";n();C();on={devnet:{id:"devnet",chainId:"D",name:"Devnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://devnet-wallet.multiversx.com",apiAddress:"https://devnet-api.multiversx.com",explorerAddress:"http://devnet-explorer.multiversx.com",apiTimeout:"4000"},testnet:{id:"testnet",chainId:"T",name:"Testnet",egldLabel:"xEGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://testnet-wallet.multiversx.com",apiAddress:"https://testnet-api.multiversx.com",explorerAddress:"http://testnet-explorer.multiversx.com",apiTimeout:"4000"},mainnet:{id:"mainnet",chainId:"1",name:"Mainnet",egldLabel:"EGLD",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"https://maiar.page.link/?apn=com.elrond.maiar.wallet&isi=1519405832&ibi=com.elrond.maiar.wallet&link=https://xportal.com/",walletConnectBridgeAddresses:["https://bridge.walletconnect.org"],walletConnectV2RelayAddresses:["wss://relay.walletconnect.com"],walletAddress:"https://wallet.multiversx.com",apiAddress:"https://api.multiversx.com",explorerAddress:"https://explorer.multiversx.com",apiTimeout:"4000"}},{chainId:Et,egldLabel:Ed}=on["devnet"],{chainId:Ct,egldLabel:Cd}=on["testnet"],{chainId:Pt,egldLabel:Pd}=on["mainnet"],Nd={["devnet"]:Et,["testnet"]:Ct,["mainnet"]:Pt},Dd={[Et]:"devnet",[Ct]:"testnet",[Pt]:"mainnet"}});var nn=f(()=>{"use strict";n()});var Dt=f(()=>{"use strict";n()});var oo,tn=f(()=>{"use strict";n();oo="transaction-status-toast-id"});var X,Rt=f(()=>{"use strict";n();X=require("@multiversx/sdk-web-wallet-provider")});var Mt=f(()=>{"use strict";n()});var Fe,no,Ot,to,Ft,Se,v=f(()=>{"use strict";n();_e();Nt();nn();Dt();tn();Rt();Mt();Fe=5e4,no=1e9,Ot=1,to="logout",Ft="login",Se="0"});var Be,ro=f(()=>{"use strict";n();Be=()=>Date.now()/1e3});var Bt=f(()=>{"use strict";n()});var Ut=f(()=>{"use strict";n()});var rn=f(()=>{"use strict";n();ro();Bt();Ut()});var cn={};Te(cn,{getItem:()=>ec,localStorageKeys:()=>ae,removeItem:()=>oc,setItem:()=>Qi});var ae,an,Qi,ec,oc,we=f(()=>{"use strict";n();rn();ae={loginExpiresAt:"sdk-dapp-login-expires-at",logoutEvent:"sdk-dapp-logout-event"},an=typeof localStorage!="undefined",Qi=({key:e,data:o,expires:t})=>{!an||localStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},ec=e=>{if(!an)return;let o=localStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Be()>=t.expires?(localStorage.removeItem(String(e)),null):t.data},oc=e=>{!an||localStorage.removeItem(String(e))}});var sn={};Te(sn,{clear:()=>Ht,getItem:()=>Gt,removeItem:()=>$t,setItem:()=>Wt,storage:()=>nc});var Wt,Gt,$t,Ht,nc,Vt=f(()=>{"use strict";n();Wt=({key:e,data:o,expires:t})=>{sessionStorage.setItem(String(e),JSON.stringify({expires:t,data:o}))},Gt=e=>{let o=sessionStorage.getItem(String(e));if(!o)return null;let t=JSON.parse(o);return!t||!t.hasOwnProperty("expires")||!t.hasOwnProperty("data")?null:Date.now()>=t.expires?(sessionStorage.removeItem(String(e)),null):t.data},$t=e=>sessionStorage.removeItem(String(e)),Ht=()=>sessionStorage.clear(),nc={setItem:Wt,getItem:Gt,removeItem:$t,clear:Ht}});var le,ve=f(()=>{"use strict";n();we();Vt();le={session:sn,local:cn}});var pn,P,Y,N=f(()=>{"use strict";n();pn=require("@reduxjs/toolkit");v();P=(0,pn.createAction)(to),Y=(0,pn.createAction)(Ft,e=>({payload:e}))});var dn,jt,zt,ao,mn,qt,_n,d_,ln,__,tc,rc,l_,g_,u_,ac,io,co=f(()=>{"use strict";n();dn=require("@multiversx/sdk-core"),jt=require("@reduxjs/toolkit"),zt=require("redux-persist");v();ve();we();N();ao={balance:"...",address:"",nonce:0,txCount:0,scrCount:0,claimableRewards:Se},mn={address:"",websocketEvent:null,accounts:{"":ao},ledgerAccount:null,publicKey:"",walletConnectAccount:null,isAccountLoading:!0,accountLoadingError:null},qt=(0,jt.createSlice)({name:"accountInfoSlice",initialState:mn,reducers:{setAddress:(e,o)=>{let t=o.payload;e.address=t,e.publicKey=new dn.Address(t).hex()},setAccount:(e,o)=>{let t=e.address===o.payload.address;e.accounts={[e.address]:t?o.payload:ao},e.shard=o.payload.shard,e.isAccountLoading=!1,e.accountLoadingError=null},setAccountNonce:(e,o)=>{let{address:t}=e;e.accounts[t].nonce=o.payload},setAccountShard:(e,o)=>{e.shard=o.payload},setLedgerAccount:(e,o)=>{e.ledgerAccount=o.payload},updateLedgerAccount:(e,o)=>{e.ledgerAccount!=null&&(e.ledgerAccount.index=o.payload.index,e.ledgerAccount.address=o.payload.address)},setWalletConnectAccount:(e,o)=>{e.walletConnectAccount=o.payload},setIsAccountLoading:(e,o)=>{e.isAccountLoading=o.payload,e.accountLoadingError=null},setAccountLoadingError:(e,o)=>{e.accountLoadingError=o.payload,e.isAccountLoading=!1},setWebsocketEvent:(e,o)=>{e.websocketEvent={timestamp:Date.now(),message:o.payload}}},extraReducers:e=>{e.addCase(P,()=>(le.local.removeItem(ae.loginExpiresAt),mn)),e.addCase(Y,(o,t)=>{let{address:r}=t.payload;o.address=r,o.publicKey=new dn.Address(r).hex()}),e.addCase(zt.REHYDRATE,(o,t)=>{var T;if(!((T=t.payload)!=null&&T.account))return;let{account:r}=t.payload,{address:s,shard:p,accounts:m,publicKey:l}=r;o.address=s,o.shard=p;let _=m&&s in m;o.accounts=_?m:mn.accounts,o.publicKey=l})}}),{setAccount:_n,setAddress:d_,setAccountNonce:ln,setAccountShard:__,setLedgerAccount:tc,updateLedgerAccount:rc,setWalletConnectAccount:l_,setIsAccountLoading:g_,setAccountLoadingError:u_,setWebsocketEvent:ac}=qt.actions,io=qt.reducer});function Ue(){return new Date().setHours(new Date().getHours()+24)}function We(e){le.local.setItem({key:ae.loginExpiresAt,data:e,expires:e})}var gn=f(()=>{"use strict";n();ve();we()});var Xt,Kt,Yt,k_,Zt,ic,cc,I_,sc,Jt,A_,so,po=f(()=>{"use strict";n();Xt=require("@reduxjs/toolkit");gn();A();N();Kt={loginMethod:"",walletConnectLogin:null,ledgerLogin:null,tokenLogin:null,walletLogin:null,extensionLogin:null,operaLogin:null,isLoginSessionInvalid:!1},Yt=(0,Xt.createSlice)({name:"loginInfoSlice",initialState:Kt,reducers:{setLoginMethod:(e,o)=>{e.loginMethod=o.payload},setTokenLogin:(e,o)=>{e.tokenLogin=o.payload},setTokenLoginSignature:(e,o)=>{(e==null?void 0:e.tokenLogin)!=null&&(e.tokenLogin.signature=o.payload)},setWalletLogin:(e,o)=>{e.walletLogin=o.payload},setWalletConnectLogin:(e,o)=>{e.walletConnectLogin=o.payload},setLedgerLogin:(e,o)=>{e.ledgerLogin=o.payload},invalidateLoginSession:e=>{e.isLoginSessionInvalid=!0},setLogoutRoute:(e,o)=>{e.logoutRoute=o.payload}},extraReducers:e=>{e.addCase(P,()=>Kt),e.addCase(Y,(o,t)=>{o.loginMethod=t.payload.loginMethod,We(Ue())})}}),{setLoginMethod:k_,setWalletConnectLogin:Zt,setLedgerLogin:ic,setTokenLogin:cc,setTokenLoginSignature:I_,setWalletLogin:sc,invalidateLoginSession:Jt,setLogoutRoute:A_}=Yt.actions,so=Yt.reducer});var er,Qt,or,P_,pc,N_,mc,mo,_o=f(()=>{"use strict";n();er=require("@reduxjs/toolkit");N();Qt={},or=(0,er.createSlice)({name:"modalsSlice",initialState:Qt,reducers:{setTxSubmittedModal:(e,o)=>{e.txSubmittedModal=o.payload},setNotificationModal:(e,o)=>{e.notificationModal=o.payload},clearTxSubmittedModal:e=>{e.txSubmittedModal=void 0},clearNotificationModal:e=>{e.notificationModal=void 0}},extraReducers:e=>{e.addCase(P,()=>Qt)}}),{setTxSubmittedModal:P_,setNotificationModal:pc,clearTxSubmittedModal:N_,clearNotificationModal:mc}=or.actions,mo=or.reducer});var Ge=f(()=>{"use strict";n()});var tr=f(()=>{"use strict";n();Ge()});function un(e){return e[Math.floor(Math.random()*e.length)]}var rr=f(()=>{"use strict";n()});var fn=f(()=>{"use strict";n();nn()});var ge=f(()=>{"use strict";n();tr();rr();fn()});var ar,ir,hn,dc,cr,q_,K_,lo,go=f(()=>{"use strict";n();ar=require("@reduxjs/toolkit"),ir=h(require("lodash.omit"));ge();hn={id:"not-configured",chainId:"",name:"NOT CONFIGURED",egldLabel:"",decimals:"18",digits:"4",gasPerDataByte:"1500",walletConnectDeepLink:"",walletConnectBridgeAddress:"",walletConnectV2RelayAddress:"",walletConnectV2ProjectId:"",walletConnectV2Options:{},walletAddress:"",apiAddress:"",explorerAddress:"",apiTimeout:"4000"},dc={network:hn,chainID:"-1"},cr=(0,ar.createSlice)({name:"appConfig",initialState:dc,reducers:{initializeNetworkConfig:(e,o)=>{let t=un(o.payload.walletConnectBridgeAddresses),r=un(o.payload.walletConnectV2RelayAddresses),s=(0,ir.default)(o.payload,"walletConnectBridgeAddresses","walletConnectV2RelayAddresses");e.network=w(u(u({},e.network),s),{walletConnectBridgeAddress:t,walletConnectV2RelayAddress:r})},setChainID:(e,o)=>{e.chainID=o.payload}}}),{initializeNetworkConfig:q_,setChainID:K_}=cr.actions,lo=cr.reducer});var sr,Tn,pr,el,ol,nl,uo,fo=f(()=>{"use strict";n();sr=require("@reduxjs/toolkit");C();N();Tn={isSigning:!1,signedSessions:{}},pr=(0,sr.createSlice)({name:"signedMessageInfoSliceState",initialState:Tn,reducers:{setSignSession:(e,o)=>{let{sessionId:t,signedSession:r,errorMessage:s}=o.payload;s&&(e.errorMessage=s),e.isSigning=r.status==="pending",e.signedSessions[t]=r},setSignSessionState:(e,o)=>u(u({},e),o.payload),clearSignedMessageInfo:()=>Tn},extraReducers:e=>{e.addCase(P,()=>Tn)}}),{setSignSession:el,clearSignedMessageInfo:ol,setSignSessionState:nl}=pr.actions,uo=pr.reducer});var dr,_r,mr,lr,_c,xn,ml,dl,_l,ll,ho,To=f(()=>{"use strict";n();dr=require("@reduxjs/toolkit"),_r=require("redux-persist");tn();C();ro();N();mr={customToasts:[],transactionToasts:[],failTransactionToast:null},lr=(0,dr.createSlice)({name:"toastsSlice",initialState:mr,reducers:{addCustomToast:(e,o)=>{e.customToasts.push(w(u({},o.payload),{type:"custom",toastId:o.payload.toastId||`custom-toast-${e.customToasts.length+1}`}))},removeCustomToast:(e,o)=>{e.customToasts=e.customToasts.filter(t=>t.toastId!==o.payload)},addTransactionToast:(e,o)=>{e.transactionToasts.push({type:"transaction",startTimestamp:Be(),toastId:o.payload||`custom-toast-${e.transactionToasts.length+1}`})},removeTransactionToast:(e,o)=>{e.transactionToasts=e.transactionToasts.filter(t=>t.toastId!==o.payload)},addFailTransactionToast:(e,o)=>{e.failTransactionToast=w(u({},o.payload),{toastId:oo})},removeFailTransactionToast:e=>{e.failTransactionToast=null}},extraReducers:e=>{e.addCase(P,()=>mr),e.addCase(_r.REHYDRATE,(o,t)=>{var s,p;let r=(p=(s=t.customToasts)==null?void 0:s.filter(m=>!("component"in m)))!=null?p:[];o.customToasts=r})}}),{addCustomToast:_c,removeCustomToast:xn,addTransactionToast:ml,removeTransactionToast:dl,addFailTransactionToast:_l,removeFailTransactionToast:ll}=lr.actions,ho=lr.reducer});var gr,bn,Sn,wn,lc,yn,ur,hl,gc,vn,xo,yo=f(()=>{"use strict";n();gr=require("@reduxjs/toolkit");N();bn="Transaction failed",Sn="Transaction successful",wn="Processing transaction",lc="Transaction submitted",yn={},ur=(0,gr.createSlice)({name:"transactionsInfo",initialState:yn,reducers:{setTransactionsDisplayInfo(e,o){let{sessionId:t,transactionsDisplayInfo:r}=o.payload;t!=null&&(e[t]={errorMessage:(r==null?void 0:r.errorMessage)||bn,successMessage:(r==null?void 0:r.successMessage)||Sn,processingMessage:(r==null?void 0:r.processingMessage)||wn,submittedMessage:(r==null?void 0:r.submittedMessage)||lc,transactionDuration:r==null?void 0:r.transactionDuration})},clearTransactionsInfoForSessionId(e,o){o.payload!=null&&delete e[o.payload]},clearTransactionsInfo:()=>yn},extraReducers:e=>{e.addCase(P,()=>yn)}}),{clearTransactionsInfo:hl,setTransactionsDisplayInfo:gc,clearTransactionsInfoForSessionId:vn}=ur.actions,xo=ur.reducer});function Ie(e){return e!=null&&(Sc(e)||Ic(e))}function Ae(e){return e!=null&&(wc(e)||Ac(e))}function Le(e){return e!=null&&(vc(e)||Lc(e))}function kn(e){return e!=null&&kc(e)}function Sc(e){return e!=null&&uc.includes(e)}function wc(e){return e!=null&&fc.includes(e)}function vc(e){return e!=null&&hc.includes(e)}function kc(e){return e!=null&&Tc.includes(e)}function Ic(e){return e!=null&&xc.includes(e)}function Ac(e){return e!=null&&yc.includes(e)}function Lc(e){return e!=null&&bc.includes(e)}var uc,fc,hc,Tc,xc,yc,bc,Ee=f(()=>{"use strict";n();A();uc=["sent"],fc=["success"],hc=["fail","cancelled","timedOut"],Tc=["timedOut"],xc=["pending"],yc=["success"],bc=["fail","invalid"]});var fr,hr,$e,Ec,Tr,xr,yr,Cc,bo,Pc,Nc,Il,Dc,So,In,wo,vo=f(()=>{"use strict";n();fr=require("@reduxjs/toolkit"),hr=require("redux-persist");A();Ee();N();$e={signedTransactions:{},transactionsToSign:null,signTransactionsError:null,signTransactionsCancelMessage:null,customTransactionInformationForSessionId:{}},Ec={signWithoutSending:!1,sessionInformation:null,redirectAfterSign:!1},Tr=(0,fr.createSlice)({name:"transactionsSlice",initialState:$e,reducers:{moveTransactionsToSignedState:(e,o)=>{var _,T;let{sessionId:t,transactions:r,errorMessage:s,status:p,redirectRoute:m}=o.payload,l=((_=e.customTransactionInformationForSessionId)==null?void 0:_[t])||Ec;e.signedTransactions[t]={transactions:r,status:p,errorMessage:s,redirectRoute:m,customTransactionInformation:l},((T=e==null?void 0:e.transactionsToSign)==null?void 0:T.sessionId)===t&&(e.transactionsToSign=$e.transactionsToSign)},clearSignedTransaction:(e,o)=>{e.signedTransactions[o.payload]&&delete e.signedTransactions[o.payload]},clearTransactionToSign:e=>{e!=null&&e.transactionsToSign&&(e.transactionsToSign=null)},updateSignedTransaction:(e,o)=>{e.signedTransactions=u(u({},e.signedTransactions),o.payload)},updateSignedTransactions:(e,o)=>{let{sessionId:t,status:r,errorMessage:s,transactions:p}=o.payload;e.signedTransactions[t]!=null&&(e.signedTransactions[t].status=r,s!=null&&(e.signedTransactions[t].errorMessage=s),p!=null&&(e.signedTransactions[t].transactions=p))},updateSignedTransactionStatus:(e,o)=>{var l,_,T,b,x,D;let{sessionId:t,status:r,errorMessage:s,transactionHash:p}=o.payload,m=(_=(l=e.signedTransactions)==null?void 0:l[t])==null?void 0:_.transactions;if(m!=null){e.signedTransactions[t].transactions=m.map(Z=>Z.hash===p?w(u({},Z),{status:r,errorMessage:s}):Z);let R=(b=(T=e.signedTransactions[t])==null?void 0:T.transactions)==null?void 0:b.every(Z=>Ae(Z.status)),re=(D=(x=e.signedTransactions[t])==null?void 0:x.transactions)==null?void 0:D.every(Z=>Le(Z.status));R&&(e.signedTransactions[t].status="success"),re&&(e.signedTransactions[t].status="fail")}},setTransactionsToSign:(e,o)=>{e.transactionsToSign=o.payload;let{sessionId:t,customTransactionInformation:r}=o.payload;e.customTransactionInformationForSessionId[t]=r,e.signTransactionsError=null},clearAllTransactionsToSign:e=>{e.transactionsToSign=$e.transactionsToSign,e.signTransactionsError=null},clearAllSignedTransactions:e=>{e.signedTransactions=$e.signedTransactions},setSignTransactionsError:(e,o)=>{e.signTransactionsError=o.payload},setSignTransactionsCancelMessage:(e,o)=>{e.signTransactionsCancelMessage=o.payload}},extraReducers:e=>{e.addCase(P,()=>$e),e.addCase(hr.REHYDRATE,(o,t)=>{var m;if(o.signTransactionsCancelMessage=null,!((m=t.payload)!=null&&m.transactions))return;let{signedTransactions:r,customTransactionInformationForSessionId:s}=t.payload.transactions,p=Object.entries(r).reduce((l,[_,T])=>{let b=new Date(_),x=new Date;return x.setHours(x.getHours()+5),x-b>0||(l[_]=T),l},{});s!=null&&(o.customTransactionInformationForSessionId=s),r!=null&&(o.signedTransactions=p)})}}),{updateSignedTransactionStatus:xr,updateSignedTransactions:yr,setTransactionsToSign:Cc,clearAllTransactionsToSign:bo,clearAllSignedTransactions:Pc,clearSignedTransaction:Nc,clearTransactionToSign:Il,setSignTransactionsError:Dc,setSignTransactionsCancelMessage:So,moveTransactionsToSignedState:In}=Tr.actions,wo=Tr.reducer});var I=f(()=>{"use strict";n();co();po();_o();go();fo();To();yo();vo()});var An=f(()=>{"use strict";n()});var kr,vr,Ir,Yl,Zl,Jl,Ql,eg,og,Ar,Lr=f(()=>{"use strict";n();kr=require("@reduxjs/toolkit");N();vr={visible:!1,modal:{showHeader:!0,showFooter:!1,headerText:"",footerText:""}},Ir=(0,kr.createSlice)({name:"dappModalsSlice",initialState:vr,reducers:{setDappModalVisibility:(e,o)=>{e.visible=o.payload},setDappModalHeaderVisibility:(e,o)=>{e.modal.showHeader=o.payload},setDappModalFooterVisibility:(e,o)=>{e.modal.showFooter=o.payload},setDappModalHeaderText:(e,o)=>{e.modal.headerText=o.payload},setDappModalFooterText:(e,o)=>{e.modal.footerText=o.payload},setDappModalConfig:(e,o)=>{e.modal=u({},o.payload)}},extraReducers:e=>{e.addCase(P,()=>vr)}}),{setDappModalVisibility:Yl,setDappModalHeaderVisibility:Zl,setDappModalFooterVisibility:Jl,setDappModalHeaderText:Ql,setDappModalFooterText:eg,setDappModalConfig:og}=Ir.actions,Ar=Ir.reducer});var Er,Fc,Bc,fe,Lo=f(()=>{"use strict";n();Er=require("@reduxjs/toolkit");An();co();Lr();po();_o();go();fo();To();yo();vo();Fc={["account"]:io,["dappModal"]:Ar,["loginInfo"]:so,["modals"]:mo,["networkConfig"]:lo,["signedMessageInfo"]:uo,["toasts"]:ho,["transactionsInfo"]:xo,["transactions"]:wo},Bc=(e={})=>(0,Er.combineReducers)(u(u({},Fc),e)),fe=Bc});var Dr={};Te(Dr,{default:()=>Qc,sessionStorageReducers:()=>Ln});function ie(e,o=[]){return{key:e,version:1,storage:Pr.default,blacklist:o}}var W,Cr,Pr,Uc,Ve,Wc,Gc,$c,Hc,Vc,jc,zc,qc,Kc,Nr,Xc,Ln,Yc,Zc,Jc,Qc,Rr=f(()=>{"use strict";n();W=require("redux-persist"),Cr=h(require("redux-persist/lib/storage")),Pr=h(require("redux-persist/lib/storage/session"));Lo();I();co();po();_o();go();fo();To();yo();vo();An();Uc={persistReducersStorageType:"localStorage"},Ve={["account"]:"sdk-dapp-account",["loginInfo"]:"sdk-dapp-login-info",["modals"]:"sdk-dapp-modals",["networkConfig"]:"sdk-dapp-network-config"},Wc=ie(Ve["account"]),Gc=ie(Ve["loginInfo"]),$c=ie(Ve["modals"]),Hc=ie(Ve["networkConfig"]),Vc={2:e=>w(u({},e),{networkConfig:hn})};jc=ie("sdk-dapp-transactionsInfo"),zc=ie("sdk-dapp-transactions",["transactionsToSign"]),qc=ie("sdk-dapp-toasts"),Kc=ie("sdk-dapp-signedMessageInfo"),Nr={key:"sdk-dapp-store",version:2,storage:Cr.default,whitelist:Object.keys(Ve),migrate:(0,W.createMigrate)(Vc,{debug:!1})},Xc=w(u({},Nr),{whitelist:[]}),Ln={["toasts"]:(0,W.persistReducer)(qc,ho),["transactions"]:(0,W.persistReducer)(zc,wo),["transactionsInfo"]:(0,W.persistReducer)(jc,xo),["signedMessageInfo"]:(0,W.persistReducer)(Kc,uo)},Yc=w(u({},Ln),{["account"]:(0,W.persistReducer)(Wc,io),["loginInfo"]:(0,W.persistReducer)(Gc,so),["modals"]:(0,W.persistReducer)($c,mo),["networkConfig"]:(0,W.persistReducer)(Hc,lo)}),Zc=Uc.persistReducersStorageType==="localStorage",Jc=Zc?(0,W.persistReducer)(Nr,fe(Ln)):(0,W.persistReducer)(Xc,fe(Yc)),Qc=Jc});var Mr={};Te(Mr,{default:()=>es});var es,Or=f(()=>{"use strict";n();Lo();es=fe()});var Fr={};Te(Fr,{default:()=>ns});var H,os,ns,Br=f(()=>{"use strict";n();H=require("redux-persist"),os=[H.FLUSH,H.REHYDRATE,H.PAUSE,H.PERSIST,H.PURGE,H.REGISTER],ns=os});var Gr={};Te(Gr,{default:()=>Wr});function Wr(e){return(0,Ur.persistStore)(e)}var Ur,$r=f(()=>{"use strict";n();Ur=require("redux-persist")});var Ba=B((oA,Fa)=>{n();var Xp=typeof a=="object"&&a&&a.Object===Object&&a;Fa.exports=Xp});var Wa=B((nA,Ua)=>{n();var Yp=Ba(),Zp=typeof self=="object"&&self&&self.Object===Object&&self,Jp=Yp||Zp||Function("return this")();Ua.exports=Jp});var Vn=B((tA,Ga)=>{n();var Qp=Wa(),em=Qp.Symbol;Ga.exports=em});var ja=B((rA,Va)=>{n();var $a=Vn(),Ha=Object.prototype,om=Ha.hasOwnProperty,nm=Ha.toString,Ke=$a?$a.toStringTag:void 0;function tm(e){var o=om.call(e,Ke),t=e[Ke];try{e[Ke]=void 0;var r=!0}catch(p){}var s=nm.call(e);return r&&(o?e[Ke]=t:delete e[Ke]),s}Va.exports=tm});var qa=B((aA,za)=>{n();var rm=Object.prototype,am=rm.toString;function im(e){return am.call(e)}za.exports=im});var Za=B((iA,Ya)=>{n();var Ka=Vn(),cm=ja(),sm=qa(),pm="[object Null]",mm="[object Undefined]",Xa=Ka?Ka.toStringTag:void 0;function dm(e){return e==null?e===void 0?mm:pm:Xa&&Xa in Object(e)?cm(e):sm(e)}Ya.exports=dm});var Qa=B((cA,Ja)=>{n();var _m=Array.isArray;Ja.exports=_m});var oi=B((sA,ei)=>{n();function lm(e){return e!=null&&typeof e=="object"}ei.exports=lm});var ti=B((pA,ni)=>{n();var gm=Za(),um=Qa(),fm=oi(),hm="[object String]";function Tm(e){return typeof e=="string"||!um(e)&&fm(e)&&gm(e)==hm}ni.exports=Tm});var Km={};module.exports=Oe(Km);n();n();var jm=h(require("react")),zm=require("@fortawesome/free-solid-svg-icons"),qm=require("@fortawesome/react-fontawesome");n();n();var yt=h(require("react")),bt=require("@fortawesome/free-solid-svg-icons"),Yi=require("@fortawesome/react-fontawesome"),Zi=h(require("classnames"));n();var Xi=`.dapp-core-component__copyButton-styles__copy {
  color: #6c757d;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Xi));n();n();n();var cp=h(require("react")),sp=require("@fortawesome/free-solid-svg-icons"),pp=require("@fortawesome/react-fontawesome"),mp=h(require("classnames"));n();var Ji=`.dapp-core-component__main__link-style {
  color: #1392ff;
}

.dapp-core-component__main__link-second-style {
  color: #9dabbd;
}
.dapp-core-component__main__link-second-style:hover {
  color: #1392ff;
}
.dapp-core-component__main__link-second-style:hover svg {
  color: #1392ff;
}

.dapp-core-component__main__wrapper {
  min-height: 100vh;
}

.dapp-core-component__main__opacity-6 {
  opacity: 0.6;
}

.dapp-core-component__main__qr-code-svg-container {
  width: 15rem;
  height: 15rem;
}

.dapp-core-component__main__pair-list {
  max-height: 20rem;
  overflow-y: auto;
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__pair-loader,
.dapp-core-component__main__pair-image {
  width: 4rem;
  overflow: hidden;
  height: auto;
  min-width: 2rem;
  margin: 0;
  padding: 0;
}

.dapp-core-component__main__pair-remove {
  padding: 0 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

.dapp-core-component__main__dapp-icon {
  padding: 5px;
  border-radius: 50%;
  background-color: #fafafa;
  width: 4.6428571429rem;
  height: 4.6428571429rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
}
.dapp-core-component__main__dapp-icon.dapp-core-component__main__icon-medium {
  width: 80px;
  height: 80px;
}

.dapp-core-component__main__centering {
  display: flex;
  justify-content: center;
  align-items: center;
}

.dapp-core-component__main__border-n {
  border: none;
}

.dapp-core-component__main__t-shadow {
  text-shadow: 0px 4px 7px rgba(0, 0, 0, 0.3);
}

.dapp-core-component__main__btn {
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.04);
}

.dapp-core-component__main__media {
  display: flex;
  align-items: flex-start;
}

.dapp-core-component__main__media-body {
  flex: 1;
}

.dapp-core-component__main__connect-btns .dapp-core-component__main__btn {
  display: flex;
  height: 48px;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: #1392ff;
  padding: 15px 20px 15px 16px;
  border-radius: 6px;
  gap: 8px;
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__btn:hover, .dapp-core-component__main__connect-btns .dapp-core-component__main__btn:active {
  background-color: #1392ff;
  background-image: linear-gradient(rgba(0, 0, 0, 0.1) 0 0);
}
.dapp-core-component__main__connect-btns .dapp-core-component__main__info {
  color: #1392ff;
  margin-left: 48px;
}

.dapp-core-component__main__btn {
  transition: none;
}

.dapp-core-component__main__trust-badge {
  width: 20px;
  height: 20px;
}

/* Navbar
  -------------------------------------------------- */
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name {
  display: flex;
  align-items: center;
  font-size: 1.7142857143rem;
  font-weight: 400;
  color: #9ba5b4;
  margin-left: 1.6428571429rem;
}
.dapp-core-component__main__navbar .dapp-core-component__main__dapp-name:before {
  content: "";
  display: block;
  width: 1px;
  height: 3rem;
  margin-right: 1.7142857143rem;
}

/* Dashboard
  -------------------------------------------------- */
.dapp-core-component__main__action-btn {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;
  margin: 0 7px;
}
.dapp-core-component__main__action-btn button {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  background-color: #fff;
}
.dapp-core-component__main__action-btn button:hover {
  opacity: 0.9;
}

.dapp-core-component__main__transactions .dapp-core-component__main__transaction-icon {
  border-radius: 50%;
  width: 40px;
  height: 40px;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table td {
  white-space: nowrap;
  vertical-align: middle;
}
.dapp-core-component__main__transactions.dapp-core-component__main__table tr:last-of-type td {
  border-bottom: 1px solid #dee2e6;
}

/* Loader, Transaction Success/Fail
  -------------------------------------------------- */
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div {
  position: absolute;
  top: 27px;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  background: rgba(0, 0, 0, 0.1);
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
}
.dapp-core-component__main__page-state .dapp-core-component__main__bg-blue .dapp-core-component__main__lds-ellipsis div {
  background: rgba(255, 255, 255, 0.4);
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(1) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis1 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(2) {
  left: 6px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(3) {
  left: 26px;
  animation: dapp-core-component__main__lds-ellipsis2 0.6s infinite;
}
.dapp-core-component__main__page-state .dapp-core-component__main__lds-ellipsis div:nth-child(4) {
  left: 45px;
  animation: dapp-core-component__main__lds-ellipsis3 0.6s infinite;
}
@keyframes dapp-core-component__main__lds-ellipsis1 {
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis3 {
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
}
@keyframes dapp-core-component__main__lds-ellipsis2 {
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(19px, 0);
  }
}

.dapp-core-component__main__btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: rgba(0, 0, 0, 0);
  border: 1px solid rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  cursor: pointer;
}
.dapp-core-component__main__btn .dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn .dapp-core-component__main__disabled {
  opacity: 0.65;
  cursor: default;
}
.dapp-core-component__main__btn:hover {
  color: #212529;
  text-decoration: none;
}
.dapp-core-component__main__btn:focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(27, 70, 194, 0.25);
}
.dapp-core-component__main__btn:disabled {
  opacity: 0.65;
  cursor: default;
}

.dapp-core-component__main__btn-primary {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-primary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #1b46c2;
  border-color: #007bff;
}
.dapp-core-component__main__btn-primary:hover {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-primary:focus {
  color: #fff;
  background-color: #1b46c2;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}
.dapp-core-component__main__btn-secondary .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #6c757d;
  border-color: #007bff;
}
.dapp-core-component__main__btn-secondary:hover {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
}
.dapp-core-component__main__btn-secondary:focus {
  color: #fff;
  background-color: #6c757d;
  border-color: #0062cc;
  box-shadow: 0 0 0 0.2rem rgba(38, 143, 255, 0.5);
}

.dapp-core-component__main__btn-warning {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #ffc107;
  border-color: #ffc107;
}
.dapp-core-component__main__btn-warning:hover {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}
.dapp-core-component__main__btn-warning:focus {
  color: #212529;
  background-color: #e0a800;
  border-color: #d39e00;
  box-shadow: 0 0 0 0.2rem rgba(222, 170, 12, 0.5);
}

.dapp-core-component__main__btn-dark {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}
.dapp-core-component__main__btn-dark .dapp-core-component__main__disabled {
  color: #fff;
  background-color: #343a40;
  border-color: #343a40;
}
.dapp-core-component__main__btn-dark:hover {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
}
.dapp-core-component__main__btn-dark:focus {
  color: #fff;
  background-color: #23272b;
  border-color: #1d2124;
  box-shadow: 0 0 0 0.2rem rgba(82, 88, 93, 0.5);
}

.dapp-core-component__main__btn-link {
  font-weight: 400;
  color: #007bff;
  text-decoration: none;
}
.dapp-core-component__main__btn-link:hover {
  color: #0056b3;
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:focus, .dapp-core-component__main__btn-link.dapp-core-component__main__focus {
  text-decoration: underline;
}
.dapp-core-component__main__btn-link:disabled, .dapp-core-component__main__btn-link.dapp-core-component__main__disabled {
  color: #6c757d;
  pointer-events: none;
}

.dapp-core-component__main__btn-light {
  color: #212529;
  background-color: #fafafa;
  border-color: #fafafa;
}
.dapp-core-component__main__btn-light .dapp-core-component__main__focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}
.dapp-core-component__main__btn-light .dapp-core-component__main__disabled {
  color: #212529;
  background-color: #f8f9fa;
  border-color: #f8f9fa;
}
.dapp-core-component__main__btn-light:hover {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
}
.dapp-core-component__main__btn-light:focus {
  color: #212529;
  background-color: #e2e6ea;
  border-color: #dae0e5;
  box-shadow: 0 0 0 0.2rem rgba(216, 217, 219, 0.5);
}

.dapp-core-component__main__badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out, border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

@media (prefers-reduced-motion: reduce) {
  .dapp-core-component__main__badge {
    transition: none;
  }
}
a.dapp-core-component__main__badge:hover, a.dapp-core-component__main__badge:focus {
  text-decoration: none;
}

.dapp-core-component__main__badge:empty {
  display: none;
}

.dapp-core-component__main__btn .dapp-core-component__main__badge {
  position: relative;
  top: -1px;
}

.dapp-core-component__main__badge-pill {
  padding-right: 0.6em;
  padding-left: 0.6em;
  border-radius: 10rem;
}

.dapp-core-component__main__badge-primary {
  color: #fff;
  background-color: #007bff;
}

a.dapp-core-component__main__badge-primary:hover, a.dapp-core-component__main__badge-primary:focus {
  color: #fff;
  background-color: #0062cc;
}

a.dapp-core-component__main__badge-primary:focus, a.dapp-core-component__main__badge-primary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.5);
}

.dapp-core-component__main__badge-secondary {
  color: #fff;
  background-color: #6c757d;
}

a.dapp-core-component__main__badge-secondary:hover, a.dapp-core-component__main__badge-secondary:focus {
  color: #fff;
  background-color: #545b62;
}

a.dapp-core-component__main__badge-secondary:focus, a.dapp-core-component__main__badge-secondary.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(108, 117, 125, 0.5);
}

.dapp-core-component__main__badge-success {
  color: #fff;
  background-color: #28a745;
}

a.dapp-core-component__main__badge-success:hover, a.dapp-core-component__main__badge-success:focus {
  color: #fff;
  background-color: #1e7e34;
}

a.dapp-core-component__main__badge-success:focus, a.dapp-core-component__main__badge-success.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.5);
}

.dapp-core-component__main__badge-info {
  color: #fff;
  background-color: #17a2b8;
}

a.dapp-core-component__main__badge-info:hover, a.dapp-core-component__main__badge-info:focus {
  color: #fff;
  background-color: #117a8b;
}

a.dapp-core-component__main__badge-info:focus, a.dapp-core-component__main__badge-info.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(23, 162, 184, 0.5);
}

.dapp-core-component__main__badge-warning {
  color: #212529;
  background-color: #ffc107;
}

a.dapp-core-component__main__badge-warning:hover, a.dapp-core-component__main__badge-warning:focus {
  color: #212529;
  background-color: #d39e00;
}

a.dapp-core-component__main__badge-warning:focus, a.dapp-core-component__main__badge-warning.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(255, 193, 7, 0.5);
}

.dapp-core-component__main__badge-danger {
  color: #fff;
  background-color: #dc3545;
}

a.dapp-core-component__main__badge-danger:hover, a.dapp-core-component__main__badge-danger:focus {
  color: #fff;
  background-color: #bd2130;
}

a.dapp-core-component__main__badge-danger:focus, a.dapp-core-component__main__badge-danger.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(220, 53, 69, 0.5);
}

.dapp-core-component__main__badge-light {
  color: #212529;
  background-color: #f8f9fa;
}

a.dapp-core-component__main__badge-light:hover, a.dapp-core-component__main__badge-light:focus {
  color: #212529;
  background-color: #dae0e5;
}

a.dapp-core-component__main__badge-light:focus, a.dapp-core-component__main__badge-light.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(248, 249, 250, 0.5);
}

.dapp-core-component__main__badge-dark {
  color: #fff;
  background-color: #343a40;
}

a.dapp-core-component__main__badge-dark:hover, a.dapp-core-component__main__badge-dark:focus {
  color: #fff;
  background-color: #1d2124;
}

a.dapp-core-component__main__badge-dark:focus, a.dapp-core-component__main__badge-dark.dapp-core-component__main__focus {
  outline: 0;
  box-shadow: 0 0 0 0.2rem rgba(52, 58, 64, 0.5);
}

.dapp-core-component__main__card {
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 5px;
}
.dapp-core-component__main__card > hr {
  margin-right: 0;
  margin-left: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group {
  border-top: inherit;
  border-bottom: inherit;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:first-child {
  border-top-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__list-group:last-child {
  border-bottom-width: 0;
}
.dapp-core-component__main__card > .dapp-core-component__main__card-header + .dapp-core-component__main__list-group,
.dapp-core-component__main__card > .dapp-core-component__main__list-group + .dapp-core-component__main__card-footer {
  border-top: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header {
  border: 0;
  padding: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header:first-child {
  border-radius: calc(0.25rem - 1px) calc(0.25rem - 1px) 0 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item {
  border-bottom: 1px solid #dee2e6;
  padding: 1.1rem 1rem;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item h6,
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item .dapp-core-component__main__h6 {
  line-height: 2.5rem;
  letter-spacing: unset;
  margin: 0;
}
.dapp-core-component__main__card .dapp-core-component__main__card-header .dapp-core-component__main__card-header-item.dapp-core-component__main__compact {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  background-color: #fafafa;
}

.dapp-core-component__main__card-body {
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
}

.dapp-core-component__main__card-title {
  margin-bottom: 0.75rem;
}

.dapp-core-component__main__card-subtitle {
  margin-top: -0.375rem;
  margin-bottom: 0;
}

.dapp-core-component__main__card-text:last-child {
  margin-bottom: 0;
}

.dapp-core-component__main__card-link:hover {
  text-decoration: none;
}
.dapp-core-component__main__card-link + .dapp-core-component__main__card-link {
  margin-left: 1.25rem;
}

.dapp-core-component__main__card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.03);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__card-footer {
  padding: 0.75rem 1.25rem;
  background-color: rgba(0, 0, 0, 0.03);
  border-top: 1px solid rgba(0, 0, 0, 0.125);
}

.dapp-core-component__main__text {
  color: #9dabbd;
}

.dapp-core-component__main__text-white {
  color: #fff;
}

.dapp-core-component__main__text-monospace {
  font-family: SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-size-xl .dapp-core-component__main__trim {
  max-width: 13rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim {
  max-width: 10rem;
}

.dapp-core-component__main__table .dapp-core-component__main__trim-only-sm .dapp-core-component__main__trim {
  max-width: none;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-left {
  text-align: left;
}

.dapp-core-component__main__text-right {
  text-align: right;
}

.dapp-core-component__main__text-center {
  text-align: center;
}

.dapp-core-component__main__text-primary {
  color: #1b46c2;
}

.dapp-core-component__main__text-secondary {
  color: #6c757d;
}

.dapp-core-component__main__text-primary-highlight {
  color: #1b46c2;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-success {
  color: #28a745;
}

.dapp-core-component__main__text-justify {
  text-align: justify;
}

.dapp-core-component__main__text-wrap {
  white-space: normal;
}

.dapp-core-component__main__text-nowrap {
  white-space: nowrap;
}

.dapp-core-component__main__text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dapp-core-component__main__text-body {
  color: #212529;
}

.dapp-core-component__main__text-muted {
  color: #6c757d;
}

.dapp-core-component__main__text-black-50 {
  color: rgba(0, 0, 0, 0.5);
}

.dapp-core-component__main__text-white-50 {
  color: rgba(255, 255, 255, 0.5);
}

@media (min-width: 992px) {
  .dapp-core-component__main__text-lg-right {
    text-align: right;
  }
}

.dapp-core-component__main__text-hide {
  font: 0/0 a;
  color: rgba(0, 0, 0, 0);
  text-shadow: none;
  background-color: rgba(0, 0, 0, 0);
  border: 0;
}

.dapp-core-component__main__text-decoration-none {
  text-decoration: none;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__text-reset {
  color: inherit;
}

.dapp-core-component__main__text-lowercase {
  text-transform: lowercase;
}

.dapp-core-component__main__text-uppercase {
  text-transform: uppercase;
}

.dapp-core-component__main__text-capitalize {
  text-transform: capitalize;
}

.dapp-core-component__main__text-danger {
  color: #d9534f;
}

.dapp-core-component__main__text-warning {
  color: #f0ad4e;
}

.dapp-core-component__main__trim-fs-sm {
  color: inherit;
}

.dapp-core-component__main__text-break {
  word-break: break-word;
  word-wrap: break-word;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__m-0 {
  margin: 0;
}

.dapp-core-component__main__mt-0 {
  margin-top: 0;
}

.dapp-core-component__main__my-0 {
  margin-top: 0;
  margin-bottom: 0;
}

.dapp-core-component__main__mr-0 {
  margin-right: 0;
}

.dapp-core-component__main__mx-0 {
  margin-right: 0;
  margin-left: 0;
}

.dapp-core-component__main__mb-0 {
  margin-bottom: 0;
}

.dapp-core-component__main__my-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.dapp-core-component__main__ml-0 {
  margin-left: 0;
}

.dapp-core-component__main__mx-0 {
  margin-left: 0;
  margin-right: 0;
}

.dapp-core-component__main__m-1 {
  margin: 0.25rem;
}

.dapp-core-component__main__mt-1 {
  margin-top: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__mr-1 {
  margin-right: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.dapp-core-component__main__mb-1 {
  margin-bottom: 0.25rem;
}

.dapp-core-component__main__my-1 {
  margin-bottom: 0.25rem;
  margin-top: 0.25rem;
}

.dapp-core-component__main__ml-1 {
  margin-left: 0.25rem;
}

.dapp-core-component__main__mx-1 {
  margin-right: 0.25rem;
  margin-left: 0.25rem;
}

.dapp-core-component__main__m-2 {
  margin: 0.5rem;
}

.dapp-core-component__main__mt-2 {
  margin-top: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-bottom: 0.5rem;
  margin-top: 0.5rem;
}

.dapp-core-component__main__mr-2 {
  margin-right: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__mb-2 {
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__my-2 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}

.dapp-core-component__main__ml-2 {
  margin-left: 0.5rem;
}

.dapp-core-component__main__mx-2 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.dapp-core-component__main__m-3 {
  margin: 1rem;
}

.dapp-core-component__main__mt-3,
.dapp-core-component__main__my-3 {
  margin-top: 1rem;
}

.dapp-core-component__main__mr-3 {
  margin-right: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__mb-3 {
  margin-bottom: 1rem;
}

.dapp-core-component__main__my-3 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.dapp-core-component__main__ml-3 {
  margin-left: 1rem;
}

.dapp-core-component__main__mx-3 {
  margin-left: 1rem;
  margin-right: 1rem;
}

.dapp-core-component__main__m-4 {
  margin: 1.5rem;
}

.dapp-core-component__main__mt-4 {
  margin-top: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__mr-4 {
  margin-right: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-right: 1.5rem;
  margin-left: 1.5rem;
}

.dapp-core-component__main__mb-4 {
  margin-bottom: 1.5rem;
}

.dapp-core-component__main__my-4 {
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.dapp-core-component__main__ml-4 {
  margin-left: 1.5rem;
}

.dapp-core-component__main__mx-4 {
  margin-left: 1.5rem;
  margin-right: 1.5rem;
}

.dapp-core-component__main__m-5 {
  margin: 3rem;
}

.dapp-core-component__main__mt-5 {
  margin-top: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__mr-5 {
  margin-right: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-right: 3rem;
  margin-left: 3rem;
}

.dapp-core-component__main__mb-5 {
  margin-bottom: 3rem;
}

.dapp-core-component__main__my-5 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.dapp-core-component__main__ml-5 {
  margin-left: 3rem;
}

.dapp-core-component__main__mx-5 {
  margin-left: 3rem;
  margin-right: 3rem;
}

.dapp-core-component__main__p-0 {
  padding: 0;
}

.dapp-core-component__main__pt-0 {
  padding-top: 0;
}

.dapp-core-component__main__py-0 {
  padding-top: 0;
  padding-bottom: 0;
}

.dapp-core-component__main__pr-0 {
  padding-right: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__pb-0 {
  padding-bottom: 0;
}

.dapp-core-component__main__py-0 {
  padding-bottom: 0;
  padding-top: 0;
}

.dapp-core-component__main__pl-0 {
  padding-left: 0;
}

.dapp-core-component__main__px-0 {
  padding-left: 0;
  padding-right: 0;
}

.dapp-core-component__main__p-1 {
  padding: 0.25rem;
}

.dapp-core-component__main__pt-1 {
  padding-top: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pr-1 {
  padding-right: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-right: 0.25rem;
  padding-left: 0.25rem;
}

.dapp-core-component__main__pb-1 {
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__py-1 {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.dapp-core-component__main__pl-1 {
  padding-left: 0.25rem;
}

.dapp-core-component__main__px-1 {
  padding-left: 0.25rem;
  padding-right: 0.25rem;
}

.dapp-core-component__main__p-2 {
  padding: 0.5rem;
}

.dapp-core-component__main__pt-2 {
  padding-top: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
}

.dapp-core-component__main__pr-2 {
  padding-right: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__pb-2 {
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.dapp-core-component__main__pl-2 {
  padding-left: 0.5rem;
}

.dapp-core-component__main__px-2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.dapp-core-component__main__p-3 {
  padding: 1rem;
}

.dapp-core-component__main__pt-3 {
  padding-top: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.dapp-core-component__main__pr-3 {
  padding-right: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__pb-3 {
  padding-bottom: 1rem;
}

.dapp-core-component__main__py-3 {
  padding-bottom: 1rem;
  padding-top: 1rem;
}

.dapp-core-component__main__pl-3 {
  padding-left: 1rem;
}

.dapp-core-component__main__px-3 {
  padding-right: 1rem;
  padding-left: 1rem;
}

.dapp-core-component__main__p-4 {
  padding: 1.5rem;
}

.dapp-core-component__main__pt-4 {
  padding-top: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pr-4 {
  padding-right: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-left: 1.5rem;
  padding-right: 1.5rem;
}

.dapp-core-component__main__pb-4 {
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__py-4 {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
}

.dapp-core-component__main__pl-4 {
  padding-left: 1.5rem;
}

.dapp-core-component__main__px-4 {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
}

.dapp-core-component__main__p-5 {
  padding: 3rem;
}

.dapp-core-component__main__pt-5 {
  padding-top: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-bottom: 3rem;
  padding-top: 3rem;
}

.dapp-core-component__main__pr-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__pb-5 {
  padding-right: 3rem;
}

.dapp-core-component__main__py-5 {
  padding-top: 3rem;
  padding-bottom: 3rem;
}

.dapp-core-component__main__pl-5 {
  padding-left: 3rem;
}

.dapp-core-component__main__px-5 {
  padding-right: 3rem;
  padding-left: 3rem;
}

.dapp-core-component__main__m-n1 {
  margin: -0.25rem;
}

.dapp-core-component__main__mt-n1 {
  margin-top: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-top: -0.25rem;
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__mr-n1 {
  margin-right: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__mb-n1 {
  margin-bottom: -0.25rem;
}

.dapp-core-component__main__my-n1 {
  margin-bottom: -0.25rem;
  margin-top: -0.25rem;
}

.dapp-core-component__main__ml-n1 {
  margin-left: -0.25rem;
}

.dapp-core-component__main__mx-n1 {
  margin-left: -0.25rem;
  margin-right: -0.25rem;
}

.dapp-core-component__main__m-n2 {
  margin: -0.5rem;
}

.dapp-core-component__main__mt-n2 {
  margin-top: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-top: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mr-n2 {
  margin-right: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.dapp-core-component__main__mb-n2 {
  margin-bottom: -0.5rem;
}

.dapp-core-component__main__my-n2 {
  margin-bottom: -0.5rem;
  margin-top: -0.5rem;
}

.dapp-core-component__main__ml-n2 {
  margin-left: -0.5rem;
}

.dapp-core-component__main__mx-n2 {
  margin-left: -0.5rem;
  margin-right: -0.5rem;
}

.dapp-core-component__main__m-n3 {
  margin: -1rem;
}

.dapp-core-component__main__mt-n3 {
  margin-top: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-top: -1rem;
  margin-bottom: -1rem;
}

.dapp-core-component__main__mr-n3 {
  margin-right: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-right: -1rem;
  margin-left: -1rem;
}

.dapp-core-component__main__mb-n3 {
  margin-bottom: -1rem;
}

.dapp-core-component__main__my-n3 {
  margin-bottom: -1rem;
  margin-top: -1rem;
}

.dapp-core-component__main__ml-n3 {
  margin-left: -1rem;
}

.dapp-core-component__main__mx-n3 {
  margin-left: -1rem;
  margin-right: -1rem;
}

.dapp-core-component__main__m-n4 {
  margin: -1.5rem;
}

.dapp-core-component__main__mt-n4 {
  margin-top: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-top: -1.5rem;
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__mr-n4 {
  margin-right: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__mb-n4 {
  margin-bottom: -1.5rem;
}

.dapp-core-component__main__my-n4 {
  margin-bottom: -1.5rem;
  margin-top: -1.5rem;
}

.dapp-core-component__main__ml-n4 {
  margin-left: -1.5rem;
}

.dapp-core-component__main__mx-n4 {
  margin-left: -1.5rem;
  margin-right: -1.5rem;
}

.dapp-core-component__main__m-n5 {
  margin: -3rem;
}

.dapp-core-component__main__mt-n5 {
  margin-top: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-bottom: -3rem;
  margin-top: -3rem;
}

.dapp-core-component__main__mr-n5 {
  margin-right: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-left: -3rem;
  margin-right: -3rem;
}

.dapp-core-component__main__mb-n5 {
  margin-bottom: -3rem;
}

.dapp-core-component__main__my-n5 {
  margin-top: -3rem;
  margin-bottom: -3rem;
}

.dapp-core-component__main__ml-n5 {
  margin-left: -3rem;
}

.dapp-core-component__main__mx-n5 {
  margin-right: -3rem;
  margin-left: -3rem;
}

.dapp-core-component__main__m-auto {
  margin: auto;
}

.dapp-core-component__main__mt-auto {
  margin-top: auto;
}

.dapp-core-component__main__my-auto {
  margin-bottom: auto;
  margin-top: auto;
}

.dapp-core-component__main__mr-auto {
  margin-right: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

.dapp-core-component__main__mb-auto {
  margin-bottom: auto;
}

.dapp-core-component__main__my-auto {
  margin-top: auto;
  margin-bottom: auto;
}

.dapp-core-component__main__ml-auto {
  margin-left: auto;
}

.dapp-core-component__main__mx-auto {
  margin-left: auto;
  margin-right: auto;
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-1 {
    margin-right: 0.25rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mr-lg-4 {
    margin-right: 1.5rem;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__mx-lg-4 {
    margin-left: 1.5rem;
    margin-right: 1.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__mb-xl-2 {
    margin-bottom: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__pr-sm-2 {
    padding-right: 0.5rem;
  }
}

@media (min-width: 576px) {
  .dapp-core-component__main__px-sm-2 {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}

@media (min-width: 1200px) {
  .dapp-core-component__main__pr-xl-0,
  .dapp-core-component__main__px-xl-0 {
    padding-right: 0;
  }
}

.dapp-core-component__main__col {
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__row {
  display: flex;
  flex-wrap: wrap;
  margin-right: -15px;
  margin-left: -15px;
}

.dapp-core-component__main__col {
  flex-basis: 0;
  flex-grow: 1;
  max-width: 100%;
}

.dapp-core-component__main__col-xl,
.dapp-core-component__main__col-xl-auto,
.dapp-core-component__main__col-xl-12,
.dapp-core-component__main__col-xl-11,
.dapp-core-component__main__col-xl-10,
.dapp-core-component__main__col-xl-9,
.dapp-core-component__main__col-xl-8,
.dapp-core-component__main__col-xl-7,
.dapp-core-component__main__col-xl-6,
.dapp-core-component__main__col-xl-5,
.dapp-core-component__main__col-xl-4,
.dapp-core-component__main__col-xl-3,
.dapp-core-component__main__col-xl-2,
.dapp-core-component__main__col-xl-1,
.dapp-core-component__main__col-lg,
.dapp-core-component__main__col-lg-auto,
.dapp-core-component__main__col-lg-12,
.dapp-core-component__main__col-lg-11,
.dapp-core-component__main__col-lg-10,
.dapp-core-component__main__col-lg-9,
.dapp-core-component__main__col-lg-8,
.dapp-core-component__main__col-lg-7,
.dapp-core-component__main__col-lg-6,
.dapp-core-component__main__col-lg-5,
.dapp-core-component__main__col-lg-4,
.dapp-core-component__main__col-lg-3,
.dapp-core-component__main__col-lg-2,
.dapp-core-component__main__col-lg-1,
.dapp-core-component__main__col-md,
.dapp-core-component__main__col-md-auto,
.dapp-core-component__main__col-md-12,
.dapp-core-component__main__col-md-11,
.dapp-core-component__main__col-md-10,
.dapp-core-component__main__col-md-9,
.dapp-core-component__main__col-md-8,
.dapp-core-component__main__col-md-7,
.dapp-core-component__main__col-md-6,
.dapp-core-component__main__col-md-5,
.dapp-core-component__main__col-md-4,
.dapp-core-component__main__col-md-3,
.dapp-core-component__main__col-md-2,
.dapp-core-component__main__col-md-1,
.dapp-core-component__main__col-sm,
.dapp-core-component__main__col-sm-auto,
.dapp-core-component__main__col-sm-12,
.dapp-core-component__main__col-sm-11,
.dapp-core-component__main__col-sm-10,
.dapp-core-component__main__col-sm-9,
.dapp-core-component__main__col-sm-8,
.dapp-core-component__main__col-sm-7,
.dapp-core-component__main__col-sm-6,
.dapp-core-component__main__col-sm-5,
.dapp-core-component__main__col-sm-4,
.dapp-core-component__main__col-sm-3,
.dapp-core-component__main__col-sm-2,
.dapp-core-component__main__col-sm-1,
.dapp-core-component__main__col,
.dapp-core-component__main__col-auto,
.dapp-core-component__main__col-12,
.dapp-core-component__main__col-11,
.dapp-core-component__main__col-10,
.dapp-core-component__main__col-9,
.dapp-core-component__main__col-8,
.dapp-core-component__main__col-7,
.dapp-core-component__main__col-6,
.dapp-core-component__main__col-5,
.dapp-core-component__main__col-4,
.dapp-core-component__main__col-3,
.dapp-core-component__main__col-2,
.dapp-core-component__main__col-1 {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
}

.dapp-core-component__main__col-1 {
  flex: 0 0 8.3333333333%;
  max-width: 8.3333333333%;
}

@media (min-width: 576px) {
  .dapp-core-component__main__col-sm-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-sm-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-sm-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-sm-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-sm-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-sm-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-sm-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-sm-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-sm-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-sm-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-sm-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-sm-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 768px) {
  .dapp-core-component__main__col-md-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-md-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-md-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-md-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-md-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-md-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-md-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-md-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-md-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-md-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-md-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-md-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 992px) {
  .dapp-core-component__main__col-lg-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-lg-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-lg-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-lg-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-lg-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-lg-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-lg-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-lg-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-lg-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-lg-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-lg-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-lg-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
@media (min-width: 1200px) {
  .dapp-core-component__main__col-xl-1 {
    flex: 0 0 8.3333333333%;
    max-width: 8.3333333333%;
  }
  .dapp-core-component__main__col-xl-2 {
    flex: 0 0 16.6666666667%;
    max-width: 16.6666666667%;
  }
  .dapp-core-component__main__col-xl-3 {
    flex: 0 0 25%;
    max-width: 25%;
  }
  .dapp-core-component__main__col-xl-4 {
    flex: 0 0 33.3333333333%;
    max-width: 33.3333333333%;
  }
  .dapp-core-component__main__col-xl-5 {
    flex: 0 0 41.6666666667%;
    max-width: 41.6666666667%;
  }
  .dapp-core-component__main__col-xl-6 {
    flex: 0 0 50%;
    max-width: 50%;
  }
  .dapp-core-component__main__col-xl-7 {
    flex: 0 0 58.3333333333%;
    max-width: 58.3333333333%;
  }
  .dapp-core-component__main__col-xl-8 {
    flex: 0 0 66.6666666667%;
    max-width: 66.6666666667%;
  }
  .dapp-core-component__main__col-xl-9 {
    flex: 0 0 75%;
    max-width: 75%;
  }
  .dapp-core-component__main__col-xl-10 {
    flex: 0 0 83.3333333333%;
    max-width: 83.3333333333%;
  }
  .dapp-core-component__main__col-xl-11 {
    flex: 0 0 91.6666666667%;
    max-width: 91.6666666667%;
  }
  .dapp-core-component__main__col-xl-12 {
    flex: 0 0 100%;
    max-width: 100%;
  }
}
.dapp-core-component__main__flex-row {
  flex-direction: row;
}

.dapp-core-component__main__flex-column {
  flex-direction: column;
}

.dapp-core-component__main__flex-row-reverse {
  flex-direction: row-reverse;
}

.dapp-core-component__main__flex-column-reverse {
  flex-direction: column-reverse;
}

.dapp-core-component__main__flex-wrap {
  flex-wrap: wrap;
}

.dapp-core-component__main__flex-nowrap {
  flex-wrap: nowrap;
}

.dapp-core-component__main__flex-wrap-reverse {
  flex-wrap: wrap-reverse;
}

.dapp-core-component__main__flex-fill {
  flex: 1 1 auto;
}

.dapp-core-component__main__flex-grow-0 {
  flex-grow: 0;
}

.dapp-core-component__main__flex-grow-1 {
  flex-grow: 1;
}

.dapp-core-component__main__flex-shrink-0 {
  flex-shrink: 0;
}

.dapp-core-component__main__flex-shrink-1 {
  flex-shrink: 1;
}

.dapp-core-component__main__justify-content-start {
  justify-content: flex-start;
}

.dapp-core-component__main__justify-content-end {
  justify-content: flex-end;
}

.dapp-core-component__main__justify-content-center {
  justify-content: center;
}

.dapp-core-component__main__justify-content-between {
  justify-content: space-between;
}

.dapp-core-component__main__justify-content-around {
  justify-content: space-around;
}

.dapp-core-component__main__align-items-start {
  align-items: flex-start;
}

.dapp-core-component__main__align-items-end {
  align-items: flex-end;
}

.dapp-core-component__main__align-items-center {
  align-items: center;
}

.dapp-core-component__main__align-items-baseline {
  align-items: baseline;
}

.dapp-core-component__main__align-items-stretch {
  align-items: stretch;
}

.dapp-core-component__main__align-content-start {
  align-content: flex-start;
}

.dapp-core-component__main__align-content-end {
  align-content: flex-end;
}

.dapp-core-component__main__align-content-center {
  align-content: center;
}

.dapp-core-component__main__align-content-between {
  align-content: space-between;
}

.dapp-core-component__main__align-content-around {
  align-content: space-around;
}

.dapp-core-component__main__align-content-stretch {
  align-content: stretch;
}

.dapp-core-component__main__align-self-auto {
  align-self: auto;
}

.dapp-core-component__main__align-self-start {
  align-self: flex-start;
}

.dapp-core-component__main__align-self-end {
  align-self: flex-end;
}

.dapp-core-component__main__align-self-center {
  align-self: center;
}

.dapp-core-component__main__align-self-baseline {
  align-self: baseline;
}

.dapp-core-component__main__align-self-stretch {
  align-self: stretch;
}

@media (min-width: 576px) {
  .dapp-core-component__main__flex-sm-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-row {
    flex-direction: row;
  }
}

@media (min-width: 992px) {
  .dapp-core-component__main__flex-lg-wrap {
    flex-wrap: wrap;
  }
}

.dapp-core-component__main__form-check {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: auto;
  padding-left: 0;
}

.dapp-core-component__main__form-check-label {
  margin-bottom: 0;
}

.dapp-core-component__main__form-check-input {
  position: absolute;
  margin-top: 0.3rem;
  margin-left: -1.25rem;
}
.dapp-core-component__main__form-check-input[disabled] ~ .dapp-core-component__main__form-check-label, .dapp-core-component__main__form-check-input:disabled ~ .dapp-core-component__main__form-check-label {
  color: #6c757d;
}

.dapp-core-component__main__form-group {
  display: flex;
  flex: 0 0 auto;
  flex-flow: row wrap;
  align-items: center;
  margin-bottom: 0;
}

.dapp-core-component__main__form-control {
  display: block;
  width: 100%;
  height: calc(1.5em + 0.75rem + 2px);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  color: #495057;
  background-color: #fff;
  background-clip: padding-box;
  border: 1px solid #ced4da;
  border-radius: 0.25rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.dapp-core-component__main__form-label {
  padding-top: calc(0.375rem + 1px);
  padding-bottom: calc(0.375rem + 1px);
  margin-bottom: 0;
  font-size: inherit;
  line-height: 1.5;
}

input,
button,
select,
optgroup,
textarea {
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
}

.dapp-core-component__main__form-group {
  margin-bottom: 1rem;
}

.dapp-core-component__main__form-text {
  display: block;
  margin-top: 0.25rem;
  color: inherit;
}

.dapp-core-component__main__border {
  border: 1px solid #dee2e6;
}

.dapp-core-component__main__border-top {
  border-top: 1px solid #dee2e6;
}

.dapp-core-component__main__border-right {
  border-right: 1px solid #dee2e6;
}

.dapp-core-component__main__border-bottom {
  border-bottom: 1px solid #dee2e6;
}

.dapp-core-component__main__border-left {
  border-left: 1px solid #dee2e6;
}

.dapp-core-component__main__border-0 {
  border: 0;
}

.dapp-core-component__main__border-top-0 {
  border-top: 0;
}

.dapp-core-component__main__border-right-0 {
  border-right: 0;
}

.dapp-core-component__main__border-bottom-0 {
  border-bottom: 0;
}

.dapp-core-component__main__border-left-0 {
  border-left: 0;
}

.dapp-core-component__main__rounded-sm {
  border-radius: 0.2rem;
}

.dapp-core-component__main__rounded {
  border-radius: 0.25rem;
}

.dapp-core-component__main__rounded-top {
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-right {
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.dapp-core-component__main__rounded-bottom {
  border-bottom-right-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-left {
  border-top-left-radius: 0.25rem;
  border-bottom-left-radius: 0.25rem;
}

.dapp-core-component__main__rounded-lg {
  border-radius: 0.3rem;
}

.dapp-core-component__main__rounded-circle {
  border-radius: 50%;
}

.dapp-core-component__main__rounded-pill {
  border-radius: 50rem;
}

.dapp-core-component__main__rounded-0 {
  border-radius: 0;
}

.dapp-core-component__main__h1,
.dapp-core-component__main__h2,
.dapp-core-component__main__h3,
.dapp-core-component__main__h4,
.dapp-core-component__main__h5,
.dapp-core-component__main__h6 {
  margin-bottom: 0.5rem;
  font-weight: 500;
  line-height: 1.2;
}

.dapp-core-component__main__h1 {
  font-size: 2.5rem;
}

.dapp-core-component__main__h2 {
  font-size: 2rem;
}

.dapp-core-component__main__h3 {
  font-size: 1.75rem;
}

.dapp-core-component__main__h4 {
  font-size: 1.5rem;
}

.dapp-core-component__main__h5 {
  font-size: 1.25rem;
}

.dapp-core-component__main__h6 {
  font-size: 1rem;
}

.dapp-core-component__main__lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

.dapp-core-component__main__bg-warning,
.dapp-core-component__main__warning {
  background-color: #f0ad4e;
}

.dapp-core-component__main__bg-danger,
.dapp-core-component__main__danger {
  background-color: #d9534f;
}

.dapp-core-component__main__bg-success,
.dapp-core-component__main__success {
  background-color: #5cb85c;
}

.dapp-core-component__main__position-relative {
  position: relative;
}

.dapp-core-component__main__position-absolute {
  position: absolute;
}

.dapp-core-component__main__w-25 {
  width: 25%;
}

.dapp-core-component__main__w-50 {
  width: 50%;
}

.dapp-core-component__main__w-75 {
  width: 75%;
}

.dapp-core-component__main__w-100 {
  width: 100%;
}

.dapp-core-component__main__w-auto {
  width: auto;
}

.dapp-core-component__main__h-25 {
  height: 25%;
}

.dapp-core-component__main__h-50 {
  height: 50%;
}

.dapp-core-component__main__h-75 {
  height: 75%;
}

.dapp-core-component__main__h-100 {
  height: 100%;
}

.dapp-core-component__main__h-auto {
  height: auto;
}

.dapp-core-component__main__mw-100 {
  max-width: 100%;
}

.dapp-core-component__main__mh-100 {
  max-height: 100%;
}

.dapp-core-component__main__min-vw-100 {
  min-width: 100vw;
}

.dapp-core-component__main__min-vh-100 {
  min-height: 100vh;
}

.dapp-core-component__main__vw-100 {
  width: 100vw;
}

.dapp-core-component__main__vh-100 {
  height: 100vh;
}

@media (min-width: 768px) {
  .dapp-core-component__main__w-md-auto {
    width: auto;
  }
}

.dapp-core-component__main__locked-icon {
  color: inherit;
}

.dapp-core-component__main__shadow-sm {
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
}

.dapp-core-component__main__shadow {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15);
}

.dapp-core-component__main__shadow-lg {
  box-shadow: 0 1rem 3rem rgba(0, 0, 0, 0.175);
}

.dapp-core-component__main__shadow-none {
  box-shadow: none;
}

.dapp-core-component__main__d-none {
  display: none;
}

.dapp-core-component__main__d-inline {
  display: inline;
}

.dapp-core-component__main__d-inline-block {
  display: inline-block;
}

.dapp-core-component__main__d-block {
  display: block;
}

.dapp-core-component__main__d-table {
  display: table;
}

.dapp-core-component__main__d-table-row {
  display: table-row;
}

.dapp-core-component__main__d-table-cell {
  display: table-cell;
}

.dapp-core-component__main__d-flex {
  display: -ms-flexbox;
  display: flex;
}

.dapp-core-component__main__d-inline-flex {
  display: -ms-inline-flexbox;
  display: inline-flex;
}

@media (min-width: 576px) {
  .dapp-core-component__main__d-sm-flex {
    display: flex;
  }
}

.dapp-core-component__main__table {
  width: 100%;
  margin-bottom: 1rem;
  color: #212529;
}
.dapp-core-component__main__table th,
.dapp-core-component__main__table td {
  padding: 0.75rem;
  vertical-align: top;
  border-top: 1px solid #dee2e6;
}
.dapp-core-component__main__table thead th {
  vertical-align: bottom;
  border-bottom: 2px solid #dee2e6;
}
.dapp-core-component__main__table tbody + tbody {
  border-top: 2px solid #dee2e6;
}

.dapp-core-component__main__table-sm th,
.dapp-core-component__main__table-sm td {
  padding: 0.3rem;
}

.dapp-core-component__main__table-bordered {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered th,
.dapp-core-component__main__table-bordered td {
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__table-bordered thead th,
.dapp-core-component__main__table-bordered thead td {
  border-bottom-width: 2px;
}

.dapp-core-component__main__table-borderless th,
.dapp-core-component__main__table-borderless td,
.dapp-core-component__main__table-borderless thead th,
.dapp-core-component__main__table-borderless tbody + tbody {
  border: 0;
}

.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(0, 0, 0, 0.05);
}

.dapp-core-component__main__table .dapp-core-component__main__thead-dark th {
  color: #fff;
  background-color: #343a40;
  border-color: #454d55;
}
.dapp-core-component__main__table .dapp-core-component__main__thead-light th {
  color: #495057;
  background-color: #e9ecef;
  border-color: #dee2e6;
}

.dapp-core-component__main__table-dark {
  color: #fff;
  background-color: #343a40;
}
.dapp-core-component__main__table-dark th,
.dapp-core-component__main__table-dark td,
.dapp-core-component__main__table-dark thead th {
  border-color: #454d55;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-bordered {
  border: 0;
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-striped tbody tr:nth-of-type(odd) {
  background-color: rgba(255, 255, 255, 0.05);
}
.dapp-core-component__main__table-dark.dapp-core-component__main__table-hover tbody tr {
  color: #fff;
  background-color: rgba(255, 255, 255, 0.075);
}

.dapp-core-component__main__table-responsive {
  display: block;
  width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}
.dapp-core-component__main__table-responsive > .dapp-core-component__main__table-bordered {
  border: 0;
}

@media (max-width: 767.98px) {
  .dapp-core-component__main__transactions.dapp-core-component__main__table .dapp-core-component__main__transaction-hash .dapp-core-component__main__trim {
    max-width: 8rem;
  }
}
.dapp-core-component__main__font-weight-light {
  font-weight: 300;
}

.dapp-core-component__main__font-weight-lighter {
  font-weight: lighter;
}

.dapp-core-component__main__font-weight-normal {
  font-weight: 400;
}

.dapp-core-component__main__font-weight-medium {
  font-weight: 500;
}

.dapp-core-component__main__font-weight-bold {
  font-weight: 700;
}

.dapp-core-component__main__font-weight-bolder {
  font-weight: bolder;
}

.dapp-core-component__main__font-italic {
  font-style: italic;
}

.dapp-core-component__main__icon-state {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
}
.dapp-core-component__main__icon-state.dapp-core-component__main__half {
  width: 3rem;
  height: 3rem;
}

.dapp-core-component__main__side-icon {
  width: 1.5rem;
  height: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  color: transparent;
  border-radius: 50%;
  overflow: hidden;
  border: 1px solid #dee2e6;
}
.dapp-core-component__main__side-icon svg {
  color: #cacbcc;
}

div.dapp-core-component__main__side-icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
div.dapp-core-component__main__side-icon img {
  transform: scale(0.75);
}

.dapp-core-component__main__toast-messages {
  position: fixed;
  bottom: 1rem;
  left: 1rem;
  right: 1rem;
  z-index: 9999;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast {
  display: block;
  flex-basis: 4rem;
  width: 100%;
  max-width: 28.875rem;
  transition: transform 0.3s ease-in-out;
  transform: translateX(120%);
  background-color: #fff;
  border-radius: 0.25rem;
  border-color: #dee2e6;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__clickable {
  cursor: pointer;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible {
  margin-bottom: 0.3rem;
  transform: translateX(0);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress {
  background-color: rgba(0, 0, 0, 0);
  height: inherit;
  font-size: inherit;
  line-height: inherit;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar {
  background-color: rgba(108, 117, 125, 0.2);
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast.dapp-core-component__main__toast-visible .dapp-core-component__main__progress .dapp-core-component__main__progress-bar .dapp-core-component__main__content-height {
  z-index: -1;
  visibility: hidden;
  width: 0;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__toast .dapp-core-component__main__tx-description {
  margin-bottom: 0.2rem;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close {
  opacity: 1;
}
.dapp-core-component__main__toast-messages .dapp-core-component__main__close svg {
  color: #6c757d;
}

.dapp-core-component__main__data-field .dapp-core-component__main__textarea {
  -moz-appearance: textfield-multiline;
  -webkit-appearance: textarea;
  border: 1px solid #000;
  background-color: #fff;
  border-radius: 5px;
  font: medium -moz-fixed;
  font-size: 0.8125rem;
  min-height: 5rem;
  overflow: auto;
  padding: 0.375rem 0.75rem;
  resize: both;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea:hover {
  border: 1px solid #000;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea .dapp-core-component__main__highlighted {
  font-size: 1rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__textarea.dapp-core-component__main__sc-call {
  min-height: 3rem;
}
.dapp-core-component__main__data-field .dapp-core-component__main__form-control:disabled,
.dapp-core-component__main__data-field .dapp-core-component__main__form-control[readonly] {
  background-color: #fafafa;
}
.dapp-core-component__main__data-field .dapp-core-component__main__small-font {
  font-size: 0.8125rem;
}

.dapp-core-component__main__progress-steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps {
  display: flex;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps hr {
  background-color: #6c757d;
  border-color: #dee2e6;
  z-index: 2;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot {
  height: 0.625rem;
  width: 0.625rem;
  border-radius: 50%;
  margin: 0 1rem;
  z-index: 5;
  background-color: #6c757d;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:first-of-type {
  margin-left: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot:last-of-type {
  margin-right: 0;
}
.dapp-core-component__main__progress-steps .dapp-core-component__main__steps .dapp-core-component__main__dot.dapp-core-component__main__bg-primary {
  background-color: #1b46c2;
}

a.dapp-core-component__main__side-link::after {
  right: 0;
  left: unset;
  width: calc(100% - 1.85rem);
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Ji));n();n();var Kr=require("react"),Ce=require("react-redux");n();var En=require("@reduxjs/toolkit"),zr=require("react-redux/lib/utils/Subscription");I();n();var Sr=h(require("lodash.throttle"));v();n();var br=h(require("lodash.isequal")),ko=require("reselect");C();var g=(0,ko.createSelectorCreator)(ko.defaultMemoize,br.default),He=(e,o)=>e!=""&&Boolean(o);I();gn();ve();we();var Rc=[to],Io=!1,Mc=(0,Sr.default)(()=>{We(Ue())},5e3),wr=e=>o=>t=>{var _;if(Rc.includes(t.type))return o(t);let r=e.getState(),s=le.local.getItem(ae.loginExpiresAt);if(!He((_=r==null?void 0:r.loginInfo)==null?void 0:_.loginMethod,r==null?void 0:r.account.address))return o(t);if(s==null)return We(Ue());let m=Date.now();return s-m<0&&!Io?setTimeout(()=>{Io=!0,e.dispatch(Jt())},1e3):(Io&&(Io=!1),Mc()),o(t)};n();n();function Ao(){return typeof sessionStorage!="undefined"}var Hr=Ao()?(Rr(),Oe(Dr)).default:(Or(),Oe(Mr)).default,Vr=Ao()?(Br(),Oe(Fr)).default:[],jr=Ao()?($r(),Oe(Gr)).default:e=>e;Lo();var S=(0,En.configureStore)({reducer:Hr,middleware:e=>e({serializableCheck:{ignoredActions:[...Vr,ln.type,_n.type],ignoredPaths:["payload.nonce","account.account.nonce","providers.accountProvider"]}}).concat(wr)}),qr=(0,zr.createSubscription)(S),Ng=jr(S),Dg=(0,En.configureStore)({reducer:fe});var ts={store:S,subscription:qr},Cn=(0,Kr.createContext)(ts),Fg=(0,Ce.createStoreHook)(Cn),F=(0,Ce.createDispatchHook)(Cn),k=(0,Ce.createSelectorHook)(Cn);n();n();I();var oe=e=>e.account,Pe=g(oe,e=>e.address),je=g(oe,Pe,(e,o)=>o in e.accounts?e.accounts[o]:ao),rs=g(oe,je,(e,o)=>{let s=e,{accounts:t}=s,r=Me(s,["accounts"]);return w(u({},r),{account:o})}),as=g(je,e=>e.balance),is=g(je,e=>{var o;return((o=e==null?void 0:e.nonce)==null?void 0:o.valueOf())||0}),Gg=g(oe,e=>e.shard),cs=g(oe,e=>e.ledgerAccount),$g=g(oe,e=>e.walletConnectAccount),Hg=g(oe,e=>e.isAccountLoading),Vg=g(oe,e=>e.accountLoadingError),Xr=g(oe,e=>e.websocketEvent);n();var V=e=>e.loginInfo,Kg=g(V,e=>e.loginMethod),Pn=g(V,Pe,(e,o)=>He(e.loginMethod,o)),Xg=g(V,e=>e.walletConnectLogin),Yg=g(V,e=>e.ledgerLogin),ss=g(V,e=>e.walletLogin),Zg=g(V,e=>e.isLoginSessionInvalid),ps=g(V,e=>e.tokenLogin),Jg=g(V,e=>e.logoutRoute);n();var Yr=e=>e.modals,ou=g(Yr,e=>e.txSubmittedModal),ms=g(Yr,e=>e.notificationModal);n();var ne=e=>e.networkConfig,Eo=g(ne,e=>e.chainID),ds=g(ne,e=>e.network.walletConnectBridgeAddress),_s=g(ne,e=>e.network.walletConnectV2RelayAddress),ls=g(ne,e=>e.network.walletConnectV2ProjectId),gs=g(ne,e=>e.network.walletConnectV2Options),Zr=g(ne,e=>e.network.walletConnectDeepLink),ce=g(ne,e=>e.network),Jr=g(ce,e=>e.apiAddress),Qr=g(ce,e=>e.explorerAddress),ea=g(ce,e=>e.egldLabel);n();var Co=e=>e.signedMessageInfo,iu=g(Co,e=>e.isSigning),cu=g(Co,e=>e.errorMessage),su=g(Co,e=>{let o=Object.keys(e.signedSessions),t=o.length;return e.signedSessions[o[t-1]]}),pu=g(Co,e=>{let o=Object.keys(e.signedSessions),t=o.length;return o.length>0?o[t-1]:""});n();var Nn=e=>e.toasts,_u=g(Nn,e=>e.customToasts),lu=g(Nn,e=>e.transactionToasts),gu=g(Nn,e=>e.failTransactionToast);n();I();var oa={errorMessage:bn,successMessage:Sn,processingMessage:wn},us=e=>e.transactionsInfo,fs=g(us,(e,o)=>o,(e,o)=>o!=null&&(e==null?void 0:e[Number(o)])||oa);n();n();var G=require("@multiversx/sdk-core");v();n();function Dn(e){try{return i.Buffer.from(e,"base64").toString()===atob(e)}catch(o){return!1}}function Po(e){var s,p,m;let{data:o}=e,t=Dn(o!=null?o:"")?G.TransactionPayload.fromEncoded(o):new G.TransactionPayload(o),r=new G.Transaction(u({value:e.value.valueOf(),data:t,nonce:e.nonce.valueOf(),receiver:new G.Address(e.receiver),sender:new G.Address(e.sender),gasLimit:(s=e.gasLimit.valueOf())!=null?s:Fe,gasPrice:(p=e.gasPrice.valueOf())!=null?p:no,chainID:e.chainID.valueOf(),version:new G.TransactionVersion((m=e.version)!=null?m:Ot)},e.options?{options:new G.TransactionOptions(e.options)}:{}));return r.applySignature({hex:()=>e.signature||""},new G.Address(e.sender)),r}n();n();n();n();n();var Ts=require("@multiversx/sdk-core/out");n();v();n();var na=require("@multiversx/sdk-core");C();n();var xs=require("@multiversx/sdk-core");n();n();n();var bs=require("@multiversx/sdk-core"),Ss=h(require("bignumber.js"));A();n();n();n();n();n();var Mn="blocks";n();n();n();n();O();n();n();O();n();n();n();var Cs=h(require("bignumber.js"));v();n();var Es=require("@multiversx/sdk-core"),On=h(require("bignumber.js"));v();n();var As=h(require("bignumber.js"));n();On.default.config({ROUNDING_MODE:On.default.ROUND_FLOOR});n();n();n();v();n();n();n();n();v();n();v();n();var Ps=require("@multiversx/sdk-core");v();n();var ze=require("@multiversx/sdk-core"),Rs=require("@multiversx/sdk-network-providers");v();n();n();var Ns=h(require("bignumber.js"));v();n();v();n();n();n();n();n();v();n();O();var Os=["reDelegateRewards","claimRewards","unBond"],Fs=["wrapEgld","unwrapEgld"],Bs=["unStake"],Us=["unDelegate"];n();n();n();v();n();var Hs=h(require("bignumber.js"));n();n();n();n();O();n();var Vs=h(require("bignumber.js"));n();n();n();var Ks=h(require("bignumber.js"));C();n();n();n();n();A();n();n();n();n();O();n();n();n();var Xs=h(require("linkifyjs"));n();v();n();var Ys=h(require("bignumber.js"));n();C();n();n();A();n();A();n();n();n();O();n();O();n();var Zs=h(require("bignumber.js"));v();O();n();O();n();var _a=require("react");C();n();n();O();n();n();var Js=require("@multiversx/sdk-core/out"),Qs=h(require("bignumber.js"));O();n();C();n();n();C();var Ky=[{label:"Raw",value:"raw"},{label:"Text",value:"text"},{label:"Decimal",value:"decimal"},{label:"Smart",value:"smart"}];n();var la=require("react");C();n();var op=require("react");O();var ob=["create","localMint","ESDTLocalMint","addQuantity","burn","localBurn","ESDTLocalBurn","wipe","writeLog","signalError"];Ee();var Mo=e=>e.transactions,Ne=g(Mo,e=>e.signedTransactions),np=g(Mo,e=>e.signTransactionsError),ga=g(Mo,e=>e.signTransactionsCancelMessage),Oo=e=>o=>Object.entries(o).reduce((t,[r,s])=>(e(s.status)&&(t[r]=s),t),{}),ua=g(Ne,Oo(Ie)),fa=g(Ne,Oo(Ae)),ha=g(Ne,Oo(Le)),tp=g(Ne,Oo(kn)),Ta=g(Mo,e=>{var o;return(e==null?void 0:e.transactionsToSign)==null?null:w(u({},e.transactionsToSign),{transactions:((o=e==null?void 0:e.transactionsToSign)==null?void 0:o.transactions.map(t=>Po(t)))||[]})}),rp=g(Ne,(e,o)=>o,(e,o)=>o!=null?(e==null?void 0:e[o])||{}:{});n();var ap=`.dapp-core-component__explorerLinkStyles__link .dapp-core-component__explorerLinkStyles__search {
  color: #6c757d;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(ap));n();n();var $o=h(require("react")),Wm=h(require("classnames"));v();n();n();var ya=require("react");n();I();n();n();n();n();n();n();n();n();var lp=require("@multiversx/sdk-extension-provider"),gp=require("@multiversx/sdk-hw-provider"),up=require("@multiversx/sdk-opera-provider"),fp=require("@multiversx/sdk-web-wallet-provider");v();A();n();var te=e=>`Unable to perform ${e}, Provider not initialized`,Fo=class{init(){return Promise.resolve(!1)}login(o){throw new Error(te(`login with options: ${o}`))}logout(o){throw new Error(te(`logout with options: ${o}`))}getAddress(){throw new Error(te("getAddress"))}isInitialized(){return!1}isConnected(){return Promise.resolve(!1)}sendTransaction(o,t){throw new Error(te(`sendTransaction with transactions: ${o} options: ${t}`))}signTransaction(o,t){throw new Error(te(`signTransaction with transactions: ${o} options: ${t}`))}signTransactions(o,t){throw new Error(te(`signTransactions with transactions: ${o} options: ${t}`))}signMessage(o,t){throw new Error(te(`signTransactions with ${o} and options ${t}`))}sendCustomMessage({method:o,params:t}){throw new Error(te(`sendCustomMessage with method: ${o} params: ${t}`))}sendCustomRequest(o){throw new Error(te(`sendSessionEvent with options: ${o}`))}ping(){return Promise.resolve(!1)}},hp=new Fo;I();n();n();var Tp=h(require("axios"));n();n();n();A();n();n();n();Ee();n();n();n();var xp=h(require("axios"));n();var yp=h(require("axios"));n();var Sp=h(require("axios"));n();n();var wp=h(require("axios"));n();var vp=h(require("axios"));n();I();A();n();n();I();n();n();var va=require("@multiversx/sdk-core");n();N();C();n();Ee();n();I();A();n();I();C();n();n();n();C();n();n();n();n();var Ap=h(require("axios")),Lp=h(require("swr"));n();n();var ka=require("react");n();var Pp=require("react"),La=require("@multiversx/sdk-web-wallet-provider"),Np=h(require("qs"));v();I();A();n();n();var Ia,hv={search:(Ia=window==null?void 0:window.location)==null?void 0:Ia.search,removeParams:[]};n();var Ea=require("react");fn();n();var Da=require("react"),Gp=require("@multiversx/sdk-extension-provider");v();n();I();A();n();n();n();n();var Na=require("react"),Fp=require("@multiversx/sdk-extension-provider");v();n();n();n();n();n();n();n();n();var qe=h(require("react"));var sk=(0,qe.createContext)({}),pk=S.getState();n();var Pa=h(require("react"));n();var Gn=h(require("react")),Rp=h(require("axios"));n();n();I();n();n();var $p=h(require("axios"));I();C();Ge();n();_e();I();n();n();var Sm=require("react");n();n();n();I();n();n();var jp=h(require("bignumber.js"));v();I();C();n();var Vp=h(require("bignumber.js"));v();n();var zp=require("@multiversx/sdk-core"),qp=h(require("bignumber.js"));v();n();n();n();n();var Oa=h(require("axios"));n();n();n();function Ra(e){return y(this,null,function*(){return yield new Promise(o=>{setTimeout(()=>o(),e)})})}n();var Ma=(e,o,t,r=0)=>y(void 0,null,function*(){try{return yield e(...t)}catch(s){return r<o.retries?((o==null?void 0:o.delay)!=null&&(yield Ra(o.delay)),yield Ma(e,o,t,r+1)):null}}),Hn=(e,o={retries:5,delay:500})=>(...t)=>y(void 0,null,function*(){return yield Ma(e,o,t)});var Kp=4;var QI=Hn((e,o)=>y(void 0,null,function*(){let{data:t}=yield Oa.default.get(`${e}/${Mn}?from=${Kp}&size=1&fields=hash,timestamp${o?"&shard="+o:""}`),[r]=t;return r}));n();n();ro();n();n();var xm=h(ti());n();var ri,bA={origin:typeof window!="undefined"?(ri=window==null?void 0:window.location)==null?void 0:ri.origin:"",apiAddress:"https://api.multiversx.com",expirySeconds:60*60*24,tokenExpirationToastWarningSeconds:5*60};n();A();Ee();n();var ai=require("react");n();n();var vm=require("react"),km=require("@multiversx/sdk-extension-provider");_e();N();A();n();ge();n();var wm=require("react"),ii=require("@multiversx/sdk-core");I();n();var Im=require("react"),Am=require("@multiversx/sdk-opera-provider");_e();N();A();ge();n();var Cm=require("react");_e();I();n();n();rn();n();n();var Lm=h(require("platform"));n();N();C();Ge();ve();we();n();n();n();n();n();n();n();n();n();A();n();Ge();ve();n();n();n();I();n();n();var Em=h(require("axios"));n();n();var ci=require("react"),Nm=require("@multiversx/sdk-hw-provider");_e();N();I();A();ge();n();var Pm=require("react");n();var zn=require("react");N();I();A();ge();n();var qn=require("react");N();I();A();ge();n();n();var si=require("react");n();var Fm=require("react");n();var Kn=require("react"),Om=require("socket.io-client");I();n();n();n();n();var pi=e=>e.dappModal,Bm=g(pi,e=>e.visible),Go=g(pi,e=>e.modal),n3=g(Go,e=>e.showHeader),t3=g(Go,e=>e.showFooter),r3=g(Go,e=>e.headerText),a3=g(Go,e=>e.footerText);n();var Um=`.dapp-core-component__trim-styles__trim {
  display: flex;
  max-width: 100%;
  overflow: hidden;
  position: relative;
  white-space: nowrap;
  /* IE fix */
  /* SAFARI fix */
}
.dapp-core-component__trim-styles__trim.dapp-core-component__trim-styles__overflow .dapp-core-component__trim-styles__ellipsis {
  display: block;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: left;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
  flex-shrink: 1;
  font-size: 1px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  direction: rtl;
  text-align: right;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span {
  font-size: 0.875rem;
  pointer-events: none;
  user-select: none;
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  flex-shrink: 0;
  display: none;
  pointer-events: none;
  user-select: none;
}
@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
@supports (-webkit-hyphens: none) {
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right,
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left {
    letter-spacing: -0.001em;
  }
  .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right {
    text-overflow: clip;
  }
}
.dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__hidden-text-ref {
  position: absolute;
  display: block;
  color: transparent;
}
@media (max-width: 1199.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 26rem;
  }
}
@media (max-width: 991.98px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 12rem;
  }
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__trim {
    max-width: 8rem;
  }
}

.dapp-core-component__trim-styles__trim-wrapper {
  display: flex;
  max-width: 100%;
  overflow: hidden;
}

a:hover > .dapp-core-component__trim-styles__trim span {
  color: #0056b3;
}
a:hover > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent;
}

a > .dapp-core-component__trim-styles__trim span,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span {
  color: #1b46c2;
}
a > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref,
.dapp-core-component__trim-styles__text-primary > .dapp-core-component__trim-styles__trim span.dapp-core-component__trim-styles__hidden-text-ref {
  color: transparent;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim {
  max-width: 10rem;
}

.dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
  max-width: none;
}
@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-only-sm .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}

.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__left span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__right span,
.dapp-core-component__trim-styles__trim-fs-sm .dapp-core-component__trim-styles__trim .dapp-core-component__trim-styles__ellipsis {
  font-size: 0.875rem;
}

@media (max-width: 768px) {
  .dapp-core-component__trim-styles__table .dapp-core-component__trim-styles__trim-size-xl .dapp-core-component__trim-styles__trim {
    max-width: 13rem;
  }
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Um));n();var $m=h(require("react")),Hm=h(require("classnames"));n();var Gm=`.dapp-core-component__styles__scResultWrapper {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-bottom: 1rem;
}
@media (min-width: 576px) {
  .dapp-core-component__styles__scResultWrapper {
    flex-direction: row;
  }
}
.dapp-core-component__styles__scResultWrapper .dapp-core-component__styles__label {
  color: #6c757d;
}
@media (min-width: 576px) {
  .dapp-core-component__styles__scResultWrapper .dapp-core-component__styles__label {
    text-align: right;
  }
}
.dapp-core-component__styles__scResultWrapper .dapp-core-component__styles__data {
  display: flex;
  align-items: center;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Gm));n();var Vm=`.dapp-core-component__styles__scResultHash .dapp-core-component__styles__hash {
  font-size: 1rem;
  align-items: flex-end;
  max-width: none;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__hash [class*=left] span,
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__hash [class*=right] span {
  font-size: 1rem;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__copy {
  margin-left: 0.5rem;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__copy svg path {
  fill: #cacbcc;
  transition: all 200ms;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__copy svg path:hover {
  fill: #1f43f4;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__explorer {
  color: #cacbcc;
  transition: all 200ms;
}
.dapp-core-component__styles__scResultHash .dapp-core-component__styles__explorer:hover {
  color: #1f43f4;
  text-decoration: none;
}`;document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(Vm));
/*!
 * The buffer module from node.js, for the browser.
 *
 * @author   Feross Aboukhadijeh <https://feross.org>
 * @license  MIT
 */
/*! ieee754. BSD-3-Clause License. Feross Aboukhadijeh <https://feross.org/opensource> */
